---
title: "Chemical phenotyping maize root bacteria"
subtitle: "Thoenen_et_al_BX_tolerance"
author: "Lisa Thoenen"
date: "`r Sys.Date()`"
output:
  html_document:
    fig_caption: yes
    fig_height: 10
    fig_width: 10
    toc: yes
    toc_float: yes
  pdf_document:
    toc: yes
editor_options:
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r, include=FALSE}
rm(list = ls())
```

```{r, warning=FALSE, include=FALSE}
# install.packages("MESS")
# #Plotly to annotate plots and explore them
# #install.packages("plotly")
# install.packages("metacoder")
# 
# #lme4 and lmertest for stats, multcomp and lsmeans for posthoc
# install.packages("lme4")
# install.packages("lmerTest")
# install.packages("multcomp")
# install.packages("emmeans")
# install.packages("magrittr")
# install.packages("readxl")
# install.packages("lme4")
# install.packages("lmerTest")
# install.packages("multcomp")
# install.packages("emmeans")
# 
# #Formatting and reading tools
# install.packages("magrittr")
# install.packages("tidyverse")
# install.packages("readxl")
# install.packages("lubridate")
# install.packages("broom")
# install.packages("dplyr")
# install.packages("tidyr")
# install.packages("stringr")
# install.packages("readr")
# install.packages("purrr")
# 
# #Plotstuff
# install.packages("ggthemes")
# install.packages("ggforce")
# install.packages("KEGGREST")
# install.packages("randomForest")
# install.packages("ggplot2")
# install.packages("rstatix")
# install.packages("ggpmisc")
# install.packages("ggbeeswarm")
# install.packages("gtable")
# install.packages("grid")
```

```{r, warning=FALSE, include=FALSE}
#MESS for AUC function
library("MESS")
#Plotly to annotate plots and explore them
#library("plotly")

library("metacoder")

#lme4 and lmertest for stats, multcomp and lsmeans for posthoc
library("lme4")
library("lmerTest")
library("multcomp")
library("emmeans")
#Formatting and reading tools
library("magrittr")
#library("tidyverse")
library("readxl")
library("lubridate")
library("broom")
library("dplyr")
library("tidyr")
library("stringr")
library("readr")
library("purrr")
library("rstatix")
### the dplyr package gets overwritten by other packages so often it is necessary to load it specifically again when needed for exapmle like that: dplyr::select
#Plotstuff
library("ggthemes")
library("ggforce")
#library("KEGGREST")
#library("randomForest")
library("ggpubr")
library("ggpmisc")
library("ggbeeswarm")
library("gtable")
library("grid")
```

```{r, include=FALSE}
# For plotting to show only one asterik instead of multiple for given p-values. 
symnum.args <- list(cutpoints = c(0, 0.05, 1), symbols = c("*", ""))
```

# I) MAIZE ROOT BACTERIA

**Load metadata**: Metadata strains with plate layout (indicating which strain is in which well) and database with taxonomic information about the strains. 

```{r, include=FALSE}
Strain_Data_raw <- read_excel("Input/DESIGN_MBOA_Pheno_60strains_RUN1_RUN2_RUN3_RUN4.xlsx", sheet = 4)

Strain_Data_raw$Strain <- gsub("LPD2_E2", "LPD2", Strain_Data_raw$Strain)
Strain_Data_raw$Strain <- gsub("LPD1", "LPD11", Strain_Data_raw$Strain)
Strain_Data_raw$Strain <- gsub("LPE1.3.1", "LPE13", Strain_Data_raw$Strain)

database <- read_excel("Input/__Database_MRB_isolates_sequences.xlsx")
Strain_Data <- left_join(Strain_Data_raw, database, by = "Strain") %>% dplyr::select(Strain, Column, Row, Well, Replicate, Plate, Phylum, Class, Order, Family, Genus, ZOTU, Morphology_MBOA)

```

```{r, include=FALSE}
Strain_Data$Family <- gsub("Pseudomonadales", "Pseudomonadaceae", Strain_Data$Family)
Strain_Data$Family <- gsub("Chitinophagales", "Chitinophagaceae", Strain_Data$Family)

Strain_Data %<>%
  mutate(Family = case_when(Strain == "LAC11" ~ "Moraxellaceae",
                            Strain != "LAC11" ~ Family))


Strain_Data$Well_Plate <- interaction(Strain_Data$Well, Strain_Data$Plate, sep = "_")
```

**Load growth data**: Growth data from the plates. 

```{r, include=FALSE}
results1 <- data_frame()

# Provide a lookup table for plate information, treatments are spread accross sheets in the table to be read.
# in the first column describe the plates and in the second column the experiment (rep(times = # plates - 2))
treatlist <- data_frame("Treat" =
              c("Dummy",
                "media",
                "5000_uM_MBOA_TSB_B",
                "2500_uM_MBOA_TSB_B",
                "1250_uM_MBOA_TSB_B",
                "625_uM_MBOA_TSB_B",
                "500_uM_MBOA_TSB_B",
                "250_uM_MBOA_TSB_B",
                "0_uM_DMSO_TSB_B",
                "5000_uM_MBOA_TSB_A",
                "2500_uM_MBOA_TSB_A",
                "1250_uM_MBOA_TSB_A",
                "625_uM_MBOA_TSB_A",
                "500_uM_MBOA_TSB_A",
                "250_uM_MBOA_TSB_A",
                "0_uM_DMSO_TSB_A",
                "Dummy"),
  "Run" = c("NULL", rep("R3", times = 15), "NULL"), 
  "Plate" = c("NULL", "NULL", rep("PlateB", times = 7), rep("PlateA", times = 7), "NULL"),
  "Media" = c("NULL", "NULL", rep ("TSB", times = 14), "NULL"),
  "Chem" = c("NULL", "media", rep("MBOA", times = 6), "DMSO", rep("MBOA", times = 6), "DMSO", "NULL"),
  "Conc" = c("NULL", "NULL",  rep(c("5000", "2500", "1250", "625", "500", "250", "0"), times = 2), "NULL"))

#Initialize / clear df

#Loop through sheets.

for (i in 1:nrow(treatlist)) {
  tmp <- read_excel("Input/60Strains_screen_Run3_211021.xlsx", sheet = i, range = "B26:CU69") %>%
    as.data.frame() %>% 
    dplyr::select(-2) %>% # Drop Temperature
    pivot_longer(names_to = "Well", values_to = "Density", cols = 3:97) %>%  #Make long
    # pivot_longer(key = "Well", value = "Density", 2:ncol(.)) %>%
    mutate(Treat = as.character(treatlist[i,1]), #Add treatment
    Run = as.character(treatlist[i,2]),
    Plate = as.character(treatlist[i,3]),
    Media = as.character(treatlist[i,4]),
    Chem = as.character(treatlist[i,5]),
    Conc = as.character(treatlist[i,6]),
    Well_Plate = interaction(Well, Plate, sep = "_")) #Add Replicate
results1 <- bind_rows(results1, tmp)
}
```

```{r, include=FALSE}
results2 <- data_frame()

# Provide a lookup table for plate information, treatments are spread accross sheets in the table to be read.
# in the first column describe the plates and in the second column the experiment (rep(times = # plates - 2))
treatlist <- data_frame("Treat" =
              c("Dummy",
                "Ctrl_1_DMSOH_TSB_B",
                "Ctrl_1_DMSOH_DMSO_TSB_A",
                "50_AMPO_TSB_B",
                "50_AMPO_TSB_A",
                "25_AMPO_TSB_B",
                "25_AMPO_TSB_A",
                "10_AMPO_TSB_B",
                "10_AMPO_TSB_A",
                "media",
                "5000_1_MBOA_TSB_B",
                "2500_1_MBOA_TSB_B",
                "1250_1_MBOA_TSB_B",
                "625_1_MBOA_TSB_B",
                "500_1_MBOA_TSB_B",
                "250_1_MBOA_TSB_B",
                "Ctrl_1_DMSO_TSB_B",
                "5000_1_MBOA_TSB_A",
                "2500_1_MBOA_TSB_A",
                "1250_1_MBOA_TSB_A",
                "625_1_MBOA_TSB_A",
                "500_1_MBOA_TSB_A",
                "250_1_MBOA_TSB_A",
                "Ctrl_1_DMSO_TSB_A",
                "Dummy"),
"Run" = c("NULL", rep("R4", times = 23), "NULL"), # times = all plates but without the dummy plates
"Plate" = c("NULL", rep(c("PlateB", "PlateA"), times = 4), "NULL", rep("PlateB", times = 7), 
            rep("PlateA", times = 7), "NULL"),
"Media" = c("NULL", rep ("TSB", times = 23), "NULL"),
"Chem" = c("NULL", rep("DMSOH", times = 2), rep("AMPO", times = 6), "media", rep("MBOA", times = 6), 
           "DMSO", rep("MBOA", times = 6), "DMSO", "NULL"), 
"Conc" = c("NULL",  rep("0", times = 2), rep("50", times = 2), rep("25", times = 2), rep("10", times = 2), "NULL", rep(c("5000", "2500", "1250", "625", "500", "250", "0"), times = 2), "NULL"))

#Initialize / clear df

#Loop through sheets.

for (i in 1:nrow(treatlist)) {
  tmp <- read_excel("Input/60Strains_screen_Run4_211210_start1.xlsx", sheet = i, range = "B26:CU27") %>%   # change for duration runs
    dplyr::select(-2) %>% # Drop Temperature
    pivot_longer(names_to = "Well", values_to = "Density", cols = 3:97) %>%  #Make long
    mutate(Treat = as.character(treatlist[i,1]), #Add treatment
    Run = as.character(treatlist[i,2]),
    Plate = as.character(treatlist[i,3]),
    Media = as.character(treatlist[i,4]),
    Chem = as.character(treatlist[i,5]),
    Conc = as.character(treatlist[i,6]),
    Well_Plate = interaction(Well, Plate, sep = "_")) #Add Replicate
results2 <- bind_rows(results2, tmp)
}
```

```{r, include=FALSE}
results3 <- data_frame()

# Provide a lookup table for plate information, treatments are spread accross sheets in the table to be read.
# in the first column describe the plates and in the second column the experiment (rep(times = # plates - 2))
treatlist <- data_frame("Treat" =
              c("Dummy",
                "Ctrl_1_DMSOH_TSB_B",
                "Ctrl_1_DMSOH_DMSO_TSB_A",
                "50_AMPO_TSB_B",
                "50_AMPO_TSB_A",
                "25_AMPO_TSB_B",
                "25_AMPO_TSB_A",
                "10_AMPO_TSB_B",
                "10_AMPO_TSB_A",
                "media",
                "5000_1_MBOA_TSB_B",
                "2500_1_MBOA_TSB_B",
                "1250_1_MBOA_TSB_B",
                "625_1_MBOA_TSB_B",
                "500_1_MBOA_TSB_B",
                "250_1_MBOA_TSB_B",
                "Ctrl_1_DMSO_TSB_B",
                "5000_1_MBOA_TSB_A",
                "2500_1_MBOA_TSB_A",
                "1250_1_MBOA_TSB_A",
                "625_1_MBOA_TSB_A",
                "500_1_MBOA_TSB_A",
                "250_1_MBOA_TSB_A",
                "Ctrl_1_DMSO_TSB_A",
                "Dummy"),
"Run" = c("NULL", rep("R4", times = 23), "NULL"), # times = all plates but without the dummy plates
"Plate" = c("NULL", rep(c("PlateB", "PlateA"), times = 4), "NULL", rep("PlateB", times = 7), 
            rep("PlateA", times = 7), "NULL"),
"Media" = c("NULL", rep ("TSB", times = 23), "NULL"),
"Chem" = c("NULL", rep("DMSOH", times = 2), rep("AMPO", times = 6), "media", rep("MBOA", times = 6), 
           "DMSO", rep("MBOA", times = 6), "DMSO", "NULL"), 
"Conc" = c("NULL",  rep("0", times = 2), rep("50", times = 2), rep("25", times = 2), rep("10", times = 2), "NULL", rep(c("5000", "2500", "1250", "625", "500", "250", "0"), times = 2), "NULL"))


#Initialize / clear df

#Loop through sheets.

for (i in 1:nrow(treatlist)) {
  tmp <- read_excel("Input/60Strains_screen_Run4_211210_start2.xlsx", sheet = i, range = "B26:CU68") %>%   # change for duration runs
    dplyr::select(-2) %>% # Drop Temperature
    pivot_longer(names_to = "Well", values_to = "Density", cols = 3:97) %>%  #Make long
    mutate(Treat = as.character(treatlist[i,1]), #Add treatment
    Run = as.character(treatlist[i,2]),
    Plate = as.character(treatlist[i,3]),
    Media = as.character(treatlist[i,4]),
    Chem = as.character(treatlist[i,5]),
    Conc = as.character(treatlist[i,6]),
    Well_Plate = interaction(Well, Plate, sep = "_")) #Add Replicate
results3 <- bind_rows(results3, tmp)
}
```

```{r, include=FALSE}
results4 <- data_frame()

# Provide a lookup table for plate information, treatments are spread accross sheets in the table to be read.
# in the first column describe the plates and in the second column the experiment (rep(times = # plates - 2))
treatlist <- data_frame("Treat" =
              c("Dummy",
                "Ctrl_DMSO_TSB_A",
                "250_BOA_TSB_A",
                "500_BOA_TSB_A",
                "625_BOA_TSB_A",
                "1250_BOA_TSB_A",
                "2500_BOA_TSB_A", 
                "5000_BOA_TSB_A",
                "Ctrl_DMSO_TSB_B",
                "250_BOA_TSB_B",
                "500_BOA_TSB_B",
                "625_BOA_TSB_B",
                "1250_BOA_TSB_B",
                "2500_BOA_TSB_B",
                "5000_BOA_TSB_B",
                "media",
                "10_AMPO_TSB_A",
                "10_AMPO_TSB_B",
                "25_AMPO_TSB_A",
                "25_AMPO_TSB_B",
                "50_AMPO_TSB_A",
                "50_AMPO_TSB_B",
                "H_DMSO_TSB_A",
                "H_DMSO_TSB_B",
                "Dummy"),
  "Run" = c("NULL", rep("R1", times = 23), "NULL"), # times = all plates but without the dummy plates
  "Plate" = c("NULL", rep("PlateA", times = 7), rep("PlateB", times = 7), "NULL", rep(c("PlateA", "PlateB"), times = 4), "NULL"),
  "Media" = c("NULL", rep ("TSB", times = 23), "NULL"),
  "Chem" = c("NULL", "DMSO", rep("BOA", times = 6), "DMSO", rep("BOA", times = 6), "NULL", rep("AMPO", times = 6), rep("DMSO", times = 2), "NULL"),
  "Conc" = c("NULL", rep(c("0", "250", "500", "625", "1250", "2500", "5000"), times = 2), "NULL", c("10", "10", "25", "25", "50", "50", "H", "H", "NULL")))


#Initialize / clear df

#Loop through sheets.

for (i in 1:nrow(treatlist)) {
  tmp <- read_excel("Input/220221_screen60_MRB_BOA_AMPO.xlsx", sheet = i, range = "B26:CU68") %>%   # change for duration runs
    dplyr::select(-2) %>% # Drop Temperature
    pivot_longer(names_to = "Well", values_to = "Density", cols = 3:97) %>%  #Make long
    mutate(Treat = as.character(treatlist[i,1]), #Add treatment
    Run = as.character(treatlist[i,2]),
    Plate = as.character(treatlist[i,3]),
    Media = as.character(treatlist[i,4]),
    Chem = as.character(treatlist[i,5]),
    Conc = as.character(treatlist[i,6]),
    Well_Plate = interaction(Well, Plate, sep = "_")) #Add Replicate
results4 <- bind_rows(results4, tmp)
}
```

```{r, include=FALSE}
results5 <- data_frame()

# Provide a lookup table for plate information, treatments are spread accross sheets in the table to be read.
# in the first column describe the plates and in the second column the experiment (rep(times = # plates - 2))
treatlist <- data_frame("Treat" =
              c("Dummy",
                "Ctrl_DMSO_TSB_A",
                "250_BOA_TSB_A",
                "500_BOA_TSB_A",
                "625_BOA_TSB_A",
                "1250_BOA_TSB_A",
                "2500_BOA_TSB_A", 
                "5000_BOA_TSB_A",
                "Ctrl_DMSO_TSB_B",
                "250_BOA_TSB_B",
                "500_BOA_TSB_B",
                "625_BOA_TSB_B",
                "1250_BOA_TSB_B",
                "2500_BOA_TSB_B",
                "5000_BOA_TSB_B",
                "media",
                "2500_DMG_TSB_A",
                "2500_DMG_TSB_B",
                "500_DMG_TSB_A",
                "500_DMG_TSB_B",
                "50_MBOA_TSB_A",
                "50_MBOA_TSB_B",
                "50_MBOA_TSB_A",
                "50_MBOA_TSB_B",
                "Dummy"),
  "Run" = c("NULL", rep("R2", times = 23), "NULL"), # times = all plates but without the dummy plates
  "Plate" = c("NULL", rep("PlateA", times = 7), rep("PlateB", times = 7), "NULL", rep(c("PlateA", "PlateB"), times = 4), "NULL"),
  "Media" = c("NULL", rep ("TSB", times = 23), "NULL"),
  "Chem" = c("NULL", "DMSO", rep("BOA", times = 6), "DMSO", rep("BOA", times = 6), "NULL", rep("DMG", times = 4), rep("MBOA", times = 4), "NULL"),
  "Conc" = c("NULL", rep(c("0", "250", "500", "625", "1250", "2500", "5000"), times = 2), "NULL", c("2500", "2500", "500", "500", "50", "50", "50", "50", "NULL")))


#Initialize / clear df

#Loop through sheets.

for (i in 1:nrow(treatlist)) {
  tmp <- read_excel("Input/220224_Screen_60_MRB_BOA_DMG_MBOlow.xlsx", sheet = i, range = "B26:CU68") %>%   # change for duration runs
    dplyr::select(-2) %>% # Drop Temperature
    pivot_longer(names_to = "Well", values_to = "Density", cols = 3:97) %>%  #Make long
    mutate(Treat = as.character(treatlist[i,1]), #Add treatment
    Run = as.character(treatlist[i,2]),
    Plate = as.character(treatlist[i,3]),
    Media = as.character(treatlist[i,4]),
    Chem = as.character(treatlist[i,5]),
    Conc = as.character(treatlist[i,6]),
    Well_Plate = interaction(Well, Plate, sep = "_")) #Add Replicate
results5 <- bind_rows(results5, tmp)
}
```

```{r, include=FALSE}
results6 <- data_frame()

# Provide a lookup table for plate information, treatments are spread accross sheets in the table to be read.
# in the first column describe the plates and in the second column the experiment (rep(times = # plates - 2))
treatlist <- data_frame("Treat" =
              c("Dummy",
                "Ctrl_DMSO_TSB_A",
                "Ctrl_DMSO_TSB_B",
                "H_DMSO_TSB_A",
                "H_DMSO_TSB_B",
                "media",
                "10_APO_TSB_A",
                "10_APO_TSB_B",
                "25_APO_TSB_A",
                "25_APO_TSB_B",
                "50_APO_TSB_A",
                "50_APO_TSB_B",
                "Dummy"),
  "Run" = c("NULL", rep("R5", times = 11), "NULL"), # times = all plates but without the dummy plates
  "Plate" = c("NULL", rep(c("PlateA", "PlateB"), times = 2), "NULL",  rep(c("PlateA", "PlateB"), times = 3), "NULL"),
  "Media" = c("NULL", rep ("TSB", times = 11), "NULL"),
  "Chem" = c("NULL", "DMSO", "DMSO", rep("DMSO", times = 2), "NULL", rep("APO", times = 6), "NULL"),
  "Conc" = c("NULL", c("0", "0"), c("H", "H"), "NULL", c("10", "10", "25", "25", "50", "50"), "NULL"))


#Initialize / clear df

#Loop through sheets.

for (i in 1:nrow(treatlist)) {
  tmp <- read_excel("Input/220228_SCOP_APO_MRB_run1.xlsx", sheet = i, range = "B26:CU68") %>%   # change for duration runs
    dplyr::select(-2) %>% # Drop Temperature
    pivot_longer(names_to = "Well", values_to = "Density", cols = 3:97) %>%  #Make long
    mutate(Treat = as.character(treatlist[i,1]), #Add treatment
    Run = as.character(treatlist[i,2]),
    Plate = as.character(treatlist[i,3]),
    Media = as.character(treatlist[i,4]),
    Chem = as.character(treatlist[i,5]),
    Conc = as.character(treatlist[i,6]),
    Well_Plate = interaction(Well, Plate, sep = "_")) #Add Replicate
results6 <- bind_rows(results6, tmp)
}
```

```{r, include=FALSE}
results7 <- data_frame()

# Provide a lookup table for plate information, treatments are spread accross sheets in the table to be read.
# in the first column describe the plates and in the second column the experiment (rep(times = # plates - 2))
treatlist <- data_frame("Treat" =
              c("Dummy",
                "Ctrl_DMSO_TSB_A",
                "Ctrl_DMSO_TSB_B",
                "media",
                "10_APO_TSB_A",
                "10_APO_TSB_B",
                "25_APO_TSB_A",
                "25_APO_TSB_B",
                "50_APO_TSB_A",
                "50_APO_TSB_B",
                "Dummy"),
  "Run" = c("NULL", rep("R6", times = 9), "NULL"), # times = all plates but without the dummy plates
  "Plate" = c("NULL", "PlateA", "PlateB", "NULL", rep(c("PlateA", "PlateB"), times = 3), "NULL"),
  "Media" = c("NULL", rep ("TSB", times = 9), "NULL"),
  "Chem" = c("NULL", "DMSO", "DMSO", "NULL", rep("APO", times = 6), "NULL"),
  "Conc" = c("NULL", "0", "0", "NULL", c("10", "10", "25", "25", "50", "50"), "NULL"))


#Initialize / clear df

#Loop through sheets.

for (i in 1:nrow(treatlist)) {
  tmp <- read_excel("Input/220328_SCO_APO_MRB_run2.xlsx", sheet = i, range = "B26:CU76") %>%   # change for duration runs
    dplyr::select(-2) %>% # Drop Temperature
    pivot_longer(names_to = "Well", values_to = "Density", cols = 3:97) %>%  #Make long
    mutate(Treat = as.character(treatlist[i,1]), #Add treatment
    Run = as.character(treatlist[i,2]),
    Plate = as.character(treatlist[i,3]),
    Media = as.character(treatlist[i,4]),
    Chem = as.character(treatlist[i,5]),
    Conc = as.character(treatlist[i,6]),
    Well_Plate = interaction(Well, Plate, sep = "_")) #Add Replicate
results7 <- bind_rows(results7, tmp)
}
```

**Formatting**:Data from the different runs are merged, then the time intervals between the measurements are calculated and last then the results are joined with the strain metadata of the strains. Next the density increase is calculated by subtracting the first measurement (represents inoculation density) from all subsequent measurements. This result data file is saved as raw results. 

```{r, include=FALSE}
results <- rbind(results1, results2, results3, results4, results5, results6, results7)
results %<>% mutate(Conc = as.character(Conc))
results %<>% filter(Media %in% "TSB")
```

```{r, include=FALSE}
###
###Get times based on difftime. 
results %<>%
  filter(!Treat %in% "Dummy") %>%
  filter(!Density %in% "NA") %>% 
  mutate(mins = difftime(Time, Time[1], units = "mins") %>% as.numeric(),
         hrs = difftime(Time, Time[1], units = "hours") %>% as.numeric()) %>% 
  left_join(., Strain_Data, by = "Well_Plate")

results %<>% dplyr::select(-Well.x, -Plate.x, -Time) %>% dplyr::rename(Well = Well.y, Plate = Plate.y) %>%  as.data.frame() 
```

```{r, include=FALSE}
results %<>%
  filter(!Strain %in% c("Ignore","None")) %>%
  group_by(Run, Well_Plate, Treat) %>%
  mutate(Density_increase = Density - Density[1]) %>%
  ungroup %>% as.data.frame()
```

```{r, include=FALSE}
results_raw <- results
```

## Growth curves
First the density increase in the medium control plate is plotted to check that there was no general contamination detected in the experiment.

```{r, message=FALSE, echo=FALSE, error=FALSE, warning = FALSE, fig.dim = c(6, 5)}
results %>%  filter(Treat %in% c("media")) %>%
  separate(Well_Plate, c("Well", "Plate")) %>%
  mutate(Row = Well %>% str_extract(.,"[A-H]"),
                    Column = Well %>% str_extract(.,"[0-9]+") %>% as.numeric()) %>% 
  ggplot(aes(x=hrs, y= Density_increase)) +
  geom_line(aes(colour = Run), linewidth = 1) +
  facet_grid(Row ~ Column, scales = "fixed") +
  theme_bw() +
  labs(x = "Time [hours]",
       y = "OD 600", 
       title = "growth curves in media control plate",
       color = "Run")
```

Next the growth of all strains in the control treatment and the no bacteria control is checked. 

```{r, message=FALSE, echo=FALSE, error=FALSE, warning = FALSE, fig.dim = c(8, 8)}
results %>%  
  filter(Chem %in% "DMSO") %>% 
  unique() %>% 
  ggplot(aes(x=hrs, y= Density_increase, color = interaction(Chem, Conc))) +
  geom_line(aes(colour = Run, linetype = as.factor(Column))) +
  facet_wrap(~ Strain, scales = "fixed") +
  theme_bw() +
  labs(x = "Time [hours]",
       y = "OD 600", 
       title = "growth curves",
       color = "Run")
```

Here the growth of two bacterial strains in one run in 6 concnetrations of MBOA and the control treatment are plotted. Pseudomonas LPD2 is MBOA-tolerant whil LRH8.O is MBOA susceptible.

```{r, , message=FALSE, echo=FALSE, error=FALSE, warning = FALSE, fig.dim = c(4, 4)}
fIG4.curve_LPD2_LRH8.O <- results %>%  
  filter(Run %in% "R4") %>% 
  filter(Conc %in% c("5000", "2500", "1250", "625", "500", "250", "0")) %>% 
  mutate(Conc = factor(Conc, levels = c("5000", "2500", "1250", "625", "500", "250", "0"))) %>% 
  mutate(Replicate = as.factor(Replicate)) %>% 
  mutate(Run = as.factor(Run)) %>% 
  filter(Chem %in% c("DMSO", "MBOA")) %>% 
  filter(Strain %in% c("LRH8.O", "LPD2")) %>% 
  unique() %>%
  ggplot(aes(x=hrs, y= Density_increase, color = Conc)) +
  geom_line(stat = "summary", linewidth = 0.5) +
  facet_wrap(~ Strain, nrow = 2, scales = "fixed") +
  theme_bw() +
  theme(legend.position = c(0.2, 0.3)) +
  theme(legend.background = element_rect(fill="transparent", 
                                  linewidth=0.05, linetype="solid"), 
        legend.text = element_text(size = 5),
        legend.key.size = unit(0.055, 'cm'), #change legend key size
        legend.key.height = unit(0.055, 'cm')) +
  scale_color_manual(values = c("#9ecae1", "#6baed6", "#4292c6", "#2171b5", "#08519c", "#08306b", "#000740"), name=" ",
                       breaks=c("5000", "2500", "1250", "625", "500", "250", "0"),
                       labels=c("5000 μM", "2500 μM", "1250 μM", "625 μM", "500 μM", "250 μM", "0 μM"))+
  theme(strip.background = element_rect(color = NULL, fill= "white", linetype="solid"), strip.text.x = element_text(size = 16/.pt, margin = margin(1, 1, 1, 1)))+
  theme(axis.text.x=element_text(size = 16/.pt),
        axis.text.y=element_text(size = 16/.pt),
        axis.title = element_text(size = 16/.pt)) +
  labs(x = "Time [hours]",
       y = "OD 600",
       linetype = "Replicate")

fIG4.curve_LPD2_LRH8.O

# ggsave(plot = fIG4.curve_LPD2_LRH8.O,  filename = "fIG4.curve_LPD2_LRH8.O.pdf", width = 4.4, height = 6, dpi = 300, scale = 1, units = "cm")

# ggsave(plot = fIG4.curve_LPD2_LRH8.O,  filename = "fIG4.curve_LPD2_LRH8.O.svg", width = 4.4, height = 6, dpi = 300, scale = 1, units = "cm")
```

**Remove strains from the analysis**: For some strains bad growth was detected in the growth curve which are removed due to bad growth, possible contamination, unclear taxonomy and antibiotic tolerant strains.

```{r, include=FALSE}
# remove strains
results %<>% filter(!Strain %in% c("LAR4", "LPE1.3.1", "LPA11", "LMX8", "LPB4.R", "LRH8.S", "LMR1", "LMU1", "LMG1", "LMF1", "LMI14", "LST17", "LAR4"))

# remove minimal media
results %<>% filter(!Media %in% c("MM")) 
```

```{r, include=FALSE}
dir.create("Output", showWarnings = F)
#write_rds(results, paste0("Output/results.rds"))
```

## Calculate area under the curve (AUC)
To quantify the total bacterial growth over time, we calculate the total area under the curve. This is done with the function "auc()" For comparison between treatments, the total AUC (AUC_raw) is normalized with the AUC of the strain grown in the control treatment (no chemicals added, just normal growth media with DMSO). The AUC_norm is used for all further analysis, plotting and calculations. 

```{r, include=FALSE, warning=FALSE}
results %<>% dplyr::select(-Well_Plate, -Plate) %>% unique()

results_AUC_MBOA <- results %>% 
  filter(Media %in% "TSB") %>% 
  filter(Chem %in% c("DMSO", "MBOA")) %>% 
  mutate(Conc = as.numeric(Conc)) %>% 
  # mutate(Conc = factor(Conc, levels = c("250", "500", "625", "1250", "2500", "5000", "0"))) %>% 
  group_by(Strain, Replicate, Media, Treat, Conc, Well, Run) %>%
  dplyr::summarize(AUC = auc(hrs, Density_increase), AUC_raw = auc(hrs, Density)) %>%
  ungroup %>% 
  mutate(Treat = as.factor(Treat), Replicate = as.factor(Replicate)) %>% 
  mutate(AUC_pos = abs(AUC)) %>% 
  mutate(Strain_Replicate_Media_Run = paste(Strain, Replicate, Media, Run)) %>% 
  arrange(desc(Conc))  %>% # this a super important step to normalize with the growth in Conc = 0 (Conc has to be a numeric, see above)
  #Split by Rep
  base::split(., .$Strain_Replicate_Media_Run) %>% #Make each strain / rep combination its own data.frane
  map_dfr(~ 
  mutate(.x, AUC_norm = AUC_pos / AUC_pos[length(AUC_pos)])) %>% #in each df, divide the AUC by the last AUC value; since Control does not start with a number, it is the last treatment
  left_join(. , Strain_Data %>% dplyr::select(Strain, Phylum, Class, Order, Family, Genus, ZOTU), by = "Strain") %>% 
  # mutate(Well_Row = Well %>% str_extract(.,"[A-H]"),
  #                   Well_Col = Well %>% str_extract(.,"[0-9]+") %>% as.numeric()) %>% 
  as.data.frame() #transform into df
## Save table with AUC
#write_rds(results_AUC_MBOA,paste0("Output/results_AUC_MBOA.rds"))
```

```{r, , include=FALSE, warning=FALSE}
# results %<>% dplyr::select(-Well_Plate, -Plate) %>% unique()

results_AUC_AMPO <- results %>% 
  filter(Media %in% "TSB") %>% 
  filter(Chem %in% c("DMSO", "AMPO")) %>% 
  mutate(Conc = as.numeric(Conc)) %>% 
  # mutate(Conc = factor(Conc, levels = c("250", "500", "625", "1250", "2500", "5000", "0"))) %>% 
  group_by(Strain, Replicate, Media, Treat, Conc, Well, Run) %>%
  dplyr::summarize(AUC = auc(hrs, Density_increase), 
            AUC_raw = auc(hrs, Density)) %>%
  ungroup %>% 
  mutate(Treat = as.factor(Treat),
         Replicate = as.factor(Replicate)) %>% 
  mutate(AUC_pos = abs(AUC)) %>% 
  mutate(Strain_Replicate_Media_Run = paste(Strain, Replicate, Media, Run)) %>% 
  arrange(desc(Conc))  %>% # this a super important step to normalize with the growth in Conc = 0 (Conc has to be a numeric, see above)
  #Split by Rep
  base::split(., .$Strain_Replicate_Media_Run) %>% #Make each strain / rep combination its own data.frane
  map_dfr(~ 
  mutate(.x, AUC_norm = AUC_pos / AUC_pos[length(AUC_pos)])) %>% #in each df, divide the AUC by the last AUC value; since Control does not start with a number, it is the last treatment
  left_join(. , Strain_Data %>% dplyr::select(Strain, Phylum, Class, Order, Family, Genus, ZOTU), by = "Strain") %>% 
  # mutate(Well_Row = Well %>% str_extract(.,"[A-H]"),
  #                   Well_Col = Well %>% str_extract(.,"[0-9]+") %>% as.numeric()) %>% 
  as.data.frame() #transform into df
## Save table with AUC
#write_rds(results_AUC_AMPO,paste0("Output/results_AUC_AMPO.rds"))
```

```{r, include=FALSE, warning=FALSE}
results_AUC_BOA <- results %>% 
  filter(Media %in% "TSB") %>% 
  filter(Chem %in% c("DMSO", "BOA")) %>% 
  filter(Conc != "H") %>% 
  mutate(Conc = as.numeric(Conc)) %>% 
  # mutate(Conc = factor(Conc, levels = c("250", "500", "625", "1250", "2500", "5000", "0"))) %>% 
  group_by(Strain, Replicate, Media, Treat, Conc, Well, Run) %>%
  dplyr::summarize(AUC = auc(hrs, Density_increase), 
            AUC_raw = auc(hrs, Density)) %>%
  ungroup %>% 
  mutate(Treat = as.factor(Treat),
         Replicate = as.factor(Replicate)) %>% 
  mutate(AUC_pos = abs(AUC)) %>% 
  mutate(Strain_Replicate_Media_Run = paste(Strain, Replicate, Media, Run)) %>% 
  arrange(desc(Conc))  %>% # this a super important step to normalize with the growth in Conc = 0 (Conc has to be a numeric, see above)
  #Split by Rep
  base::split(., .$Strain_Replicate_Media_Run) %>% #Make each strain / rep combination its own data.frane
  map_dfr(~ 
  mutate(.x, AUC_norm = AUC_pos / AUC_pos[length(AUC_pos)])) %>% #in each df, divide the AUC by the last AUC value; since Control does not start with a number, it is the last treatment
  left_join(. , Strain_Data %>% dplyr::select(Strain, Phylum, Class, Order, Family, Genus, ZOTU), by = "Strain") %>% 
  # mutate(Well_Row = Well %>% str_extract(.,"[A-H]"),
  #                   Well_Col = Well %>% str_extract(.,"[0-9]+") %>% as.numeric()) %>% 
  as.data.frame() #transform into df
## Save table with AUC
#write_rds(results_AUC_BOA, paste0("Output/results_AUC_BOA.rds"))
```

```{r, include=FALSE, warning=FALSE}
# results %<>% dplyr::select(-Well_Plate, -Plate) %>% unique()

results_AUC_DMG <- results %>% 
  filter(Media %in% "TSB") %>% 
  filter(Run %in% "R2") %>% 
  filter(Chem %in% c("DMSO", "DMG")) %>% 
  filter(Conc != "H") %>% 
  mutate(Conc = as.numeric(Conc)) %>% 
  # mutate(Conc = factor(Conc, levels = c("250", "500", "625", "1250", "2500", "5000", "0"))) %>% 
  group_by(Strain, Replicate, Media, Treat, Conc, Well, Run) %>%
  dplyr::summarize(AUC = auc(hrs, Density_increase), 
            AUC_raw = auc(hrs, Density)) %>%
  ungroup %>% 
  mutate(Treat = as.factor(Treat),
         Replicate = as.factor(Replicate)) %>% 
  mutate(AUC_pos = abs(AUC)) %>% 
  mutate(Strain_Replicate_Media_Run = paste(Strain, Replicate, Media, Run)) %>% 
  arrange(desc(Conc))  %>% # this a super important step to normalize with the growth in Conc = 0 (Conc has to be a numeric, see above)
  #Split by Rep
  base::split(., .$Strain_Replicate_Media_Run) %>% #Make each strain / rep combination its own data.frane
  map_dfr(~ 
  mutate(.x, AUC_norm = AUC_pos / AUC_pos[length(AUC_pos)])) %>% #in each df, divide the AUC by the last AUC value; since Control does not start with a number, it is the last treatment
  left_join(. , Strain_Data %>% dplyr::select(Strain, Phylum, Class, Order, Family, Genus, ZOTU), by = "Strain") %>% 
  # mutate(Well_Row = Well %>% str_extract(.,"[A-H]"),
  #                   Well_Col = Well %>% str_extract(.,"[0-9]+") %>% as.numeric()) %>% 
  as.data.frame() #transform into df
## Save table with AUC
#write_rds(results_AUC_AMPO,paste0("Output/results_AUC_DMG.rds"))
```

```{r, include=FALSE, warning=FALSE}
results_AUC_APO <- results %>% 
  filter(Media %in% "TSB") %>% 
  filter(Chem %in% c("DMSO", "APO")) %>% 
  filter(Conc != "H") %>% 
  mutate(Conc = as.numeric(Conc)) %>% 
  # mutate(Conc = factor(Conc, levels = c("250", "500", "625", "1250", "2500", "5000", "0"))) %>% 
  group_by(Strain, Replicate, Media, Treat, Conc, Well, Run) %>%
  dplyr::summarize(AUC = auc(hrs, Density_increase), 
            AUC_raw = auc(hrs, Density)) %>%
  ungroup %>% 
  mutate(Treat = as.factor(Treat),
         Replicate = as.factor(Replicate)) %>% 
  mutate(AUC_pos = abs(AUC)) %>% 
  mutate(Strain_Replicate_Media_Run = paste(Strain, Replicate, Media, Run)) %>% 
  arrange(desc(Conc))  %>% # this a super important step to normalize with the growth in Conc = 0 (Conc has to be a numeric, see above)
  #Split by Rep
  base::split(., .$Strain_Replicate_Media_Run) %>% #Make each strain / rep combination its own data.frane
  map_dfr(~ 
  mutate(.x, AUC_norm = AUC_pos / AUC_pos[length(AUC_pos)])) %>% #in each df, divide the AUC by the last AUC value; since Control does not start with a number, it is the last treatment
  left_join(. , Strain_Data %>% dplyr::select(Strain, Phylum, Class, Order, Family, Genus, ZOTU), by = "Strain") %>% 
  # mutate(Well_Row = Well %>% str_extract(.,"[A-H]"),
  #                   Well_Col = Well %>% str_extract(.,"[0-9]+") %>% as.numeric()) %>% 
  as.data.frame() #transform into df
## Save table with AUC
#write_rds(results_AUC_APO, paste0("Output/results_AUC_APO.rds"))
```

```{r, include=FALSE}
results_AUC_MBOA$Chem <- "MBOA"
results_AUC_BOA$Chem <- "BOA"
results_AUC_AMPO$Chem <- "AMPO"
results_AUC_DMG$Chem <- "DMG"
results_AUC_APO$Chem <- "APO"
results_AUC_all_raw <- rbind(results_AUC_MBOA, results_AUC_BOA, results_AUC_AMPO, results_AUC_DMG, results_AUC_APO)
results_AUC_all_raw %<>% mutate(Chem_Conc = interaction(Chem, Conc, sep = "_"))

results_AUC_all <- results_AUC_all_raw
```

```{r, include=FALSE}
results_AUC_all %<>% na.omit()

results_AUC_all$cols_Family <- as.character(results_AUC_all$Family)

results_AUC_all[ results_AUC_all$Family=="Moraxellaceae", ]$cols_Family <- "#a73e62"
results_AUC_all[ results_AUC_all$Family=="Bacillaceae" , ]$cols_Family <- "#004586"
results_AUC_all[ results_AUC_all$Family=="Pseudomonadaceae" , ]$cols_Family <- "#4b1f6f"
results_AUC_all[ results_AUC_all$Family=="Paenibacillaceae" , ]$cols_Family <- "#8f9ec9"
results_AUC_all[ results_AUC_all$Family=="Rhizobiaceae", ]$cols_Family <- "#ff950e"
results_AUC_all[ results_AUC_all$Family=="Sphingomonadaceae", ]$cols_Family <- "#c5000b"
results_AUC_all[ results_AUC_all$Family=="Xanthomonadaceae", ]$cols_Family <- "#0084d1"
results_AUC_all[ results_AUC_all$Family=="Microbacteriaceae" , ]$cols_Family <- "#7e0021"
results_AUC_all[ results_AUC_all$Family=="Micrococcaceae" , ]$cols_Family <- "#83caff"
results_AUC_all[ results_AUC_all$Family=="Chitinophagaceae" , ]$cols_Family <- "#ff420e"
results_AUC_all[ results_AUC_all$Family=="Enterobacteriaceae" , ]$cols_Family <- "#ffd320"
results_AUC_all[ results_AUC_all$Family=="Erwiniaceae" , ]$cols_Family <- "#579d1c"
results_AUC_all[ results_AUC_all$Family=="Nocardioidaceae" , ]$cols_Family <- "#314004"
results_AUC_all[ results_AUC_all$Family=="Oxalobacteraceae" , ]$cols_Family <- "#aecf00"

temp <- data.frame(results_AUC_all$Family, results_AUC_all$cols_Family)
temp %<>% mutate(results_AUC_all.cols_Family_light = adjustcolor(results_AUC_all.cols_Family, alpha.f = 0.5))
temp <- plyr::ddply(temp, .variables="results_AUC_all.cols_Family", .fun=unique)   #library(plyr)
results_AUC_all_level_cols_Family <- as.character(temp[,2])
names(results_AUC_all_level_cols_Family) <- temp[,1]

results_AUC_all_level_cols_Family_light <- as.character(temp[,3])
names(results_AUC_all_level_cols_Family_light) <- temp[,1]
```

```{r, include=FALSE}
sorted <- results_AUC_all %>% dplyr::arrange(Family) %>% dplyr::select(Family) %>% unique()
sorted_Family <- sorted$Family
```

```{r, include=FALSE}
#write_rds(results_AUC_all, "Output/results_AUC_all.rds")
```

## Plotting AUC 
*Figures for supplement*

All AUC MBOA

```{r, message=FALSE, echo=FALSE, error=FALSE, warning = FALSE, fig.dim = c(10, 10)}
AUC_all_MBOA <- results_AUC_all %>%  
  filter(Run %in% c("R3", "R4")) %>% 
  filter(Chem %in% "MBOA") %>% 
  unique() %>% 
  ggplot(aes(x= as.factor(Conc), y= AUC_norm)) +
  #geom_boxplot(aes(colour = Family), position = position_dodge(0.5), show.legend = FALSE) +
  #geom_point(aes(colour = Family, shape = Run), size = 0.1, position = position_dodge(0.5), show.legend = FALSE)+
  stat_summary(fun.y = mean, fun.ymax = max, fun.ymin = min, alpha = 0.4) +
  geom_point(aes(colour = Family), alpha=0.4, bins = 3, show.legend = FALSE) + 
  stat_compare_means(label = "p.signif", method = "t.test", ref.group = "0", label.y = 1.4, symnum.args = symnum.args) +
  ylim(c(0, 1.8))+
  facet_wrap(~ reorder(Strain, desc(Family)), nrow = 4, scales = "fixed") +
  theme_bw() +
  theme(axis.text.x = element_text(angle = -90, size = 8, hjust = 0, vjust = 0.5)) +
  scale_color_manual(values = results_AUC_all_level_cols_Family) +
  labs(y = "AUC",
       x = "concentration [μM]", 
       title = "growth in MBOA",
       color = "Family")

AUC_all_MBOA

#ggsave(plot = AUC_all_MBOA,  filename = "AUC_all_MBOA.pdf", width = 25, height = 15, dpi = 300, scale = 1, units = "cm")
# 
#ggsave(plot = AUC_all_MBOA,  filename = "AUC_all_MBOA.svg", width = 25, height = 15, dpi = 300, scale = 1, units = "cm")
```

AUC tolerant and susceptible strain in MBOA

```{r, message=FALSE, echo=FALSE, error=FALSE, warning = FALSE, fig.dim = c(4, 4)}
fIG4.AUC_LPD2_LRH8.O <- results_AUC_all %>%  
  mutate(Strain = reorder(Strain, Family)) %>% 
  filter(Run %in% c("R3", "R4")) %>% 
  # filter(!Run %in% "R1" | !Strain %in% c("LMX92", "LMX9231", "LST11", "LMI14", "LPB4.O")) %>% 
  filter(Chem %in% "MBOA") %>% 
  filter(Strain %in% c("LRH8.O", "LPD2")) %>% 
  unique() %>% 
  ggplot(aes(x= as.factor(Conc), y= AUC_norm)) +
  geom_bar(aes(fill = Family), alpha = 0.5, stat = "summary", position = "dodge", show.legend = FALSE) +
  geom_quasirandom(aes(fill = Family), size = 0.1/.pt, alpha = 0.7, shape = 21,  width = 0.3, show.legend = FALSE)+
  geom_errorbar(stat = "summary", width=0.2) +
  # geom_point(aes(colour = Family, shape = Run), show.legend = FALSE)+
  # stat_summary(geom = "pointrange", fun.y = mean, fun.ymax = max, fun.ymin = min, alpha = 0.7, show.legend = FALSE) +
  stat_compare_means(label = "p.signif", method = "t.test", ref.group = "0", size = 3, label.y = 1, hide.ns = TRUE) +
  ylim(c(0, 1.5))+
  facet_wrap(~ Strain, nrow = 2, scales = "fixed") +
  theme_bw() +
  # theme(axis.text.x = element_text(angle = 0, hjust = 0, vjust = 0.5)) +
  theme(strip.background = element_rect(color = NULL, fill= "white", linetype="solid"), strip.text.x = element_text(size = 16/.pt, margin = margin(1, 1, 1, 1)))+
  scale_fill_manual(values = results_AUC_all_level_cols_Family) +
  scale_color_manual(values = results_AUC_all_level_cols_Family) +
  scale_x_discrete(labels= c("0", "250", "500", "625", "1250", "2500", "5000"))+
  theme(axis.text.x=element_text(size = 16/.pt),
        axis.text.y=element_text(size = 16/.pt),
        axis.title = element_text(size = 16/.pt)) +
  labs(y = "AUC norm",
       x = "concentration [μM]", 
       title = " ",
#       caption = "test",
       color = "Family")

fIG4.AUC_LPD2_LRH8.O

# ggsave(plot = fIG4.AUC_LPD2_LRH8.O,  filename = "fIG4.AUC_LPD2_LRH8.O.pdf", width = 4.4, height = 6, dpi = 300, scale = 1, units = "cm")

# ggsave(plot = fIG4.AUC_LPD2_LRH8.O,  filename = "fIG4.AUC_LPD2_LRH8.O.svg", width = 4.4, height = 6, dpi = 300, scale = 1, units = "cm")
```

All AUC BOA

```{r, message=FALSE, echo=FALSE, error=FALSE, warning = FALSE, fig.dim = c(10, 10)}
AUC_all_BOA <- results_AUC_all %>%  
  mutate(Strain = reorder(Strain, Family)) %>% 
  filter(Run %in% c("R1", "R2")) %>% 
  filter(!Run %in% "R1" | !Strain %in% c("LMX92", "LMX9231", "LST11", "LMI14", "LPB4.O")) %>% 
  filter(Chem %in% "BOA") %>% 
  unique() %>% 
  ggplot(aes(x= as.factor(Conc), y= AUC_norm)) +
  #geom_boxplot(aes(colour = Family), position = position_dodge(0.5), show.legend = FALSE) +
  #geom_point(aes(colour = Family, shape = Run), size = 0.1, position = position_dodge(0.5), show.legend = FALSE)+
  stat_summary(fun.y = mean, fun.ymax = max, fun.ymin = min, alpha = 0.4) +
  geom_point(aes(colour = Family), alpha=0.4, bins = 3, show.legend = FALSE) + 
  stat_compare_means(label = "p.signif", method = "t.test", ref.group = "0", label.y = 1.4, symnum.args = symnum.args) +
  ylim(c(0, 1.8))+
  facet_wrap(~ reorder(Strain, desc(Family)), nrow = 4, scales = "fixed") +
  theme_bw() +
  theme(axis.text.x = element_text(angle = -90, size = 8, hjust = 0, vjust = 0.5)) +
  scale_color_manual(values = results_AUC_all_level_cols_Family) +
  labs(y = "AUC",
       x = "concentration [μM]", 
       title = "growth in BOA",
       color = "Family")

AUC_all_BOA

#ggsave(plot = AUC_all_BOA,  filename = "AUC_all_BOA.pdf", width = 25, height = 15, dpi = 300, scale = 1, units = "cm")
# 
#ggsave(plot = AUC_all_BOA,  filename = "AUC_all_BOA.svg", width = 25, height = 15, dpi = 300, scale = 1, units = "cm")
```

All AUC AMPO

```{r, message=FALSE, echo=FALSE, error=FALSE, warning = FALSE, fig.dim = c(10, 10)}
AUC_all_AMPO <- results_AUC_all %>%  
  mutate(Strain = reorder(Strain, Family)) %>% 
  filter(Run %in% c("R1", "R4")) %>% 
  filter(Chem %in% "AMPO") %>% 
  unique() %>% 
  ggplot(aes(x= as.factor(Conc), y= AUC_norm)) +
  #geom_boxplot(aes(colour = Family), position = position_dodge(0.5), show.legend = FALSE) +
  #geom_point(aes(colour = Family, shape = Run), size = 0.1, position = position_dodge(0.5), show.legend = FALSE)+
  stat_summary(fun.y = mean, fun.ymax = max, fun.ymin = min, alpha = 0.4) +
  geom_point(aes(colour = Family), alpha=0.4, bins = 3, show.legend = FALSE) + 
  stat_compare_means(label = "p.signif", method = "t.test", ref.group = "0", label.y = 1.4, symnum.args = symnum.args) +
  ylim(c(0, 1.8))+
  facet_wrap(~ reorder(Strain, desc(Family)), nrow = 4, scales = "fixed") +
  theme_bw() +
  theme(axis.text.x = element_text(angle = -90, size = 8, hjust = 0, vjust = 0.5)) +
  scale_color_manual(values = results_AUC_all_level_cols_Family) +
  labs(y = "AUC",
       x = "concentration [μM]", 
       title = "growth in AMPO",
       color = "Family")

AUC_all_AMPO

#ggsave(plot = AUC_all_AMPO,  filename = "AUC_all_AMPO.pdf", width = 25, height = 15, dpi = 300, scale = 1, units = "cm")

#ggsave(plot = AUC_all_AMPO,  filename = "AUC_all_AMPO.svg", width = 25, height = 15, dpi = 300, scale = 1, units = "cm")
```

All AUC APO

```{r, message=FALSE, echo=FALSE, error=FALSE, warning = FALSE, fig.dim = c(10, 10)}
AUC_all_APO <- results_AUC_all %>%  
  mutate(Strain = reorder(Strain, Family)) %>% 
  filter(Run %in% c("R6", "R5")) %>% 
  filter(Chem %in% "APO") %>% 
  unique() %>% 
  ggplot(aes(x= as.factor(Conc), y= AUC_norm)) +
  #geom_boxplot(aes(colour = Family), position = position_dodge(0.5), show.legend = FALSE) +
  #geom_point(aes(colour = Family, shape = Run), size = 0.1, position = position_dodge(0.5), show.legend = FALSE)+
  stat_summary(fun.y = mean, fun.ymax = max, fun.ymin = min, alpha = 0.4) +
  geom_point(aes(colour = Family), alpha=0.4, bins = 3, show.legend = FALSE) + 
  stat_compare_means(label = "p.signif", method = "t.test", ref.group = "0", label.y = 1.4, symnum.args = symnum.args) +
  ylim(c(0, 1.8))+
  facet_wrap(~ reorder(Strain, desc(Family)), nrow = 4, scales = "fixed") +
  theme_bw() +
  theme(axis.text.x = element_text(angle = -90, size = 8, hjust = 0, vjust = 0.5)) +
  scale_color_manual(values = results_AUC_all_level_cols_Family) +
  labs(y = "AUC",
       x = "concentration [μM]", 
       title = "growth in APO",
       color = "Family")

AUC_all_APO

#ggsave(plot = AUC_all_APO,  filename = "AUC_all_APO.pdf", width = 25, height = 15, dpi = 300, scale = 1, units = "cm")

#ggsave(plot = AUC_all_APO,  filename = "AUC_all_APO.svg", width = 25, height = 15, dpi = 300, scale = 1, units = "cm")
```

All AUC DIMBOA-Glc

```{r, message=FALSE, echo=FALSE, error=FALSE, warning = FALSE, fig.dim = c(10, 10)}
AUC_all_DMG <- results_AUC_all %>%  
  mutate(Strain = reorder(Strain, Family)) %>% 
  filter(Run %in% c("R2")) %>% 
  filter(Chem %in% "DMG") %>% 
  unique() %>% 
  ggplot(aes(x= as.factor(Conc), y= AUC_norm)) +
  #geom_boxplot(aes(colour = Family), position = position_dodge(0.5), show.legend = FALSE) +
  #geom_point(aes(colour = Family, shape = Run), size = 0.1, position = position_dodge(0.5), show.legend = FALSE)+
  stat_summary(fun.y = mean, fun.ymax = max, fun.ymin = min, alpha = 0.4) +
  geom_point(aes(colour = Family), alpha=0.4, bins = 3, show.legend = FALSE) + 
  stat_compare_means(label = "p.signif", method = "t.test", ref.group = "0", label.y = 1.4, symnum.args = symnum.args) +
  ylim(c(0, 1.8))+
  facet_wrap(~ reorder(Strain, desc(Family)), nrow = 4, scales = "fixed") +
  theme_bw() +
  theme(axis.text.x = element_text(angle = -90, size = 8, hjust = 0, vjust = 0.5)) +
  scale_color_manual(values = results_AUC_all_level_cols_Family) +
  labs(y = "AUC",
       x = "concentration [μM]", 
       title = "growth in DIMBOA-Glc",
       color = "Family")

AUC_all_DMG

# ggsave(plot = AUC_all_DMG,  filename = "AUC_all_DMG.pdf", width = 25, height = 15, dpi = 300, scale = 1, units = "cm")
# 
# ggsave(plot = AUC_all_DMG,  filename = "AUC_all_DMG.svg", width = 25, height = 15, dpi = 300, scale = 1, units = "cm")
```

## Compare concentrations and chemicals

**MBOA and AMPO at 50 μM**: Compare low MBOA concentrations with high AMPO concentrations [50 uM]

```{r, message=FALSE, echo=FALSE, error=FALSE, warning = FALSE, fig.dim = c(6, 5)}
fIGsupp.MBOA50 <- results_AUC_all %>%
  filter(AUC_norm < 2) %>% 
  filter(Strain != "NBC") %>% 
  filter(Chem_Conc %in% c("MBOA_50")) %>% 
  unique() %>% 
  ggplot(aes(y = AUC_norm, x = reorder(Strain, desc(AUC_norm)))) + 
  geom_bar(aes(fill = Family),  size = 0.1/.pt, alpha = 0.5, stat = "summary", position = "dodge", show.legend = FALSE) +
  ggbeeswarm::geom_quasirandom(aes(color = Family), size = 0.01/.pt, alpha = 0.7,  width = 0.3, show.legend = FALSE)+
  geom_errorbar(stat = "summary", size = 0.5/.pt, width=0.2) +
  geom_hline(colour = "grey50", yintercept = 1, linewidth = 1/.pt) +
  geom_hline(colour = "grey50", yintercept = 0.75, linewidth = 1/.pt) +
  geom_hline(colour = "grey50", yintercept = 0.5, linewidth = 1/.pt) +
  scale_fill_manual(values = results_AUC_all_level_cols_Family) +
  scale_colour_manual(values = results_AUC_all_level_cols_Family) +
  theme_classic()+
  scale_y_continuous(limits = c(0, 1.7), expand = expansion(mult = c(0, 0)))+
  theme(axis.text.x=element_text(angle = -90, hjust = 0, vjust = 0.5, size = 15/.pt),
        axis.text.y=element_text(size = 15/.pt),
        axis.title = element_text(size = 15/.pt)) +
  labs(x = "", y = "AUC [MBOA 50 μM]")


fIGsupp.MBOA50

# ggsave(plot = fIGsupp.MBOA50,  filename = "fIGsupp.MBOA50.pdf", width = 8.8, height = 6, dpi = 300, scale = 1, units = "cm")
# 
# ggsave(plot = fIGsupp.MBOA50,  filename = "fIGsupp.MBOA50.svg", width = 8.8, height = 6, dpi = 300, scale = 1, units = "cm")

```

```{r, message=FALSE, echo=FALSE, error=FALSE, warning = FALSE, fig.dim = c(6, 5)}
fIGsupp.AMPO50 <- results_AUC_all %>%
   filter(AUC_norm < 2) %>% 
  filter(Strain != "NBC") %>% 
  filter(Chem_Conc %in% c("AMPO_50")) %>% 
  unique() %>% 
  ggplot(aes(y = AUC_norm, x = reorder(Strain, desc(AUC_norm)))) + 
  geom_bar(aes(fill = Family),  size = 0.1/.pt, alpha = 0.5, stat = "summary", position = "dodge", show.legend = FALSE) +
  ggbeeswarm::geom_quasirandom(aes(color = Family), size = 0.01/.pt, alpha = 0.7,  width = 0.3, show.legend = FALSE)+
  geom_errorbar(stat = "summary", size = 0.5/.pt, width=0.2) +
  geom_hline(colour = "grey50", yintercept = 1, linewidth = 1/.pt) +
  geom_hline(colour = "grey50", yintercept = 0.75, linewidth = 1/.pt) +
  geom_hline(colour = "grey50", yintercept = 0.5, linewidth = 1/.pt) +
  scale_fill_manual(values = results_AUC_all_level_cols_Family) +
  scale_colour_manual(values = results_AUC_all_level_cols_Family) +
  theme_classic()+
  scale_y_continuous(limits = c(0, 1.7), expand = expansion(mult = c(0, 0)))+
  theme(axis.text.x=element_text(angle = -90, hjust = 0, vjust = 0.5, size = 15/.pt),
        axis.text.y=element_text(size = 15/.pt),
        axis.title = element_text(size = 15/.pt)) +
  labs(x = "", y = "AUC [AMPO 50 μM]")

fIGsupp.AMPO50

# ggsave(plot = fIGsupp.AMPO50,  filename = "fIGsupp.AMPO50.pdf", width = 8.8, height = 6, dpi = 300, scale = 1, units = "cm")
# 
# ggsave(plot = fIGsupp.AMPO50,  filename = "fIGsupp.AMPO50.svg", width = 8.8, height = 6, dpi = 300, scale = 1, units = "cm")

```


To investigate how many strains are susceptible to low concentrations of MBOA and AMPO (50 μM). We check for the number of strains with AUC_norm < 0.8 and statistically test (t.test) which bacteria grow significantly less in the treatment compared to the control. 

```{r, message=FALSE, echo=FALSE, error=FALSE, warning = FALSE, fig.dim = c(6, 5)}
results_AUC_all %>% filter(Chem_Conc %in% c("MBOA_50")) %>% 
  filter(Strain != "NBC") %>% 
  group_by(Strain, Chem_Conc) %>% 
  dplyr::summarize(AUC_norm_mean = mean(AUC_norm)) %>% 
  filter(AUC_norm_mean < 0.80) %>% 
  dplyr::select(Strain) %>% unique() %>% knitr::kable(caption = "MBOA 50 μM")

sig_MBOA_growth_AUC <- results_AUC_all %>% 
  filter(Chem_Conc %in% c("MBOA_0", "MBOA_50")) %>% 
  filter(Run %in% "R2") %>% 
  filter(Strain != "NBC") %>% 
  group_by(Strain, Chem_Conc, Replicate, Run) %>% 
  unique() %>% 
  dplyr::select(Strain, AUC, Chem_Conc, Replicate, Run) %>%
  group_by(Strain) %>%
  do(tidy(t.test(AUC~Chem_Conc, data=.))) %>% 
  add_significance("p.value") %>% 
  dplyr::select(Strain, p.value, p.value.signif)  

sig_MBOA_growth_AUC %>% 
  filter(!p.value.signif %in% "ns") %>% 
  dplyr::select(Strain) %>% unique() %>% 
  knitr::kable(caption = "MBOA 50 μM")
# 2 strains
```

```{r, message=FALSE, echo=FALSE, error=FALSE, warning = FALSE, fig.dim = c(6, 5)}
# results_AUC_all %>% filter(Chem_Conc %in% c("AMPO_50")) %>% filter(Run %in% "R1") %>% 
#   filter(Strain != "NBC") %>% group_by(Strain, Chem_Conc) %>% unique() %>% 
#   dplyr::select(Strain, AUC_norm, Chem_Conc) %>% dplyr::summarize(AUC_norm_mean = mean(AUC_norm)) %>% filter(AUC_norm_mean < 0.8) 

sig_AMPO_growth_AUC <- 
  results_AUC_all %>% 
  filter(Chem_Conc %in% c("MBOA_0", "AMPO_50")) %>% 
  filter(Run %in% "R1") %>% 
  filter(Strain != "NBC") %>% 
  group_by(Strain, Chem_Conc, Replicate, Run) %>% unique() %>% 
  dplyr::select(Strain, AUC, Chem_Conc, Replicate, Run) %>%
  group_by(Strain) %>%
  do(tidy(t.test(AUC~Chem_Conc, data=.))) %>% 
  add_significance("p.value") %>% 
  dplyr::select(Strain, p.value, p.value.signif)  

sig_AMPO_strains <- sig_AMPO_growth_AUC %>% filter(!p.value.signif %in% "ns") %>% dplyr::select(Strain) %>% unique() 
sig_AMPO_strains %>% knitr::kable(caption = "AMPO 50 μM")
# 25 strains

# strains in MBOA_50 AUC_norm < 0.75 
results_AUC_all %>% filter(Chem_Conc %in% c("AMPO_50")) %>% filter(Strain != "NBC") %>% group_by(Strain, Chem_Conc) %>% dplyr::summarize(AUC_norm_mean =mean(AUC_norm)) %>% filter(AUC_norm_mean < 0.90) %>% dplyr::select(Strain) %>% unique() %>% knitr::kable(caption = "AMPO 50 μM")
```

```{r, message=FALSE, echo=FALSE, error=FALSE, warning = FALSE, fig.dim = c(6, 5)}
AUC_all_MBOA_AMPO_50 <- results_AUC_all %>% 
  filter(Chem_Conc %in% c("MBOA_0", "MBOA_50", "AMPO_50")) %>% 
  mutate(Chem_Conc = factor(Chem_Conc, levels =c("MBOA_0", "MBOA_50", "AMPO_50"))) %>% 
  ggplot(aes(x = Chem_Conc, y = AUC_norm)) +
  stat_summary(fun.y = mean, fun.ymax = max, fun.ymin = min, alpha = 0.4) +
  geom_point(aes(colour = Family), alpha=0.4, show.legend = FALSE) + 
  stat_compare_means(label = "p.signif", method = "t.test", ref.group = "MBOA_0", label.y = 1.4, symnum.args = symnum.args) +
  ylim(c(0, 1.8))+
  facet_wrap(~ reorder(Strain, desc(Family)), nrow = 4, scales = "fixed") +
  theme_bw() +
  theme(axis.text.x = element_text(angle = -90, size = 8, hjust = 0, vjust = 0.5)) +
  scale_color_manual(values = results_AUC_all_level_cols_Family) +
  scale_x_discrete(labels = c("DMSO", "MBOA 50", "AMPO 50"))+
  labs(y = "AUC",
       x = "compound & concentration [μM]", 
       title = "growth in MBOA and AMPO",
       color = "Family")

AUC_all_MBOA_AMPO_50

# ggsave(plot = AUC_all_MBOA_AMPO_50,  filename = "AUC_all_MBOA_AMPO_50.pdf", width = 25, height = 15, dpi = 300, scale = 1, units = "cm")
# # 
# ggsave(plot = AUC_all_MBOA_AMPO_50,  filename = "AUC_all_MBOA_AMPO_50.svg", width = 25, height = 15, dpi = 300, scale = 1, units = "cm")
```

```{r, message=FALSE, echo=FALSE, error=FALSE, warning = FALSE, fig.dim = c(6, 5)}
xyAMPO_MBOA_50 <- results_AUC_all %>%
  filter(Strain != "NBC") %>% 
  mutate(Run = gsub("R4", "R1", Run)) %>% 
  filter(Chem_Conc %in% c("AMPO_50", "MBOA_50")) %>% 
  dplyr::select(Strain, Family, AUC_norm, Chem_Conc, Replicate) %>% 
  unique() %>% 
  pivot_wider(values_from = AUC_norm, names_from = Chem_Conc, values_fn = mean) %>% 
  filter(MBOA_50 < 1.5) %>% 
  filter(AMPO_50 < 1.5) %>% 
  ggplot(aes(x = MBOA_50, y = AMPO_50)) +
  geom_point(aes(colour = Family), alpha = 0.5, show.legend = FALSE) +
  geom_smooth(method = "lm", color="black", formula = y ~ x)  + 
  # xlim(0, 1.5)+
  # ylim(0, 1.5)+
  geom_smooth(method = "lm", color="black", formula = y ~ x)  + 
  stat_cor(method = "pearson", label.x = 0.7, label.y = 1.2, cex = 15/.pt) +
  scale_colour_manual(values = results_AUC_all_level_cols_Family) +
  theme_bw() +
 labs(x = "AUC norm MBOA",
       y = "AUC norm AMPO")

xyAMPO_MBOA_50

# ggsave(plot = xyAMPO_MBOA_50,  filename = "xyAMPO_MBOA_50.pdf", width = 9, height = 9, dpi = 300, scale = 1, units = "cm")
# 
# ggsave(plot = xyAMPO_MBOA_50,  filename = "xyAMPO_MBOA_50.svg", width = 9, height = 9, dpi = 300, scale = 1, units = "cm")
```


### MBOA and BOA 2500 50 μM

```{r, message=FALSE, echo=FALSE, error=FALSE, warning = FALSE, fig.dim = c(6, 5)}
fIGsupp.MBOA2500 <- results_AUC_all %>%
  filter(AUC_norm < 2) %>% 
  filter(Strain != "NBC") %>% 
  filter(Chem_Conc %in% c("MBOA_2500")) %>% 
  unique() %>% 
  ggplot(aes(y = AUC_norm, x = reorder(Strain, desc(AUC_norm)))) + 
  geom_bar(aes(fill = Family),  size = 0.1/.pt, alpha = 0.5, stat = "summary", position = "dodge", show.legend = FALSE) +
  geom_quasirandom(aes(fill = Family),  size = 0.01/.pt, alpha = 0.7, shape = 21,  width = 0.3, show.legend = FALSE)+
  geom_hline(colour = "grey50", yintercept = 1, size = 1/.pt) +
  geom_hline(colour = "grey50", yintercept = 0.75, size = 1/.pt) +
  geom_hline(colour = "grey50", yintercept = 0.5, size = 1/.pt) +
  stat_compare_means(method = "anova", aes(x = reorder(Strain, desc(AUC_norm)), y = AUC_norm), cex = 7/.pt, label.y = 1.6, label.x.npc = 0.6) +
  # stat_compare_means(label = "p.signif", method = "t.test", ref.group = "LBA3", hide.ns = TRUE, cex = 7/.pt, label.y = 1.5) +
  scale_fill_manual(values = results_AUC_all_level_cols_Family) +
  theme_classic()+
  scale_y_continuous(expand = expansion(mult = c(0, NA)))+
  ylim(0, 1.7)+
  theme(axis.text.x=element_text(angle = -90, hjust = 0, vjust = 0.5, size = 15/.pt),
        axis.text.y=element_text(size = 15/.pt),
        axis.title = element_text(size = 15/.pt)) +
  labs(x = "", y = "AUC norm")

fIGsupp.MBOA2500


# ggsave(plot = fIGsupp.MBOA2500,  filename = "fIGsupp.MBOA2500.pdf", width = 8.8, height = 6, dpi = 300, scale = 3, units = "cm")

# ggsave(plot = fIGsupp.MBOA2500,  filename = "fIGsupp.MBOA2500.svg", width = 8.8, height = 6, dpi = 300, scale = 3, units = "cm")
```

```{r, message=FALSE, echo=FALSE, error=FALSE, warning = FALSE, fig.dim = c(6, 5)}
fIGsupp.BOA2500 <- results_AUC_all %>%
  filter(AUC_norm < 2) %>% 
  filter(Strain != "NBC") %>% 
  filter(Chem_Conc %in% c("BOA_2500")) %>% 
  unique() %>% 
  ggplot(aes(y = AUC_norm, x = reorder(Strain, desc(AUC_norm)))) + 
  # geom_bar(aes(fill = Family), alpha = 0.5, stat = "summary", show.legend = FALSE) +
  # geom_jitter(aes(shape = Run), alpha = 0.5, width = 0.1, show.legend = FALSE)+
  geom_bar(aes(fill = Family),  size = 0.1/.pt, alpha = 0.5, stat = "summary", position = "dodge", show.legend = FALSE) +
  geom_quasirandom(aes(fill = Family),  size = 0.01/.pt, alpha = 0.7, shape = 21,  width = 0.3, show.legend = FALSE)+
  # geom_errorbar(stat = "summary", width=0.2) +
  geom_hline(colour = "grey50", yintercept = 1, size = 1/.pt) +
  geom_hline(colour = "grey50", yintercept = 0.75, size = 1/.pt) +
  geom_hline(colour = "grey50", yintercept = 0.5, size = 1/.pt) +
  stat_compare_means(method = "anova", aes(x = reorder(Strain, desc(AUC_norm)), y = AUC_norm), cex = 7/.pt, label.y = 1.6, label.x.npc = 0.6) +
  # stat_compare_means(label = "p.signif", method = "t.test", ref.group = "LBA3", hide.ns = TRUE, cex = 7/.pt, label.y = 1.5) +
  scale_fill_manual(values = results_AUC_all_level_cols_Family) +
  theme_classic()+
  scale_y_continuous(expand = expansion(mult = c(0, NA)))+
  ylim(0, 1.7)+
  theme(axis.text.x=element_text(angle = -90, hjust = 0, vjust = 0.5, size = 15/.pt),
        axis.text.y=element_text(size = 15/.pt),
        axis.title = element_text(size = 15/.pt)) +
  labs(x = "", y = "AUC norm")

fIGsupp.BOA2500

# ggsave(plot = fIGsupp.BOA2500,  filename = "fIGsupp.BOA2500.pdf", width = 8.8, height = 6, dpi = 300, scale = 3, units = "cm")

# ggsave(plot = fIGsupp.BOA2500,  filename = "fIGsupp.BOA2500.svg", width = 8.8, height = 6, dpi = 300, scale = 3, units = "cm")
```

To investigate how many strains are susceptible to high concentrations of MBOA and BOA (2500 μM). 
We check for the number of strains with AUC_norm < 0.75 and statistically test (t.test) which bacteria grow significantly less in the treatment compared to the control. 

```{r, message=FALSE, echo=FALSE, error=FALSE, warning = FALSE, fig.dim = c(6, 5)}
# strains in MBOA_50 AUC_norm < 0.75 
# results_AUC_all %>% filter(Chem_Conc %in% c("MBOA_2500")) %>% 
#   filter(Strain != "NBC") %>% group_by(Strain, Chem_Conc) %>% 
#   dplyr::summarize(AUC_norm_mean =mean(AUC_norm)) %>% 
#   filter(AUC_norm_mean < 0.75) %>% 
#   dplyr::select(Strain) %>% 
#   unique()

sig_MBOA_growth_AUC <- results_AUC_all %>% 
  filter(Chem_Conc %in% c("MBOA_0", "MBOA_2500")) %>% 
  filter(Strain != "NBC") %>% 
  group_by(Strain, Chem_Conc, Replicate, Run) %>% 
  unique() %>% 
  dplyr::select(Strain, AUC, Chem_Conc, Replicate, Run) %>%
  group_by(Strain) %>%
  do(tidy(t.test(AUC~Chem_Conc, data=.))) %>% 
  add_significance("p.value") %>% 
  dplyr::select(Strain, p.value, p.value.signif)  

MBOA_2500_tolerant_AUC <- sig_MBOA_growth_AUC %>% 
  filter(p.value.signif %in% "ns") %>% 
  dplyr::select(Strain) %>% 
  unique() 

MBOA_2500_tolerant_AUC %>% knitr::kable(caption = "MBOA 2500 μM")
# 14 strains

sig_BOA_growth_AUC <- results_AUC_all %>% filter(Chem_Conc %in% c("BOA_0", "BOA_2500")) %>% 
  filter(Strain != "NBC") %>% group_by(Strain, Chem_Conc, Replicate, Run) %>% unique() %>% 
  dplyr::select(Strain, AUC, Chem_Conc, Replicate, Run) %>%
  # mutate(compartment_soil = interaction(compartment, soil)) %>% 
  group_by(Strain) %>%
  do(tidy(t.test(AUC~Chem_Conc, data=.))) %>% 
  add_significance("p.value") %>% 
  dplyr::select(Strain, p.value, p.value.signif)  

BOA_2500_tolerant_AUC <- sig_BOA_growth_AUC %>% filter(p.value.signif %in% "ns") %>% dplyr::select(Strain) %>% unique() 

# BOA_2500_tolerant_AUC %>% knitr::kable(caption = "BOA 2500 μM")
# 11 strains
```

```{r, message=FALSE, echo=FALSE, error=FALSE, warning = FALSE, fig.dim = c(6, 5)}
fIGsupp.BOA1250 <- results_AUC_all %>%
  filter(AUC_norm < 2) %>% 
  filter(Strain != "NBC") %>% 
  filter(Chem_Conc %in% c("BOA_1250")) %>% 
  unique() %>% 
  ggplot(aes(y = AUC_norm, x = reorder(Strain, desc(AUC_norm)))) + 
  # geom_bar(aes(fill = Family), alpha = 0.5, stat = "summary", show.legend = FALSE) +
  # geom_jitter(aes(shape = Run), alpha = 0.5, width = 0.1, show.legend = FALSE)+
  geom_bar(aes(fill = Family),  size = 0.1/.pt, alpha = 0.5, stat = "summary", position = "dodge", show.legend = FALSE) +
  geom_quasirandom(aes(fill = Family),  size = 0.01/.pt, alpha = 0.7, shape = 21,  width = 0.3, show.legend = FALSE)+
  # geom_errorbar(stat = "summary", width=0.2) +
  geom_hline(colour = "grey50", yintercept = 1, size = 1/.pt) +
  geom_hline(colour = "grey50", yintercept = 0.75, size = 1/.pt) +
  geom_hline(colour = "grey50", yintercept = 0.5, size = 1/.pt) +
  stat_compare_means(method = "anova", aes(x = reorder(Strain, desc(AUC_norm)), y = AUC_norm), cex = 7/.pt, label.y = 1.6, label.x.npc = 0.6) +
  # stat_compare_means(label = "p.signif", method = "t.test", ref.group = "LBA3", hide.ns = TRUE, cex = 7/.pt, label.y = 1.5) +
  scale_fill_manual(values = results_AUC_all_level_cols_Family) +
  theme_classic()+
  scale_y_continuous(expand = expansion(mult = c(0, NA)))+
  ylim(0, 1.7)+
  theme(axis.text.x=element_text(angle = -90, hjust = 0, vjust = 0.5, size = 15/.pt),
        axis.text.y=element_text(size = 15/.pt),
        axis.title = element_text(size = 15/.pt)) +
  labs(x = "", y = "AUC norm")

fIGsupp.BOA1250

# ggsave(plot = fIGsupp.BOA1250,  filename = "fIGsupp.BOA1250.pdf", width = 8.8, height = 6, dpi = 300, scale = 3, units = "cm")

# ggsave(plot = fIGsupp.BOA1250,  filename = "fIGsupp.BOA1250.svg", width = 8.8, height = 6, dpi = 300, scale = 3, units = "cm")
```

### AMPO and APO [50 uM]

```{r, message=FALSE, echo=FALSE, error=FALSE, warning = FALSE, fig.dim = c(6, 5)}
fIGsupp.AMPO50 <- results_AUC_all %>%
  filter(AUC_norm < 2) %>% 
  filter(AUC_norm > 0) %>% 
  filter(Strain != "NBC") %>% 
  filter(Chem_Conc %in% c("AMPO_50")) %>% 
  filter(Run %in% c("R1")) %>% 
  unique() %>% 
  ggplot(aes(y = AUC_norm, x = reorder(Strain, desc(AUC_norm)))) + 
  # geom_bar(aes(fill = Family), alpha = 0.5, stat = "summary", show.legend = FALSE) +
  # geom_jitter(aes(shape = Run), alpha = 0.5, width = 0.1, show.legend = FALSE)+
  geom_bar(aes(fill = Family),  size = 0.1/.pt, alpha = 0.5, stat = "summary", position = "dodge", show.legend = FALSE) +
  geom_quasirandom(aes(fill = Family),  size = 0.01/.pt, alpha = 0.7, shape = 21,  width = 0.3, show.legend = FALSE)+
  # geom_errorbar(stat = "summary", width=0.2) +
  geom_hline(colour = "grey50", yintercept = 1, size = 1/.pt) +
  geom_hline(colour = "grey50", yintercept = 0.75, size = 1/.pt) +
  geom_hline(colour = "grey50", yintercept = 0.5, size = 1/.pt) +
  stat_compare_means(method = "anova", aes(x = reorder(Strain, desc(AUC_norm)), y = AUC_norm), cex = 7/.pt, label.y = 1.6, label.x.npc = 0.6) +
  # stat_compare_means(label = "p.signif", method = "t.test", ref.group = "LBA3", hide.ns = TRUE, cex = 7/.pt, label.y = 1.5) +
  scale_fill_manual(values = results_AUC_all_level_cols_Family) +
  theme_classic()+
  scale_y_continuous(expand = expansion(mult = c(0, NA)))+
  ylim(0, 1.7)+
  theme(axis.text.x=element_text(angle = -90, hjust = 0, vjust = 0.5, size = 15/.pt),
        axis.text.y=element_text(size = 15/.pt),
        axis.title = element_text(size = 15/.pt)) +
  labs(x = "", y = "AUC norm")

fIGsupp.AMPO50

# ggsave(plot = fIGsupp.AMPO50,  filename = "fIGsupp.AMPO50.pdf", width = 8.8, height = 6, dpi = 300, scale = 3, units = "cm")

# ggsave(plot = fIGsupp.AMPO50,  filename = "fIGsupp.AMPO50.svg", width = 8.8, height = 6, dpi = 300, scale = 3, units = "cm")
```

```{r, message=FALSE, echo=FALSE, error=FALSE, warning = FALSE, fig.dim = c(6, 5)}
fIGsupp.APO50 <- results_AUC_all %>%
  filter(AUC_norm < 2) %>% 
  filter(AUC_norm > 0) %>% 
  filter(Strain != "NBC") %>% 
  filter(Chem_Conc %in% c("APO_50")) %>% 
  filter(Run %in% c("R5")) %>% 
  unique() %>% 
  ggplot(aes(y = AUC_norm, x = reorder(Strain, desc(AUC_norm)))) + 
  # geom_bar(aes(fill = Family), alpha = 0.5, stat = "summary", show.legend = FALSE) +
  # geom_jitter(aes(shape = Run), alpha = 0.5, width = 0.1, show.legend = FALSE)+
  geom_bar(aes(fill = Family),  size = 0.1/.pt, alpha = 0.5, stat = "summary", position = "dodge", show.legend = FALSE) +
  geom_quasirandom(aes(fill = Family),  size = 0.01/.pt, alpha = 0.7, shape = 21,  width = 0.3, show.legend = FALSE)+
  # geom_errorbar(stat = "summary", width=0.2) +
  geom_hline(colour = "grey50", yintercept = 1, size = 1/.pt) +
  geom_hline(colour = "grey50", yintercept = 0.75, size = 1/.pt) +
  geom_hline(colour = "grey50", yintercept = 0.5, size = 1/.pt) +
  stat_compare_means(method = "anova", aes(x = reorder(Strain, desc(AUC_norm)), y = AUC_norm), cex = 7/.pt, label.y = 1.6, label.x.npc = 0.6) +
  # stat_compare_means(label = "p.signif", method = "t.test", ref.group = "LBA3", hide.ns = TRUE, cex = 7/.pt, label.y = 1.5) +
  scale_fill_manual(values = results_AUC_all_level_cols_Family) +
  theme_classic()+
  scale_y_continuous(expand = expansion(mult = c(0, NA)))+
  ylim(0, 1.7)+
  theme(axis.text.x=element_text(angle = -90, hjust = 0, vjust = 0.5, size = 15/.pt),
        axis.text.y=element_text(size = 15/.pt),
        axis.title = element_text(size = 15/.pt)) +
  labs(x = "", y = "AUC norm")

fIGsupp.APO50

# ggsave(plot = fIGsupp.APO50,  filename = "fIGsupp.APO50.pdf", width = 8.8, height = 6, dpi = 300, scale = 3, units = "cm")

# ggsave(plot = fIGsupp.APO50,  filename = "fIGsupp.APO50.svg", width = 8.8, height = 6, dpi = 300, scale = 3, units = "cm")
```

```{r, message=FALSE, echo=FALSE, error=FALSE, warning = FALSE, fig.dim = c(6, 5)}
# strains in MBOA_50 AUC_norm < 0.75 
# results_AUC_all %>% filter(Chem_Conc %in% c("MBOA_2500")) %>% filter(Strain != "NBC") %>% group_by(Strain, Chem_Conc) %>% dplyr::summarize(AUC_norm_mean =mean(AUC_norm)) %>% filter(AUC_norm_mean < 0.80) %>% dplyr::select(Strain) %>% unique()

sig_APO_growth_AUC <- results_AUC_all %>% filter(Chem_Conc %in% c("MBOA_0", "APO_50")) %>% 
  filter(Strain != "NBC") %>% group_by(Strain, Chem_Conc, Replicate, Run) %>% unique() %>% 
  dplyr::select(Strain, AUC, Chem_Conc, Replicate, Run) %>%
  # mutate(compartment_soil = interaction(compartment, soil)) %>% 
  group_by(Strain) %>%
  do(tidy(t.test(AUC~Chem_Conc, data=.))) %>% 
  add_significance("p.value") %>% 
  dplyr::select(Strain, p.value, p.value.signif)  

APO_50_tolerant_AUC <- sig_APO_growth_AUC %>% filter(p.value.signif %in% "ns") %>% dplyr::select(Strain) %>% unique()
# 27 strains

sig_BOA_growth_AUC <- results_AUC_all %>% filter(Chem_Conc %in% c("BOA_0", "BOA_2500")) %>% 
  filter(Strain != "NBC") %>% group_by(Strain, Chem_Conc, Replicate, Run) %>% unique() %>% 
  dplyr::select(Strain, AUC, Chem_Conc, Replicate, Run) %>%
  # mutate(compartment_soil = interaction(compartment, soil)) %>% 
  group_by(Strain) %>%
  do(tidy(t.test(AUC~Chem_Conc, data=.))) %>% 
  add_significance("p.value") %>% 
  dplyr::select(Strain, p.value, p.value.signif)  

BOA_2500_tolerant_AUC <- sig_BOA_growth_AUC %>% filter(p.value.signif %in% "ns") %>% dplyr::select(Strain) %>% unique()
# 11 strains

 # results_AUC_all %>% filter(Chem_Conc %in% c("AMPO_50")) %>% filter(Run %in% "R1") %>% 
 #  filter(Strain != "NBC") %>% group_by(Strain, Chem_Conc) %>% unique() %>% 
 #  dplyr::select(Strain, AUC_norm, Chem_Conc) %>% dplyr::summarize(AUC_norm_mean = mean(AUC_norm)) %>% filter(Strain %in% sig_AMPO_strains$Strain) %>% filter(AUC_norm_mean < 1) 
```

## Tolerance index 
To compare the tolerance among different bacterial strains, we use the tolerance index (TI). This tolerance index is calculated from the area under the curve of the AUC of each strain in the different concentrations of the compounds, as further normalization the normalized tolerance index is calculated from AUC_norm instead of AUC. Accordingly, a strain with TI_norm = 1 is completely tolerant to the compound in each concentration (not inhibited by the compound). Tolerance index is calculated for each compound separately using results_AUC from the respective compound. 

```{r, warning=FALSE, include=FALSE}
#results_AUC_all <- read_rds("Output/results_AUC_all.rds")
```

*TI MBOA*

```{r, warning=FALSE, include=FALSE, echo=F}
AUC_tolind_MBOA <- results_AUC_all %>% 
  filter(Run %in% c("R3", "R4")) %>% 
  filter(Chem %in% "MBOA") %>% 
  filter(!Strain %in% c("no", NA)) %>% 
  group_by(Strain, Replicate, Run) %>% 
  mutate(AUC_max = paste("1")) %>%
  dplyr::summarize(TI = auc(Conc, AUC_norm), 
            TI_raw = auc(Conc, AUC_raw),
            TI_max = auc(Conc, AUC_max)) %>% 
  mutate(TI_norm = TI / TI_max)

AUC_tolind_MBOA <- left_join(AUC_tolind_MBOA %>% as.data.frame(), Strain_Data %>% as.data.frame() %>% dplyr::select(Strain, Phylum, Family, Genus), by = "Strain") %>% unique()

## Save table with AUC
#write_rds(AUC_tolind_MBOA, paste0("Output/AUC_tolind_MBOA.rds"))
```

```{r, warning=FALSE, include=FALSE}
#AUC_tolind_MBOA <- read_rds("Output/AUC_tolind_MBOA.rds")
```

```{r, message=FALSE, echo=FALSE, error=FALSE, warning = FALSE, fig.dim = c(6, 5)}
TI_MBOA_ALL <- AUC_tolind_MBOA %>% 
  filter(!Strain %in% "NBC") %>% 
  filter(TI_norm < 1.3) %>% 
  ggplot(aes(x = reorder(Strain, desc(TI_norm)), y = TI_norm)) + # to sort the x levels using y value descending
  geom_hline(colour = "grey50", yintercept = 0.75, linewidth = 0.5/.pt) +
  geom_hline(colour = "grey50", yintercept = 0.5, linewidth = 0.5/.pt) +
  geom_bar(aes(fill = Family), stat = "summary", position = "dodge", show.legend = FALSE, alpha = 0.5) +
  ggbeeswarm::geom_quasirandom(aes(color = Family), size = 0.01/.pt, alpha = 0.7,  width = 0.3, show.legend = FALSE)+
  geom_errorbar(stat = "summary", size = 0.5/.pt, width=0.2) +
  # stat_compare_means(method = "anova", aes(x = reorder(Strain, desc(TI_norm)), y = TI_norm), cex = 10/.pt, label.y = 1.1, label.x.npc = 0.6) +
  # stat_compare_means(label = "p.signif", method = "t.test", ref.group = "LMI1x", symnum.args = symnum.args, hide.ns = TRUE, cex = 4/.pt, label.y = 1) +
  ylim(0, 1.4)+
  scale_fill_manual(values = results_AUC_all_level_cols_Family) +
  scale_color_manual(values = results_AUC_all_level_cols_Family) +
  theme_classic()+
  scale_y_continuous(limits = c(0, 1.5), expand = expansion(mult = c(0, 0)))+
  theme(axis.text.x=element_text(angle = -90, hjust = 0, vjust = 0.5, size = 15/.pt),
        axis.text.y=element_text(size = 25/.pt),
        axis.title = element_text(size = 25/.pt),
        plot.title = element_text(size = 25/.pt)) +
  labs(title = "MBOA", 
       fill = "Family", 
       y = "TI", 
       x = "")

TI_MBOA_ALL


# ggsave(plot = TI_MBOA_ALL,  filename = "TI_MBOA_ALL.pdf", width = 8.8, height = 6, dpi = 300, scale = 1, units = "cm")

# ggsave(plot = TI_MBOA_ALL,  filename = "TI_MBOA_ALL.svg", width = 8.8, height = 6, dpi = 300, scale = 1, units = "cm")
```

Tolerance index of tolerant and susceptible strain in MBOA
```{r, message=FALSE, echo=FALSE, error=FALSE, warning = FALSE, fig.dim = c(6, 5)}
fIG4.TI_LPD2_LRH8.O <- AUC_tolind_MBOA %>% 
  filter(Strain %in% c("LRH8.O", "LPD2")) %>% 
  filter(TI_norm < 1.3) %>% 
  ggplot(aes(x = reorder(Strain, desc(TI_norm)), y = TI_norm)) + # to sort the x levels using y value descending
  geom_bar(aes(fill = Family), alpha = 0.5, stat = "summary", position = "dodge", show.legend = FALSE) +
  geom_quasirandom(aes(fill = Family), size = 5/.pt, alpha = 0.7, shape = 21,  width = 0.3, show.legend = FALSE)+
  geom_errorbar(stat = "summary", width=0.2) +
  geom_hline(colour = "grey50", yintercept = 1) +
  geom_hline(colour = "grey50", yintercept = 0.75) +
  geom_hline(colour = "grey50", yintercept = 0.5) +
  # stat_compare_means(method = "anova", aes(x = reorder(Strain, desc(TI_norm)), y = TI_norm), size = 4, label.y = 1.2, label.x.npc = 0.6) +
  # stat_compare_means(label = "p.signif", method = "t.test", ref.group = "LPD2", hide.ns = TRUE, size = 4, label.y = 1.1) +
  scale_fill_manual(values = results_AUC_all_level_cols_Family) +
  theme_classic()+
  # theme(axis.text.x=element_text(angle = 0, hjust = 0, vjust = 0.5)) +
  scale_y_continuous(expand = expansion(mult = c(0, NA)))+
  ylim(0, 1.3)+
  labs(fill = "Family", 
       y = "TI",
       x = "",
       title = " ",
       subtitle = " ")

fIG4.TI_LPD2_LRH8.O

# ggsave(plot = fIG4.TI_LPD2_LRH8.O,  filename = "fIG4.TI_LPD2_LRH8.O.pdf", width = 4.4, height = 6, dpi = 300, scale = 1, units = "cm")

# ggsave(plot = fIG4.TI_LPD2_LRH8.O,  filename = "fIG4.TI_LPD2_LRH8.O.svg", width = 4.4, height = 6, dpi = 300, scale = 1, units = "cm")
```

*TI MBOA without high concentrations*
```{r, warning=FALSE, include=FALSE, echo=F}
AUC_tolind_MBOA_low <- results_AUC_all %>% 
  filter(!Conc %in% c("2500", "5000")) %>% 
  filter(Run %in% c("R3", "R4")) %>% 
  filter(Chem %in% "MBOA") %>% 
  filter(!Strain %in% c("no", NA)) %>% 
  group_by(Strain, Replicate, Run) %>% 
  mutate(AUC_max = paste("1")) %>%
  dplyr::summarize(TI = auc(Conc, AUC_norm), 
            TI_raw = auc(Conc, AUC_raw),
            TI_max = auc(Conc, AUC_max)) %>% 
  mutate(TI_norm = TI / TI_max)

AUC_tolind_MBOA_low <- left_join(AUC_tolind_MBOA_low %>% as.data.frame(), Strain_Data %>% as.data.frame() %>% dplyr::select(Strain, Phylum, Family, Genus), by = "Strain") %>% unique()

## Save table with AUC
#write_rds(AUC_tolind_MBOA_low, paste0("Output/AUC_tolind_MBOA_low.rds"))
```

```{r, message=FALSE, echo=FALSE, error=FALSE, warning = FALSE, fig.dim = c(6, 5)}
TI_MBOA_low <- AUC_tolind_MBOA_low %>% 
  filter(!Strain %in% "NBC") %>% 
  filter(TI_norm < 1.3) %>% 
  ggplot(aes(x = reorder(Strain, desc(TI_norm)), y = TI_norm)) + # to sort the x levels using y value descending
  geom_hline(colour = "grey50", yintercept = 0.75, linewidth = 0.5/.pt) +
  geom_hline(colour = "grey50", yintercept = 0.5, linewidth = 0.5/.pt) +
  geom_bar(aes(fill = Family), stat = "summary", position = "dodge", show.legend = FALSE, alpha = 0.5) +
  ggbeeswarm::geom_quasirandom(aes(color = Family), size = 0.01/.pt, alpha = 0.7,  width = 0.3, show.legend = FALSE)+
  geom_errorbar(stat = "summary", size = 0.5/.pt, width=0.2) +
  # stat_compare_means(method = "anova", aes(x = reorder(Strain, desc(TI_norm)), y = TI_norm), cex = 10/.pt, label.y = 1.1, label.x.npc = 0.6) +
  # stat_compare_means(label = "p.signif", method = "t.test", ref.group = "LMI1x", symnum.args = symnum.args, hide.ns = TRUE, cex = 4/.pt, label.y = 1) +
  ylim(0, 1.4)+
  scale_fill_manual(values = results_AUC_all_level_cols_Family) +
  scale_color_manual(values = results_AUC_all_level_cols_Family) +
  theme_classic()+
  scale_y_continuous(limits = c(0, 1.5), expand = expansion(mult = c(0, 0)))+
  theme(axis.text.x=element_text(angle = -90, hjust = 0, vjust = 0.5, size = 15/.pt),
        axis.text.y=element_text(size = 25/.pt),
        axis.title = element_text(size = 25/.pt),
        plot.title = element_text(size = 25/.pt)) +
  labs(title = "MBOA", 
       fill = "Family", 
       y = "TI low", 
       x = "")

TI_MBOA_low


# ggsave(plot = TI_MBOA_low,  filename = "TI_MBOA_low.pdf", width = 8.8, height = 6, dpi = 300, scale = 1, units = "cm")
# # 
# ggsave(plot = TI_MBOA_low,  filename = "TI_MBOA_low.svg", width = 8.8, height = 6, dpi = 300, scale = 1, units = "cm")
```

```{r, warning=FALSE, include=FALSE, echo=F}
AUC_tolind_MBOA_low_cor <- left_join(AUC_tolind_MBOA_low %>% dplyr::rename(TI_norm_low = TI_norm) %>% 
                                       dplyr::select(Strain, Replicate, Run, TI_norm_low), 
                                     AUC_tolind_MBOA %>% dplyr::select(Strain, Replicate, Run, Family, TI_norm), 
                                     join_by(Strain == Strain, Replicate == Replicate, Run == Run))

TI_low_cor <- AUC_tolind_MBOA_low_cor %>% 
  filter(Strain != "NBC") %>% 
  # filter(Family != "Xanthomonadaceae") %>% 
  # mutate(Run = gsub("R4", "R1", Run)) %>% 
  ggplot(aes(x = TI_norm, y = TI_norm_low)) +
  geom_point(aes(colour = Family), alpha = 0.5, show.legend = FALSE) +
  geom_smooth(method = "lm", color="black", formula = y ~ x)  + 
  # xlim(0, 1)+
  # ylim(0, 1)+
  geom_smooth(method = "lm", color="black", formula = y ~ x)  + 
  stat_cor(method = "pearson", label.x = 0.4, label.y = 0.8, cex = 15/.pt) +
  scale_colour_manual(values = results_AUC_all_level_cols_Family) +
  theme_bw() +
 labs(x = "TI",
       y = "TI low")

TI_low_cor

# ggsave(plot = TI_low_cor,  filename = "TI_low_cor.pdf", width = 9, height = 9, dpi = 300, scale = 1, units = "cm")

# ggsave(plot = TI_low_cor,  filename = "TI_low_cor.svg", width = 9, height = 9, dpi = 300, scale = 1, units = "cm")
```

*TI MBOA at two concentrations (500 and 2500 μM)*

```{r, warning=FALSE, include=FALSE, echo=F}
AUC_tolind_MBOA_2_conc <- results_AUC_all %>% 
  filter(Conc %in% c("0", "500", "2500")) %>% 
  filter(Run %in% c("R3", "R4")) %>% 
  filter(Chem %in% "MBOA") %>% 
  filter(!Strain %in% c("no", NA)) %>% 
  unique() %>% 
  group_by(Strain, Replicate, Run) %>% 
  mutate(AUC_max = paste("1")) %>%
  dplyr::summarize(TI = auc(Conc, AUC_norm), 
            TI_raw = auc(Conc, AUC_raw),
            TI_max = auc(Conc, AUC_max)) %>% 
  mutate(TI_norm = TI / TI_max)

AUC_tolind_MBOA_2_conc <- left_join(AUC_tolind_MBOA_2_conc %>% as.data.frame(), Strain_Data %>% as.data.frame() %>% dplyr::select(Strain, Phylum, Family, Genus) %>% unique(), by = "Strain") %>% unique()

## Save table with AUC
#write_rds(AUC_tolind_MBOA_2_conc, paste0("Output/AUC_tolind_MBOA_2_conc.rds"))
```

```{r, message=FALSE, echo=FALSE, error=FALSE, warning = FALSE, fig.dim = c(6, 5)}
TI_MBOA_2conc <- AUC_tolind_MBOA_2_conc %>% 
  filter(!Strain %in% "NBC") %>% 
  filter(TI_norm < 1.3) %>% 
  ggplot(aes(x = reorder(Strain, desc(TI_norm)), y = TI_norm)) + # to sort the x levels using y value descending
  geom_hline(colour = "grey50", yintercept = 0.75, linewidth = 0.5/.pt) +
  geom_hline(colour = "grey50", yintercept = 0.5, linewidth = 0.5/.pt) +
  geom_bar(aes(fill = Family), stat = "summary", position = "dodge", show.legend = FALSE, alpha = 0.5) +
  ggbeeswarm::geom_quasirandom(aes(color = Family), size = 0.01/.pt, alpha = 0.7,  width = 0.3, show.legend = FALSE)+
  geom_errorbar(stat = "summary", size = 0.5/.pt, width=0.2) +
  #stat_compare_means(method = "anova", aes(x = reorder(Strain, desc(TI_norm)), y = TI_norm), cex = 10/.pt, label.y = 1.1, label.x.npc = 0.6) +
  # stat_compare_means(label = "p.signif", method = "t.test", ref.group = "LMI1x", symnum.args = symnum.args, hide.ns = TRUE, cex = 4/.pt, label.y = 1) +
  ylim(0, 1.4)+
  scale_fill_manual(values = results_AUC_all_level_cols_Family) +
  scale_color_manual(values = results_AUC_all_level_cols_Family) +
  theme_classic()+
  scale_y_continuous(limits = c(0, 1.5), expand = expansion(mult = c(0, 0)))+
  theme(axis.text.x=element_text(angle = -90, hjust = 0, vjust = 0.5, size = 15/.pt),
        axis.text.y=element_text(size = 25/.pt),
        axis.title = element_text(size = 25/.pt),
        plot.title = element_text(size = 25/.pt)) +
  labs(title = "MBOA with three concentrations (0, 500 and 2500 μM)", 
       fill = "Family", 
       y = "TI", 
       x = "")

TI_MBOA_2conc


# ggsave(plot = TI_MBOA_2conc,  filename = "TI_MBOA_2conc.pdf", width = 8.8, height = 6, dpi = 300, scale = 1, units = "cm")
# 
# ggsave(plot = TI_MBOA_2conc,  filename = "TI_MBOA_2conc.svg", width = 8.8, height = 6, dpi = 300, scale = 1, units = "cm")
```

```{r, warning=FALSE, include=FALSE, echo=F}
AUC_tolind_MBOA_2conc_cor <- left_join(AUC_tolind_MBOA_2_conc %>% dplyr::rename(TI_norm_2conc = TI_norm) %>% 
                                       dplyr::select(Strain, Replicate, Run, TI_norm_2conc), 
                                     AUC_tolind_MBOA %>% dplyr::select(Strain, Replicate, Run, Family, TI_norm), 
                                     join_by(Strain == Strain, Replicate == Replicate, Run == Run))

TI_2conc_cor <- AUC_tolind_MBOA_2conc_cor %>% 
  filter(Strain != "NBC") %>% 
  # filter(Family != "Xanthomonadaceae") %>% 
  # mutate(Run = gsub("R4", "R1", Run)) %>% 
  ggplot(aes(x = TI_norm, y = TI_norm_2conc)) +
  geom_point(aes(colour = Family), alpha = 0.5, show.legend = FALSE) +
  geom_smooth(method = "lm", color="black", formula = y ~ x)  + 
  xlim(0, 1.3)+
  ylim(0, 1.3)+
  geom_smooth(method = "lm", color="black", formula = y ~ x)  + 
  stat_cor(method = "pearson", label.x = 0.2, label.y = 1.2, cex = 15/.pt) +
  scale_colour_manual(values = results_AUC_all_level_cols_Family) +
  theme_bw() +
 labs(x = "TI",
       y = "TI 2conc")

TI_2conc_cor

# ggsave(plot = TI_2conc_cor,  filename = "TI_2conc_cor.pdf", width = 9, height = 9, dpi = 300, scale = 1, units = "cm")
# # 
# ggsave(plot = TI_2conc_cor,  filename = "TI_2conc_cor.svg", width = 9, height = 9, dpi = 300, scale = 1, units = "cm")
```

*AUC MBOA 625 μM*
```{r, message=FALSE, echo=FALSE, error=FALSE, warning = FALSE, fig.dim = c(6, 5)}
AUC_all_MBOA_625 <- results_AUC_all %>% 
  filter(Chem_Conc %in% c("MBOA_0", "MBOA_625")) %>% 
  mutate(Chem_Conc = factor(Chem_Conc, levels =c("MBOA_0", "MBOA_625"))) %>% 
  ggplot(aes(x = Chem_Conc, y = AUC_norm)) +
  stat_summary(fun.y = mean, fun.ymax = max, fun.ymin = min, alpha = 0.4) +
  geom_point(aes(colour = Family), alpha=0.4, show.legend = FALSE) + 
  # stat_compare_means(label = "p.signif", method = "t.test", ref.group = "MBOA_0",  symnum.args = symnum.args, hide.ns = TRUE, label.y = 1.4) +
  ylim(c(0, 1.8))+
  facet_wrap(~ reorder(Strain, desc(Family)), nrow = 4, scales = "fixed") +
  theme_bw() +
  theme(axis.text.x = element_text(angle = -90, size = 8, hjust = 0, vjust = 0.5)) +
  scale_color_manual(values = results_AUC_all_level_cols_Family) +
  scale_x_discrete(labels = c("DMSO", "MBOA 625"))+
  labs(y = "AUC",
       x = "compound & concentration [μM]", 
       title = "growth in MBOA 625 μM",
       color = "Family")

AUC_all_MBOA_625

# ggsave(plot = AUC_all_MBOA_625,  filename = "AUC_all_MBOA_625.pdf", width = 25, height = 15, dpi = 300, scale = 1, units = "cm")
# 
# ggsave(plot = AUC_all_MBOA_625,  filename = "AUC_all_MBOA_625.svg", width = 25, height = 15, dpi = 300, scale = 1, units = "cm")
```

*TI BOA*

```{r, message=FALSE, echo=FALSE, error=FALSE, warning = FALSE, fig.dim = c(6, 5)}
AUC_tolind_BOA <- results_AUC_all %>% 
  filter(Run %in% c("R1", "R2")) %>% 
  filter(!Run %in% "R1" | !Strain %in% c("LMX92", "LMX9231", "LST11", "LMI14", "LPB4.O")) %>% 
  filter(Chem %in% "BOA") %>% 
  filter(!Strain %in% c("no", NA)) %>% 
  group_by(Strain, Replicate, Run) %>% 
  mutate(AUC_max = paste("1")) %>%
  dplyr::summarize(TI = auc(Conc, AUC_norm), 
            TI_raw = auc(Conc, AUC_raw),
            TI_max = auc(Conc, AUC_max)) %>% 
  mutate(TI_norm = TI / TI_max)

AUC_tolind_BOA <- left_join(AUC_tolind_BOA, Strain_Data %>% dplyr::select(Strain, Phylum, Family, Genus), by = "Strain") %>% unique()

## Save table with AUC
#write_rds(AUC_tolind_BOA, paste0("Output/AUC_tolind_BOA.rds"))
```

```{r, message=FALSE, echo=FALSE, error=FALSE, warning = FALSE, fig.dim = c(6, 5)}
TI_BOA_ALL <- AUC_tolind_BOA %>% 
  filter(!Strain %in% "NBC") %>% 
  filter(TI_norm < 1.3) %>% 
  ggplot(aes(x = reorder(Strain, desc(TI_norm)), y = TI_norm)) + # to sort the x levels using y value descending
  geom_hline(colour = "grey50", yintercept = 0.75, size = 0.5/.pt) +
  geom_hline(colour = "grey50", yintercept = 0.5, size = 0.5/.pt) +
  geom_bar(aes(fill = Family), stat = "summary", position = "dodge", show.legend = FALSE) +
  ggbeeswarm::geom_quasirandom(aes(color = Family), size = 0.01/.pt, alpha = 0.7,  width = 0.3, show.legend = FALSE)+
  geom_errorbar(stat = "summary", size = 0.5/.pt, width=0.2) +
  # stat_compare_means(method = "anova", aes(x = reorder(Strain, desc(TI_norm)), y = TI_norm), cex = 10/.pt, label.y = 1.2, label.x.npc = 0.6) +
  # stat_compare_means(label = "p.signif", method = "t.test", ref.group = "LMI1x", symnum.args = symnum.args, hide.ns = TRUE, cex = 4/.pt, label.y = 1.1) +
  ylim(0, 1.4)+
  scale_fill_manual(values = results_AUC_all_level_cols_Family_light) +
  scale_color_manual(values = results_AUC_all_level_cols_Family) +
  theme_classic()+
  scale_y_continuous(limits = c(0, 1.5), expand = expansion(mult = c(0, 0)))+
  theme(axis.text.x=element_text(angle = -90, hjust = 0, vjust = 0.5, size = 15/.pt),
        axis.text.y=element_text(size = 25/.pt),
        axis.title = element_text(size = 25/.pt),
        plot.title = element_text(size = 25/.pt)) +
  labs(title = "BOA", 
       fill = "Family", 
       y = "TI", 
       x = "")

TI_BOA_ALL

# ggsave(plot = TI_BOA_ALL,  filename = "TI_BOA_ALL.pdf", width = 8.8, height = 6, dpi = 300, scale = 1, units = "cm")
# 
# ggsave(plot = TI_BOA_ALL,  filename = "TI_BOA_ALL.svg", width = 8.8, height = 6, dpi = 300, scale = 1, units = "cm")
```

*TI AMPO*

```{r, message=FALSE, echo=FALSE, error=FALSE, warning = FALSE, fig.dim = c(6, 5)}
AUC_tolind_AMPO <- results_AUC_all %>% 
  filter(Run %in% c("R1", "R4")) %>% 
  filter(Chem %in% "AMPO") %>% 
  filter(!Strain %in% c("no", NA)) %>% 
  group_by(Strain, Replicate, Run, Family) %>% 
  mutate(AUC_max = paste("1")) %>%
  dplyr::summarize(TI = auc(Conc, AUC_norm), 
            TI_raw = auc(Conc, AUC_raw),
            TI_max = auc(Conc, AUC_max)) %>% 
  mutate(TI_norm = TI / TI_max)

AUC_tolind_AMPO <- left_join(AUC_tolind_AMPO, Strain_Data %>% dplyr::select(Strain, Phylum, Family, Genus), by = "Strain") %>% unique() %>% dplyr::select(-Family.x) %>% dplyr::rename(Family = Family.y)

## Save table with AUC
#write_rds(AUC_tolind_AMPO, paste0("Output/AUC_tolind_AMPO.rds"))
```

```{r, message=FALSE, echo=FALSE, error=FALSE, warning = FALSE, fig.dim = c(6, 5)}
TI_AMPO_ALL <- AUC_tolind_AMPO %>% 
  filter(!Strain %in% "NBC") %>% 
  filter(TI_norm < 1.3) %>% 
  ggplot(aes(x = reorder(Strain, desc(TI_norm)), y = TI_norm)) + # to sort the x levels using y value descending
  geom_hline(colour = "grey50", yintercept = 0.75, size = 0.5/.pt) +
  geom_hline(colour = "grey50", yintercept = 0.5, size = 0.5/.pt) +
  geom_bar(aes(fill = Family), stat = "summary", position = "dodge", show.legend = FALSE) +
  ggbeeswarm::geom_quasirandom(aes(color = Family), size = 0.01/.pt, alpha = 0.7,  width = 0.3, show.legend = FALSE)+
  geom_errorbar(stat = "summary", size = 0.5/.pt, width=0.2) +
  # stat_compare_means(method = "anova", aes(x = reorder(Strain, desc(TI_norm)), y = TI_norm), cex = 10/.pt, label.y = 1.2, label.x.npc = 0.6) +
  # stat_compare_means(label = "p.signif", method = "t.test", ref.group = "LMY1", symnum.args = symnum.args, hide.ns = TRUE, cex = 4/.pt, label.y = 1.1) +
  ylim(0, 1.4)+
  scale_fill_manual(values = results_AUC_all_level_cols_Family_light) +
  scale_color_manual(values = results_AUC_all_level_cols_Family) +
  theme_classic()+
  scale_y_continuous(limits = c(0, 1.5), expand = expansion(mult = c(0, 0)))+
  theme(axis.text.x=element_text(angle = -90, hjust = 0, vjust = 0.5, size = 15/.pt),
        axis.text.y=element_text(size = 25/.pt),
        axis.title = element_text(size = 25/.pt),
        plot.title = element_text(size = 25/.pt)) +
  labs(title = "AMPO", 
       fill = "Family", 
       y = "TI", 
       x = "")

TI_AMPO_ALL

# ggsave(plot = TI_AMPO_ALL,  filename = "TI_AMPO_ALL.pdf", width = 8.8, height = 6, dpi = 300, scale = 1, units = "cm")
# 
# ggsave(plot = TI_AMPO_ALL,  filename = "TI_AMPO_ALL.svg", width = 8.8, height = 6, dpi = 300, scale = 1, units = "cm")
```



*AUC DIMBOA-Glc 2500 uM*

```{r, message=FALSE, echo=FALSE, error=FALSE, warning = FALSE, fig.dim = c(6, 5)}
AUC_DMG_ALL <- results_AUC_all %>% 
  filter(Run %in% c("R2")) %>% 
  filter(Chem %in% "DMG") %>% 
  filter(Conc %in% "2500") %>% 
  filter(AUC_norm < 2) %>% 
  filter(Run %in% "R2") %>% 
  filter(!Strain %in% "NBC") %>%
   unique() %>% 
  ggplot(aes(x = reorder(Strain, desc(AUC_norm)), y = AUC_norm)) + # to sort the x levels using y value descending
  geom_hline(colour = "grey50", yintercept = 0.75, size = 0.5/.pt) +
  geom_hline(colour = "grey50", yintercept = 0.5, size = 0.5/.pt) +
  geom_bar(aes(fill = Family), stat = "summary", position = "dodge", show.legend = FALSE) +
  ggbeeswarm::geom_quasirandom(aes(color = Family), size = 0.01/.pt, alpha = 0.7,  width = 0.3, show.legend = FALSE)+
  geom_errorbar(stat = "summary", size = 0.5/.pt, width=0.2) +
  stat_compare_means(method = "anova", aes(x = reorder(Strain, desc(AUC_norm)), y = AUC_norm), cex = 10/.pt, label.y = 1.7, label.x.npc = 0.6) +
  # stat_compare_means(label = "p.signif", method = "t.test", ref.group = "LMI11", symnum.args = symnum.args, hide.ns = TRUE, cex = 4/.pt, label.y = 1.6) +
  ylim(0, 1.4)+
  scale_fill_manual(values = results_AUC_all_level_cols_Family_light) +
  scale_color_manual(values = results_AUC_all_level_cols_Family) +
  theme_classic()+
  scale_y_continuous(limits = c(0, 2), expand = expansion(mult = c(0, 0)))+
  theme(axis.text.x=element_text(angle = -90, hjust = 0, vjust = 0.5, size = 15/.pt),
        axis.text.y=element_text(size = 25/.pt),
        axis.title = element_text(size = 25/.pt),
        plot.title = element_text(size = 25/.pt)) +
  labs(title = "DIMBOA-Glc", 
       fill = "Family", 
       y = expression("AUC = f "^-1*"(TI)"),
       x = "")

AUC_DMG_ALL

# ggsave(plot = AUC_DMG_ALL,  filename = "AUC_DMG_ALL.pdf", width = 8.8, height = 6, dpi = 300, scale = 1, units = "cm")

# ggsave(plot = AUC_DMG_ALL,  filename = "AUC_DMG_ALL.svg", width = 8.8, height = 6, dpi = 300, scale = 1, units = "cm")
```

*TI DIMBOA-Glc*

```{r, warning=FALSE, include=FALSE, echo=F}
AUC_tolind_DMG <- results_AUC_all %>% 
  filter(AUC_norm < 2) %>% 
  filter(Chem %in% "DMG") %>% 
  filter(!Strain %in% c("no", NA)) %>% 
  group_by(Strain, Replicate, Run) %>% 
  mutate(AUC_max = paste("1")) %>%
  dplyr::summarize(TI = auc(Conc, AUC_norm), 
            TI_raw = auc(Conc, AUC_raw),
            TI_max = auc(Conc, AUC_max)) %>% 
  mutate(TI_norm = TI / TI_max)

AUC_tolind_DMG <- left_join(AUC_tolind_DMG %>% as.data.frame(), Strain_Data %>% as.data.frame() %>% dplyr::select(Strain, Phylum, Family, Genus), by = "Strain") %>% unique()

## Save table with AUC
#write_rds(AUC_tolind_DMG, paste0("Output/AUC_tolind_DMG.rds"))
```

```{r, message=FALSE, echo=FALSE, error=FALSE, warning = FALSE, fig.dim = c(6, 5)}
TI_DMG_ALL <- AUC_tolind_DMG %>% 
  filter(!Strain %in% "NBC") %>% 
  filter(TI_norm < 2) %>% 
  ggplot(aes(x = reorder(Strain, desc(TI_norm)), y = TI_norm)) + # to sort the x levels using y value descending
  geom_hline(colour = "grey50", yintercept = 0.75, linewidth = 0.5/.pt) +
  geom_hline(colour = "grey50", yintercept = 0.5, linewidth = 0.5/.pt) +
  geom_hline(colour = "grey50", yintercept = 1.0, linewidth = 0.5/.pt) +
  geom_bar(aes(fill = Family), stat = "summary", position = "dodge", show.legend = FALSE, alpha = 0.5) +
  ggbeeswarm::geom_quasirandom(aes(color = Family), size = 0.01/.pt, alpha = 0.7,  width = 0.3, show.legend = FALSE)+
  geom_errorbar(stat = "summary", size = 0.5/.pt, width=0.2) +
  # stat_compare_means(method = "anova", aes(x = reorder(Strain, desc(TI_norm)), y = TI_norm), cex = 10/.pt, label.y = 1.1, label.x.npc = 0.6) +
  # stat_compare_means(label = "p.signif", method = "t.test", ref.group = "LMI1x", symnum.args = symnum.args, hide.ns = TRUE, cex = 4/.pt, label.y = 1) +
  # ylim(0, 1.4)+
  scale_fill_manual(values = results_AUC_all_level_cols_Family) +
  scale_color_manual(values = results_AUC_all_level_cols_Family) +
  theme_classic()+
  scale_y_continuous(limits = c(0, 2), expand = expansion(mult = c(0, 0)))+
  theme(axis.text.x=element_text(angle = -90, hjust = 0, vjust = 0.5, size = 15/.pt),
        axis.text.y=element_text(size = 25/.pt),
        axis.title = element_text(size = 25/.pt),
        plot.title = element_text(size = 25/.pt)) +
  labs(title = "DIMBOA-Glc", 
       fill = "Family", 
       y = "TI", 
       x = "")

TI_DMG_ALL


# ggsave(plot = TI_DMG_ALL,  filename = "TI_DMG_ALL.pdf", width = 8.8, height = 6, dpi = 300, scale = 1, units = "cm")
# 
# ggsave(plot = TI_DMG_ALL,  filename = "TI_DMG_ALL.svg", width = 8.8, height = 6, dpi = 300, scale = 1, units = "cm")
```

correlation TI with TI low
```{r, warning=FALSE, include=FALSE, echo=F}
AUC_tolind_AUC_DMG_cor <- left_join(AUC_tolind_DMG %>% 
                                    dplyr::select(Strain, Replicate, Run, TI_norm), 
                                    results_AUC_all %>% 
                                    filter(Run %in% c("R2")) %>% 
                                    filter(Chem %in% "DMG") %>% 
                                    filter(Conc %in% "2500") %>% 
                                    filter(AUC_norm < 2) %>% 
                                    filter(Run %in% "R2") %>% 
                                    filter(!Strain %in% "NBC") %>% 
                                    dplyr::select(Strain, Replicate, Run, Family, AUC_norm), 
                                    join_by(Strain == Strain, Replicate == Replicate, Run == Run))

DMG_TI_AUC_cor <- AUC_tolind_AUC_DMG_cor %>% 
  filter(Strain != "NBC") %>% 
  filter(!AUC_norm %in% NA) %>% 
  # filter(Family != "Xanthomonadaceae") %>% 
  # mutate(Run = gsub("R4", "R1", Run)) %>% 
  ggplot(aes(x = TI_norm, y = AUC_norm)) +
  geom_point(aes(colour = Family), alpha = 0.2, show.legend = FALSE) +
  geom_smooth(method = "lm", color="black", formula = y ~ x)  + 
  # xlim(0, 1)+
  # ylim(0, 1)+
  stat_cor(method = "pearson", label.x = 0.4, label.y = 1.8, cex = 15/.pt) +
  scale_colour_manual(values = results_AUC_all_level_cols_Family) +
  theme_bw() +
 labs(x = "TI",
       y = "AUC")

DMG_TI_AUC_cor

# ggsave(plot = DMG_TI_AUC_cor,  filename = "DMG_TI_AUC_cor.pdf", width = 9, height = 9, dpi = 300, scale = 1, units = "cm")
# 
# ggsave(plot = DMG_TI_AUC_cor,  filename = "DMG_TI_AUC_cor.svg", width = 9, height = 9, dpi = 300, scale = 1, units = "cm")
```




*TI APO*

```{r, message=FALSE, echo=FALSE, error=FALSE, warning = FALSE, fig.dim = c(6, 5)}
AUC_tolind_APO <- results_AUC_all %>% 
  filter(Run %in% c("R5", "R6")) %>% 
  filter(Chem %in% "APO") %>% 
  filter(!Strain %in% c("no", NA)) %>% 
  group_by(Strain, Replicate, Run, Family) %>% 
  mutate(AUC_max = paste("1")) %>%
  dplyr::summarize(TI = auc(Conc, AUC_norm), 
            TI_raw = auc(Conc, AUC_raw),
            TI_max = auc(Conc, AUC_max)) %>% 
  mutate(TI_norm = TI / TI_max)

AUC_tolind_APO <- left_join(AUC_tolind_APO, Strain_Data %>% dplyr::select(Strain, Phylum, Family, Genus), by = "Strain") %>% unique() %>% dplyr::select(-Family.x) %>% dplyr::rename(Family = Family.y)

## Save table with AUC
#write_rds(AUC_tolind_APO, paste0("Output/AUC_tolind_APO.rds"))
```

```{r, message=FALSE, echo=FALSE, error=FALSE, warning = FALSE, fig.dim = c(6, 5)}
TI_APO_ALL <- AUC_tolind_APO %>% 
  filter(!Strain %in% "NBC") %>% 
  filter(TI_norm < 1.3) %>% 
  ggplot(aes(x = reorder(Strain, desc(TI_norm)), y = TI_norm)) + # to sort the x levels using y value descending
  geom_hline(colour = "grey50", yintercept = 0.75, size = 0.5/.pt) +
  geom_hline(colour = "grey50", yintercept = 0.5, size = 0.5/.pt) +
  geom_bar(aes(fill = Family), stat = "summary", position = "dodge", show.legend = FALSE) +
  ggbeeswarm::geom_quasirandom(aes(color = Family), size = 0.01/.pt, alpha = 0.7,  width = 0.3, show.legend = FALSE)+
  geom_errorbar(stat = "summary", size = 0.5/.pt, width=0.2) +
  # stat_compare_means(method = "anova", aes(x = reorder(Strain, desc(TI_norm)), y = TI_norm), cex = 10/.pt, label.y = 1.2, label.x.npc = 0.6) +
  # stat_compare_means(label = "p.signif", method = "t.test", ref.group = "LBA112", symnum.args = symnum.args, hide.ns = TRUE, cex = 4/.pt, label.y = 1.1) +
  ylim(0, 1.4)+
  scale_fill_manual(values = results_AUC_all_level_cols_Family_light) +
  scale_color_manual(values = results_AUC_all_level_cols_Family) +
  theme_classic()+
  scale_y_continuous(limits = c(0, 1.5), expand = expansion(mult = c(0, 0)))+
  theme(axis.text.x=element_text(angle = -90, hjust = 0, vjust = 0.5, size = 15/.pt),
        axis.text.y=element_text(size = 25/.pt),
        axis.title = element_text(size = 25/.pt),
        plot.title = element_text(size = 25/.pt)) +
  labs(title = "APO", 
       fill = "Family", 
       y = "TI", 
       x = "")

TI_APO_ALL

# ggsave(plot = TI_APO_ALL,  filename = "TI_APO_ALL.pdf", width = 8.8, height = 6, dpi = 300, scale = 1, units = "cm")
# # 
# ggsave(plot = TI_APO_ALL,  filename = "TI_APO_ALL.svg", width = 8.8, height = 6, dpi = 300, scale = 1, units = "cm")
```

## p-value heatmaps

```{r, message=FALSE, echo=FALSE, error=FALSE, warning = FALSE, fig.dim = c(6, 5)}
AUC_tolind_MBOA_mean <- AUC_tolind_MBOA %>% filter(TI_norm < 1.3) %>%  group_by(Strain) %>% dplyr::summarise(TI_norm_mean = mean(TI_norm))
AUC_tolind_MBOA_sorted <- left_join(AUC_tolind_MBOA %>% filter(TI_norm < 1.3), AUC_tolind_MBOA_mean, by = "Strain") %>% as.data.frame() %>% dplyr::arrange(desc(TI_norm_mean))

AUC_tolind_MBOA_sorted_strains <- AUC_tolind_MBOA_sorted$Strain %>% unique()

MBOA.TI.aov <- AUC_tolind_MBOA_sorted %>% aov(TI_norm ~ Strain, data = .)

MBOA.TI.tukey <- AUC_tolind_MBOA_sorted %>% aov(TI_norm ~ Strain, data = .) %>% tukey_hsd(ordered = TRUE, conf.level = 0.95)

MBOA.TI.tukey.sorted <- left_join(MBOA.TI.tukey, AUC_tolind_MBOA_mean, join_by(group1 == Strain) )

MBOA.AUC.TI.all.tukey.sorted <- left_join(AUC_tolind_MBOA %>% filter(TI_norm < 1.3), MBOA.TI.tukey, join_by(Strain == group1))


MBOA_p_HM_bin <-
  MBOA.AUC.TI.all.tukey.sorted %>%
  mutate(binary = case_when(p.adj > 0.05 ~ "ns", p.adj < 0.05 ~ "sig")) %>% 
  mutate(group2 = factor(group2, level = rev(AUC_tolind_MBOA_sorted_strains)),
          group1 = factor(Strain, level = AUC_tolind_MBOA_sorted_strains)) %>%
  ggplot(aes(x=group1, y = group2))+
  geom_tile(aes(fill = binary), color = "white", show.legend = TRUE)+
  scale_fill_manual(values = c("darkblue", "darkred"), labels = c("ns", "sig", ""), na.value = "white") +
  theme_classic()+ 
  theme(axis.text.x = element_text(angle = -90, size = 8, hjust = 0, vjust = 0.5), axis.text.y = element_text(angle = 0, size = 8, hjust = 0, vjust = 0))+
  theme(legend.position = c(0.2, 0.2))+
  labs(x = "", y = "")

# MBOA_p_HM_bin

MBOA_p_HM_bar <- AUC_tolind_MBOA %>% 
  filter(!Strain %in% "NBC") %>% 
  filter(TI_norm < 1.3) %>% 
  ggplot(aes(x = reorder(Strain, desc(TI_norm)), y = TI_norm)) + # to sort the x levels using y value descending
  geom_bar(aes(fill = Family), stat = "summary", position = "dodge", show.legend = FALSE, alpha = 0.5) +
  ggbeeswarm::geom_quasirandom(aes(color = Family), size = 0.01/.pt, alpha = 0.7,  width = 0.3, show.legend = FALSE)+
  geom_errorbar(stat = "summary", size = 0.5/.pt, width=0.2) +
  scale_fill_manual(values = results_AUC_all_level_cols_Family) +
  scale_color_manual(values = results_AUC_all_level_cols_Family) +
  theme_classic()+
  scale_y_continuous(limits = c(0, 1.5), expand = expansion(mult = c(0, 0)))+
  theme(axis.text.x=element_blank())+
  labs(x = "", y = "TI")
  
#MBOA_p_HM_bar


g2 <- ggplotGrob(MBOA_p_HM_bar)
g3 <- ggplotGrob(MBOA_p_HM_bin)
MBOA_p_HM_bar_bin <- rbind(g2, g3)
MBOA_p_HM_bar_bin$widths <- unit.pmax(g2$widths, g3$widths)
grid.newpage()
grid.draw(MBOA_p_HM_bar_bin)
 
# svg("MBOA_p_HM_bar_bin.svg", height = 12, width = 7)
# grid.draw(MBOA_p_HM_bar_bin)
# dev.off()

```

```{r, message=FALSE, echo=FALSE, error=FALSE, warning = FALSE, fig.dim = c(6, 5)}
AUC_tolind_BOA_mean <- AUC_tolind_BOA %>% filter(TI_norm < 1.3) %>%  group_by(Strain) %>% dplyr::summarise(TI_norm_mean = mean(TI_norm))
AUC_tolind_BOA_sorted <- left_join(AUC_tolind_BOA %>% filter(TI_norm < 1.3), AUC_tolind_BOA_mean, by = "Strain") %>% as.data.frame() %>% dplyr::arrange(desc(TI_norm_mean))

AUC_tolind_BOA_sorted_strains <- AUC_tolind_BOA_sorted$Strain %>% unique()

BOA.TI.aov <- AUC_tolind_BOA_sorted %>% aov(TI_norm ~ Strain, data = .)

BOA.TI.tukey <- AUC_tolind_BOA_sorted %>% aov(TI_norm ~ Strain, data = .) %>% tukey_hsd(ordered = TRUE, conf.level = 0.95)

BOA.TI.tukey.sorted <- left_join(BOA.TI.tukey, AUC_tolind_BOA_mean, join_by(group1 == Strain) )

BOA.AUC.TI.all.tukey.sorted <- left_join(AUC_tolind_BOA %>% filter(TI_norm < 1.3), BOA.TI.tukey, join_by(Strain == group1))


BOA_p_HM_bin <-
  BOA.AUC.TI.all.tukey.sorted %>%
  mutate(binary = case_when(p.adj > 0.05 ~ "ns", p.adj < 0.05 ~ "sig")) %>% 
  mutate(group2 = factor(group2, level = rev(AUC_tolind_BOA_sorted_strains)),
          group1 = factor(Strain, level = AUC_tolind_BOA_sorted_strains)) %>%
  ggplot(aes(x=group1, y = group2))+
  geom_tile(aes(fill = binary), color = "white", show.legend = TRUE)+
  scale_fill_manual(values = c("darkblue", "darkred"), labels = c("ns", "sig", ""), na.value = "white") +
  theme_classic()+ 
  theme(axis.text.x = element_text(angle = -90, size = 8, hjust = 0, vjust = 0.5), axis.text.y = element_text(angle = 0, size = 8, hjust = 0, vjust = 0))+
  theme(legend.position = c(0.2, 0.2))+
  labs(x = "", y = "")

# BOA_p_HM_bin

BOA_p_HM_bar <- AUC_tolind_BOA %>% 
  filter(!Strain %in% "NBC") %>% 
  filter(TI_norm < 1.3) %>% 
  ggplot(aes(x = reorder(Strain, desc(TI_norm)), y = TI_norm)) + # to sort the x levels using y value descending
  geom_bar(aes(fill = Family), stat = "summary", position = "dodge", show.legend = FALSE, alpha = 0.5) +
  ggbeeswarm::geom_quasirandom(aes(color = Family), size = 0.01/.pt, alpha = 0.7,  width = 0.3, show.legend = FALSE)+
  geom_errorbar(stat = "summary", size = 0.5/.pt, width=0.2) +
  scale_fill_manual(values = results_AUC_all_level_cols_Family) +
  scale_color_manual(values = results_AUC_all_level_cols_Family) +
  theme_classic()+
  scale_y_continuous(limits = c(0, 1.5), expand = expansion(mult = c(0, 0)))+
  theme(axis.text.x=element_blank())+
  labs(x = "", y = "TI")
  
#BOA_p_HM_bar


g2 <- ggplotGrob(BOA_p_HM_bar)
g3 <- ggplotGrob(BOA_p_HM_bin)
BOA_p_HM_bar_bin <- rbind(g2, g3)
BOA_p_HM_bar_bin$widths <- unit.pmax(g2$widths, g3$widths)
grid.newpage()
grid.draw(BOA_p_HM_bar_bin)
 
# svg("BOA_p_HM_bar_bin.svg", height = 12, width = 7)
# grid.draw(BOA_p_HM_bar_bin)
# dev.off()

```

```{r, message=FALSE, echo=FALSE, error=FALSE, warning = FALSE, fig.dim = c(6, 5)}
AUC_tolind_AMPO_mean <- AUC_tolind_AMPO %>% filter(TI_norm < 1.3) %>%  group_by(Strain) %>% dplyr::summarise(TI_norm_mean = mean(TI_norm))
AUC_tolind_AMPO_sorted <- left_join(AUC_tolind_AMPO %>% filter(TI_norm < 1.3), AUC_tolind_AMPO_mean, by = "Strain") %>% as.data.frame() %>% dplyr::arrange(desc(TI_norm_mean))

AUC_tolind_AMPO_sorted_strains <- AUC_tolind_AMPO_sorted$Strain %>% unique()

AMPO.TI.aov <- AUC_tolind_AMPO_sorted %>% aov(TI_norm ~ Strain, data = .)

AMPO.TI.tukey <- AUC_tolind_AMPO_sorted %>% aov(TI_norm ~ Strain, data = .) %>% tukey_hsd(ordered = TRUE, conf.level = 0.95)

AMPO.TI.tukey.sorted <- left_join(AMPO.TI.tukey, AUC_tolind_AMPO_mean, join_by(group1 == Strain) )

AMPO.AUC.TI.all.tukey.sorted <- left_join(AUC_tolind_AMPO %>% filter(TI_norm < 1.3), AMPO.TI.tukey, join_by(Strain == group1))


AMPO_p_HM_bin <-
  AMPO.AUC.TI.all.tukey.sorted %>%
  mutate(binary = case_when(p.adj > 0.05 ~ "ns", p.adj < 0.05 ~ "sig")) %>% 
  mutate(group2 = factor(group2, level = rev(AUC_tolind_AMPO_sorted_strains)),
          group1 = factor(Strain, level = AUC_tolind_AMPO_sorted_strains)) %>%
  ggplot(aes(x=group1, y = group2))+
  geom_tile(aes(fill = binary), color = "white", show.legend = TRUE)+
  scale_fill_manual(values = c("darkblue", "darkred"), labels = c("ns", "sig", ""), na.value = "white") +
  theme_classic()+ 
  theme(axis.text.x = element_text(angle = -90, size = 8, hjust = 0, vjust = 0.5), axis.text.y = element_text(angle = 0, size = 8, hjust = 0, vjust = 0))+
  theme(legend.position = c(0.2, 0.2))+
  labs(x = "", y = "")

# AMPO_p_HM_bin

AMPO_p_HM_bar <- AUC_tolind_AMPO %>% 
  filter(!Strain %in% "NBC") %>% 
  filter(TI_norm < 1.3) %>% 
  ggplot(aes(x = reorder(Strain, desc(TI_norm)), y = TI_norm)) + # to sort the x levels using y value descending
  geom_bar(aes(fill = Family), stat = "summary", position = "dodge", show.legend = FALSE, alpha = 0.5) +
  ggbeeswarm::geom_quasirandom(aes(color = Family), size = 0.01/.pt, alpha = 0.7,  width = 0.3, show.legend = FALSE)+
  geom_errorbar(stat = "summary", size = 0.5/.pt, width=0.2) +
  scale_fill_manual(values = results_AUC_all_level_cols_Family) +
  scale_color_manual(values = results_AUC_all_level_cols_Family) +
  theme_classic()+
  scale_y_continuous(limits = c(0, 1.5), expand = expansion(mult = c(0, 0)))+
  theme(axis.text.x=element_blank())+
  labs(x = "", y = "TI")
  
#AMPO_p_HM_bar


g2 <- ggplotGrob(AMPO_p_HM_bar)
g3 <- ggplotGrob(AMPO_p_HM_bin)
AMPO_p_HM_bar_bin <- rbind(g2, g3)
AMPO_p_HM_bar_bin$widths <- unit.pmax(g2$widths, g3$widths)
grid.newpage()
grid.draw(AMPO_p_HM_bar_bin)
 
# svg("AMPO_p_HM_bar_bin.svg", height = 12, width = 7)
# grid.draw(AMPO_p_HM_bar_bin)
# dev.off()

```

```{r, message=FALSE, echo=FALSE, error=FALSE, warning = FALSE, fig.dim = c(6, 5)}
AUC_tolind_DMG_mean <- AUC_tolind_DMG %>% filter(TI_norm < 2) %>%  group_by(Strain) %>% dplyr::summarise(TI_norm_mean = mean(TI_norm))
AUC_tolind_DMG_sorted <- left_join(AUC_tolind_DMG %>% filter(TI_norm < 2), AUC_tolind_DMG_mean, by = "Strain") %>% as.data.frame() %>% dplyr::arrange(desc(TI_norm_mean))

AUC_tolind_DMG_sorted_strains <- AUC_tolind_DMG_sorted$Strain %>% unique()

DMG.TI.aov <- AUC_tolind_DMG_sorted %>% aov(TI_norm ~ Strain, data = .)

DMG.TI.tukey <- AUC_tolind_DMG_sorted %>% aov(TI_norm ~ Strain, data = .) %>% tukey_hsd(ordered = TRUE, conf.level = 0.95)

DMG.TI.tukey.sorted <- left_join(DMG.TI.tukey, AUC_tolind_DMG_mean, join_by(group1 == Strain) )

DMG.AUC.TI.all.tukey.sorted <- left_join(AUC_tolind_DMG %>% filter(TI_norm < 2), DMG.TI.tukey, join_by(Strain == group1))


DMG_p_HM_bin <-
  DMG.AUC.TI.all.tukey.sorted %>%
  mutate(binary = case_when(p.adj > 0.05 ~ "ns", p.adj < 0.05 ~ "sig")) %>% 
  mutate(group2 = factor(group2, level = rev(AUC_tolind_DMG_sorted_strains)),
          group1 = factor(Strain, level = AUC_tolind_DMG_sorted_strains)) %>%
  ggplot(aes(x=group1, y = group2))+
  geom_tile(aes(fill = binary), color = "white", show.legend = TRUE)+
  scale_fill_manual(values = c("darkblue", "darkred"), labels = c("ns", "sig", ""), na.value = "white") +
  theme_classic()+ 
  theme(axis.text.x = element_text(angle = -90, size = 8, hjust = 0, vjust = 0.5), axis.text.y = element_text(angle = 0, size = 8, hjust = 0, vjust = 0))+
  theme(legend.position = c(0.2, 0.2))+
  labs(x = "", y = "")

# DMG_p_HM_bin

DMG_p_HM_bar <- AUC_tolind_DMG %>% 
  filter(!Strain %in% "NBC") %>% 
  filter(TI_norm < 2) %>% 
  ggplot(aes(x = reorder(Strain, desc(TI_norm)), y = TI_norm)) + # to sort the x levels using y value descending
  geom_bar(aes(fill = Family), stat = "summary", position = "dodge", show.legend = FALSE, alpha = 0.5) +
  ggbeeswarm::geom_quasirandom(aes(color = Family), size = 0.01/.pt, alpha = 0.7,  width = 0.3, show.legend = FALSE)+
  geom_errorbar(stat = "summary", size = 0.5/.pt, width=0.2) +
  scale_fill_manual(values = results_AUC_all_level_cols_Family) +
  scale_color_manual(values = results_AUC_all_level_cols_Family) +
  theme_classic()+
  scale_y_continuous(limits = c(0, 2), expand = expansion(mult = c(0, 0)))+
  theme(axis.text.x=element_blank())+
  labs(x = "", y = "TI")
  
#DMG_p_HM_bar


g2 <- ggplotGrob(DMG_p_HM_bar)
g3 <- ggplotGrob(DMG_p_HM_bin)
DMG_p_HM_bar_bin <- rbind(g2, g3)
DMG_p_HM_bar_bin$widths <- unit.pmax(g2$widths, g3$widths)
grid.newpage()
grid.draw(DMG_p_HM_bar_bin)
 
# svg("DMG_p_HM_bar_bin.svg", height = 12, width = 7)
# grid.draw(DMG_p_HM_bar_bin)
# dev.off()

```

```{r, message=FALSE, echo=FALSE, error=FALSE, warning = FALSE, fig.dim = c(6, 5)}
AUC_tolind_DMG_mean <- AUC_tolind_DMG %>% filter(TI_norm < 1.3) %>%  group_by(Strain) %>% dplyr::summarise(TI_norm_mean = mean(TI_norm))
AUC_tolind_DMG_sorted <- left_join(AUC_tolind_DMG %>% filter(TI_norm < 1.3), AUC_tolind_DMG_mean, by = "Strain") %>% as.data.frame() %>% dplyr::arrange(desc(TI_norm_mean))

AUC_tolind_DMG_sorted_strains <- AUC_tolind_DMG_sorted$Strain %>% unique()

DMG.TI.tukey <- AUC_tolind_DMG_sorted %>% aov(TI_norm ~ Strain, data = .) %>% tukey_hsd(ordered = TRUE, conf.level = 0.95)

DMG.TI.tukey.sorted <- left_join(DMG.TI.tukey, AUC_tolind_DMG_mean, join_by(group1 == Strain) )

DMG_p_HM <- DMG.TI.tukey.sorted %>%
  mutate(binary = case_when(p.adj > 0.05 ~ "ns", p.adj < 0.05 ~ "sig")) %>% 
  mutate(group2 = factor(group2, level = AUC_tolind_DMG_sorted_strains),
           group1 = factor(group1, level = rev(AUC_tolind_DMG_sorted_strains))) %>%
  arrange(group1, group2) %>% 
  ggplot(aes(x=group1, y = group2, fill=binary))+
  geom_tile(color = "white")+
  scale_fill_manual(values = c("darkblue", "darkred"), na.value = "white") +
  theme_classic()+ 
  theme(axis.text.x = element_text(angle = -90, size = 8, hjust = 0, vjust = 0.5), axis.text.y = element_text(angle = 0, size = 8, hjust = 0, vjust = 0))+
  coord_fixed() +
  labs(x = "susceptible -> tolerant",
         y = "susceptible -> tolerant",
       fill = "",
         title = "DIMBOA-Glc tolerance index",
         subtitle = "p-values of pairwise comparisons")

DMG_p_HM

# ggsave(plot = DMG_p_HM,  filename = "DMG_p_HM.pdf", width = 19, height = 16, dpi = 300, scale = 1, units = "cm")
# # 
# ggsave(plot = DMG_p_HM,  filename = "DMG_p_HM.svg", width = 19, height = 16, dpi = 300, scale = 1, units = "cm")
```



*p-value APO*
```{r, message=FALSE, echo=FALSE, error=FALSE, warning = FALSE, fig.dim = c(6, 5)}
AUC_tolind_APO_mean <- AUC_tolind_APO %>% filter(TI_norm < 1.3) %>%  group_by(Strain) %>% dplyr::summarise(TI_norm_mean = mean(TI_norm))
AUC_tolind_APO_sorted <- left_join(AUC_tolind_APO %>% filter(TI_norm < 1.3), AUC_tolind_APO_mean, by = "Strain") %>% as.data.frame() %>% dplyr::arrange(desc(TI_norm_mean))

AUC_tolind_APO_sorted_strains <- AUC_tolind_APO_sorted$Strain %>% unique()

APO.TI.aov <- AUC_tolind_APO_sorted %>% aov(TI_norm ~ Strain, data = .)

APO.TI.tukey <- AUC_tolind_APO_sorted %>% aov(TI_norm ~ Strain, data = .) %>% tukey_hsd(ordered = TRUE, conf.level = 0.95)

APO.TI.tukey.sorted <- left_join(APO.TI.tukey, AUC_tolind_APO_mean, join_by(group1 == Strain) )

APO.AUC.TI.all.tukey.sorted <- left_join(AUC_tolind_APO %>% filter(TI_norm < 1.3), APO.TI.tukey, join_by(Strain == group1))


APO_p_HM_bin <-
  APO.AUC.TI.all.tukey.sorted %>%
  mutate(binary = case_when(p.adj > 0.05 ~ "ns", p.adj < 0.05 ~ "sig")) %>% 
  mutate(group2 = factor(group2, level = rev(AUC_tolind_APO_sorted_strains)),
          group1 = factor(Strain, level = AUC_tolind_APO_sorted_strains)) %>%
  ggplot(aes(x=group1, y = group2))+
  geom_tile(aes(fill = binary), color = "white", show.legend = TRUE)+
  scale_fill_manual(values = c("darkblue", "darkred"), labels = c("ns", "sig", ""), na.value = "white") +
  theme_classic()+ 
  theme(axis.text.x = element_text(angle = -90, size = 8, hjust = 0, vjust = 0.5), axis.text.y = element_text(angle = 0, size = 8, hjust = 0, vjust = 0))+
  theme(legend.position = c(0.2, 0.2))+
  labs(x = "", y = "")

# APO_p_HM_bin

APO_p_HM_bar <- AUC_tolind_APO %>% 
  filter(!Strain %in% "NBC") %>% 
  filter(TI_norm < 1.3) %>% 
  ggplot(aes(x = reorder(Strain, desc(TI_norm)), y = TI_norm)) + # to sort the x levels using y value descending
  geom_bar(aes(fill = Family), stat = "summary", position = "dodge", show.legend = FALSE, alpha = 0.5) +
  ggbeeswarm::geom_quasirandom(aes(color = Family), size = 0.01/.pt, alpha = 0.7,  width = 0.3, show.legend = FALSE)+
  geom_errorbar(stat = "summary", size = 0.5/.pt, width=0.2) +
  scale_fill_manual(values = results_AUC_all_level_cols_Family) +
  scale_color_manual(values = results_AUC_all_level_cols_Family) +
  theme_classic()+
  scale_y_continuous(limits = c(0, 1.5), expand = expansion(mult = c(0, 0)))+
  theme(axis.text.x=element_blank())+
  labs(x = "", y = "TI")
  
#APO_p_HM_bar


g2 <- ggplotGrob(APO_p_HM_bar)
g3 <- ggplotGrob(APO_p_HM_bin)
APO_p_HM_bar_bin <- rbind(g2, g3)
APO_p_HM_bar_bin$widths <- unit.pmax(g2$widths, g3$widths)
grid.newpage()
grid.draw(APO_p_HM_bar_bin)
 
# svg("APO_p_HM_bar_bin.svg", height = 12, width = 7)
# grid.draw(APO_p_HM_bar_bin)
# dev.off()

```

## Correlation tolerance index 
To investigate if the tolerance of maize root bacteria to different benzoxazinonids and aminophenoxazinones depend on each other, we correlated the TIs of all bacteria in the different chemicals with each other. 

```{r, message=FALSE, echo=FALSE, error=FALSE, warning = FALSE, fig.dim = c(6, 5)}
AUC_tolind_MBOA$Chem <- "MBOA"
AUC_tolind_AMPO$Chem <- "AMPO"
AUC_tolind_BOA$Chem <- "BOA"
AUC_tolind_APO$Chem <- "APO"
AUC_tolind_DMG$Chem <- "DMG"

AUC_tolind_MBOA$Strain_Replicate <- interaction(AUC_tolind_MBOA$Strain, AUC_tolind_MBOA$Replicate)
AUC_tolind_AMPO$Strain_Replicate <- interaction(AUC_tolind_AMPO$Strain, AUC_tolind_AMPO$Replicate)
AUC_tolind_BOA$Strain_Replicate <- interaction(AUC_tolind_BOA$Strain, AUC_tolind_BOA$Replicate)
AUC_tolind_APO$Strain_Replicate <- interaction(AUC_tolind_APO$Strain, AUC_tolind_APO$Replicate)
AUC_tolind_DMG$Strain_Replicate <- interaction(AUC_tolind_DMG$Strain, AUC_tolind_DMG$Replicate)


# AUC_2500_DMG <- results_AUC_all %>% 
#   filter(Run %in% c("R2")) %>% 
#   filter(Chem %in% "DMG") %>% 
#   filter(Conc %in% "2500") %>% 
#   dplyr::select(Strain, Replicate, Run, AUC, AUC_raw, AUC_pos, AUC_norm, Phylum, Family, Genus, Chem) %>% 
#   dplyr::rename(TI = AUC, TI_raw = AUC_raw, TI_max = AUC_pos, TI_norm = AUC_norm)
# 
# AUC_2500_DMG$Strain_Replicate <- interaction(AUC_2500_DMG$Strain, AUC_2500_DMG$Replicate)

AUC_tolind_MBOA_AMPO_BOA_APO_DMG <- rbind(AUC_tolind_MBOA, AUC_tolind_AMPO, AUC_tolind_APO, AUC_tolind_BOA, AUC_tolind_DMG)

AUC_tolind_MBOA_AMPO_BOA_APO_DMG_cor <- AUC_tolind_MBOA_AMPO_BOA_APO_DMG %>% 
  mutate(Run = replace(Run, Run == "R3", "R1")) %>% 
  mutate(Run = replace(Run, Run == "R4", "R2")) %>% 
  mutate(Run = replace(Run, Run == "R5", "R1")) %>% 
  mutate(Run = replace(Run, Run == "R6", "R2")) %>% 
  dplyr::select(Strain, Replicate, Phylum, Family, Genus, Run, TI_norm, Chem) %>%  
  pivot_wider(values_from = TI_norm, names_from = Chem, values_fn = mean) %>% 
  dplyr::rename(TI_norm_MBOA = MBOA, TI_norm_AMPO = AMPO, TI_norm_BOA = BOA, TI_norm_APO = APO, TI_norm_DMG = DMG)

#write_rds(AUC_tolind_MBOA_AMPO_BOA_APO_DMG_cor, "Output/AUC_tolind_MBOA_AMPO_BOA_APO_DMG_cor.rds")
```

```{r, include=FALSE}
#AUC_tolind_MBOA_AMPO_BOA_APO_DMG_cor <- read_rds("Output/AUC_tolind_MBOA_AMPO_BOA_APO_DMG_cor.rds")

write.csv(AUC_tolind_MBOA_AMPO_BOA_APO_DMG_cor, "Output/AUC_tolind_MBOA_AMPO_BOA_APO_DMG_cor.csv")
```

*MBOA ~ AMPO*

```{r, message=FALSE, echo=FALSE, error=FALSE, warning = FALSE, fig.dim = c(5, 5)}
fIG4.TIcor_MBOA_AMPO <- AUC_tolind_MBOA_AMPO_BOA_APO_DMG_cor %>% 
  filter(!Strain %in% "LMX8") %>% 
  filter(TI_norm_AMPO < 1.2) %>%
  filter(TI_norm_MBOA < 1.2) %>%
  ggplot(aes(x = TI_norm_MBOA, y = TI_norm_AMPO)) +
  geom_point(aes(colour = Family), alpha = 0.5, show.legend = FALSE) +
  geom_smooth(method = "lm", color= "black", formula = y ~ x)  + 
  xlim(0, 1.25)+
  ylim(0, 1.25)+
  stat_cor(method = "pearson", label.x = 0.2, label.y = 1.25, cex = 15/.pt) +
  scale_colour_manual(values = results_AUC_all_level_cols_Family) +
  theme_bw() +
  labs(x = "TI MBOA",
       y = "TI AMPO")

fIG4.TIcor_MBOA_AMPO

# ggsave(plot = fIG4.TIcor_MBOA_AMPO,  filename = "fIG4.TIcor_MBOA_AMPO.pdf", width = 9, height = 9, dpi = 300, scale = 1, units = "cm")
# 
# ggsave(plot = fIG4.TIcor_MBOA_AMPO,  filename = "fIG4.TIcor_MBOA_AMPO.svg", width = 9, height = 9, dpi = 300, scale = 1, units = "cm")
```

```{r, include=FALSE }
aov(TI_norm_MBOA ~ TI_norm_AMPO, data = AUC_tolind_MBOA_AMPO_BOA_APO_DMG_cor %>% filter(TI_norm_AMPO < 1.2) %>%
  filter(TI_norm_MBOA < 1.2)) %>% summary()
``` 

*MBOA ~ BOA*

```{r, message=FALSE, echo=FALSE, error=FALSE, warning = FALSE, fig.dim = c(5, 5)}
fIG4.TIcor_BOA_MBOA <- AUC_tolind_MBOA_AMPO_BOA_APO_DMG_cor %>% 
  filter(TI_norm_MBOA < 1.5) %>%
  filter(TI_norm_BOA < 1.5) %>%
  ggplot(aes(x = TI_norm_MBOA, y = TI_norm_BOA)) +
  geom_point(aes(colour = Family), alpha = 0.5, show.legend = FALSE) +
  geom_smooth(method = "lm", color="black", formula = y ~ x)  + 
  xlim(0, 1.25)+
  ylim(0, 1.25)+
  stat_cor(method = "pearson", label.x = 0.2, label.y = 1.25, cex = 15/.pt) +
  scale_colour_manual(values = results_AUC_all_level_cols_Family) +
  theme_bw() +
  labs(x = "TI MBOA",
       y = "TI BOA")

fIG4.TIcor_BOA_MBOA

# ggsave(plot = fIG4.TIcor_BOA_MBOA,  filename = "fIG4.TIcor_BOA_MBOA.pdf", width = 9, height = 9, dpi = 300, scale = 1, units = "cm")
# 
# ggsave(plot = fIG4.TIcor_BOA_MBOA,  filename = "fIG4.TIcor_BOA_MBOA.svg", width = 9, height = 9, dpi = 300, scale = 1, units = "cm")
```

```{r, include=FALSE}
aov(TI_norm_MBOA ~ TI_norm_BOA, data = AUC_tolind_MBOA_AMPO_BOA_APO_DMG_cor %>% filter(TI_norm_BOA < 1.2) %>%
  filter(TI_norm_MBOA < 1.2)) %>% summary()
``` 

*APO ~ AMPO*

```{r, message=FALSE, echo=FALSE, error=FALSE, warning = FALSE,  fig.dim = c(5, 5)}
fIG4.TIcor_APO_AMPO <- AUC_tolind_MBOA_AMPO_BOA_APO_DMG_cor %>% 
  filter(TI_norm_AMPO < 1.2) %>%
  filter(TI_norm_APO < 1.2) %>%
  ggplot(aes(x = TI_norm_AMPO, y = TI_norm_APO)) +
  geom_point(aes(colour = Family), alpha = 0.5, show.legend = FALSE) +
  geom_smooth(method = "lm", color="black", formula = y ~ x)  + 
  xlim(0, 1.5)+
  ylim(0, 1.5)+
  stat_cor(method = "pearson", label.x = 0.2, label.y = 1.5, cex = 15/.pt) +
  scale_colour_manual(values = results_AUC_all_level_cols_Family) +
  theme_bw() +
  labs(x = "TI AMPO",
       y = "TI APO")

fIG4.TIcor_APO_AMPO

# ggsave(plot = fIG4.TIcor_APO_AMPO,  filename = "fIG4.TIcor_APO_AMPO.pdf", width = 9, height = 9, dpi = 300, scale = 1, units = "cm")
# 
# ggsave(plot = fIG4.TIcor_APO_AMPO,  filename = "fIG4.TIcor_APO_AMPO.svg", width = 9, height = 9, dpi = 300, scale = 1, units = "cm")
```

```{r, include=FALSE}
aov(TI_norm_APO ~ TI_norm_AMPO, data = AUC_tolind_MBOA_AMPO_BOA_APO_DMG_cor %>% filter(TI_norm_AMPO < 1.2) %>%
  filter(TI_norm_APO < 1.2)) %>% summary()
```

*BOA ~ APO*

```{r, message=FALSE, echo=FALSE, error=FALSE, warning = FALSE, fig.dim = c(5, 5)}
fIG4.TIcor_BOA_APO <- AUC_tolind_MBOA_AMPO_BOA_APO_DMG_cor %>% 
  filter(TI_norm_BOA < 1.2) %>%
  filter(TI_norm_APO < 1.2) %>%
  ggplot(aes(x = TI_norm_BOA, y = TI_norm_APO)) +
  geom_point(aes(colour = Family), alpha = 0.5, show.legend = FALSE) +
  geom_smooth(method = "lm", color="black", formula = y ~ x)  + 
  xlim(0, 1.25)+
  ylim(0, 1.25)+
  stat_cor(method = "pearson", label.x = 0.2, label.y = 1.2, cex = 15/.pt) +
  scale_colour_manual(values = results_AUC_all_level_cols_Family) +
  scale_colour_manual(values = results_AUC_all_level_cols_Family) +
  theme_bw() +
  labs(x = "TI BOA",
       y = "TI APO")

fIG4.TIcor_BOA_APO

# ggsave(plot = fIG4.TIcor_BOA_APO,  filename = "fIG4.TIcor_BOA_APO.pdf", width = 9, height = 9, dpi = 300, scale = 1, units = "cm")
# 
# ggsave(plot = fIG4.TIcor_BOA_APO,  filename = "fIG4.TIcor_BOA_APO.svg", width = 9, height = 9, dpi = 300, scale = 1, units = "cm")

```

```{r, include=FALSE}
aov(TI_norm_BOA ~ TI_norm_APO, data = AUC_tolind_MBOA_AMPO_BOA_APO_DMG_cor %>% filter(TI_norm_BOA < 1.2) %>%
  filter(TI_norm_APO < 1.2)) %>% summary()
```

*MBOA ~ DIMBOA-Glc*

```{r, message=FALSE, echo=FALSE, error=FALSE, warning = FALSE, fig.dim = c(5, 5)}
TIcor_MBOA_DMG <- AUC_tolind_MBOA_AMPO_BOA_APO_DMG_cor %>% 
  # filter(AUC_norm_DMG < 1.2) %>%
  filter(TI_norm_DMG < 1.2) %>%
  filter(TI_norm_MBOA < 1.2) %>%
  ggplot(aes(x = TI_norm_MBOA, y = TI_norm_DMG)) +
  geom_point(aes(colour = Family), alpha = 0.5, show.legend = FALSE) +
  geom_smooth(method = "lm", color="black", formula = y ~ x)  + 
  stat_cor(method = "pearson", label.x = 0.5, label.y = 1.3) +
  scale_colour_manual(values = results_AUC_all_level_cols_Family) +
  theme_bw() +
  labs(x = "TI_norm MBOA",
       y = "TI_norm DMG")

TIcor_MBOA_DMG

# ggsave(plot = TIcor_MBOA_DMG,  filename = "TIcor_MBOA_DMG.pdf", width = 9, height = 9, dpi = 300, scale = 1, units = "cm")
# 
# ggsave(plot = TIcor_MBOA_DMG,  filename = "fIG4.TIcor_MBOA_DMG.svg", width = 9, height = 9, dpi = 300, scale = 1, units = "cm")
```

```{r, include=FALSE}
aov(TI_norm_MBOA ~ TI_norm_DMG, data = AUC_tolind_MBOA_AMPO_BOA_APO_DMG_cor %>% filter(TI_norm_AMPO < 2) %>%
  filter(TI_norm_MBOA < 2)) %>% summary()
``` 

## Tolerance groups
```{r, include=FALSE}
#AUC_tolind_MBOA_AMPO_BOA_APO_DMG_cor <- read_rds("Output/AUC_tolind_MBOA_AMPO_BOA_APO_DMG_cor.rds")
```

```{r, include=FALSE}
BX_tolerance <- AUC_tolind_MBOA_AMPO_BOA_APO_DMG_cor %>% as.data.frame() %>% 
  filter(TI_norm_MBOA < 1.2 & TI_norm_AMPO < 1.2 & TI_norm_BOA < 1.2 & TI_norm_APO < 1.2) %>% 
  dplyr::group_by(Strain, Family) %>% 
  dplyr::summarize(TI_norm_MBOA_mean = mean(TI_norm_MBOA), 
            TI_norm_AMPO_mean = mean(TI_norm_AMPO), 
            TI_norm_BOA_mean = mean(TI_norm_BOA), 
            TI_norm_APO_mean = mean(TI_norm_APO)) %>% 
  dplyr::select(Strain, Family, TI_norm_MBOA_mean, TI_norm_AMPO_mean, TI_norm_BOA_mean, TI_norm_APO_mean)

BX_tolerance %<>% 
  mutate(MBOA_type = case_when(TI_norm_MBOA_mean > 0.75 ~ "tolerant",
                              TI_norm_MBOA_mean < 0.5 ~ "susceptible")) %>% 
  mutate(AMPO_type = case_when(TI_norm_AMPO_mean > 0.75 ~ "tolerant",
                              TI_norm_AMPO_mean < 0.5 ~ "susceptible")) %>% 
  mutate(BOA_type = case_when(TI_norm_BOA_mean > 0.75 ~ "tolerant",
                              TI_norm_BOA_mean < 0.5 ~ "susceptible")) %>% 
  mutate(APO_type = case_when(TI_norm_APO_mean > 0.75 ~ "tolerant",
                              TI_norm_APO_mean < 0.5 ~ "susceptible")) %>% 
  # mutate(DMG_type = case_when(AUC_norm_DMG_mean > 0.75 ~ "tolerant",
  #                             AUC_norm_DMG_mean < 0.5 ~ "susceptible")) %>% 
  replace_na(list(MBOA_type = "intermediate", 
                  AMPO_type = "intermediate", 
                  BOA_type = "intermediate", 
                  APO_type = "intermediate"))
```

```{r, include=FALSE}
BX_tolerance %<>% as.data.frame()

BX_tolerance_table <- cbind(BX_tolerance %>% dplyr::select(MBOA_type) %>% table() %>% as.data.frame() %>% dplyr::rename(MBOA = Freq),
      BX_tolerance %>% as.data.frame() %>% dplyr::select(BOA_type) %>% table() %>% as.data.frame() %>% dplyr::rename(BOA = Freq),
      BX_tolerance %>% as.data.frame() %>% dplyr::select(AMPO_type) %>% table() %>% as.data.frame() %>% dplyr::rename(AMPO = Freq),
      BX_tolerance %>% as.data.frame() %>% dplyr::select(APO_type) %>% table() %>% as.data.frame() %>% dplyr::rename(APO = Freq)) # %>%
     # dplyr::select(-3,-5,-7) %>% dplyr::rename(group = '.')

write.csv(BX_tolerance_table, "Output/BX_tolerance_table.csv")
```

```{r, echo=FALSE}
BX_tolerance_table %>% knitr::kable()
```

```{r, include=FALSE}
MBOA_type_table <- BX_tolerance %>% as.data.frame() %>% dplyr::select(MBOA_type, Family) %>% table()
```

```{r, include=FALSE}
MBOA_type_test <- fisher.test(BX_tolerance$Family, BX_tolerance$MBOA_type,
            alternative = "two.sided",
            conf.int = TRUE, conf.level = 0.95)

BOA_type_test <- fisher.test(BX_tolerance$Family, BX_tolerance$BOA_type,
            alternative = "two.sided",
            conf.int = TRUE, conf.level = 0.95)

AMPO_type_test <- fisher.test(BX_tolerance$Family, BX_tolerance$AMPO_type,  
            alternative = "two.sided",
            conf.int = TRUE, conf.level = 0.95)

APO_type_test <- fisher.test(BX_tolerance$Family, BX_tolerance$APO_type,
            alternative = "two.sided",
            conf.int = TRUE, conf.level = 0.95)

p.value <- rbind(MBOA_type_test$p.value, BOA_type_test$p.value, AMPO_type_test$p.value, APO_type_test$p.value)
Compound <- rbind("MBOA", "BOA", "AMPO", "APO")
Compound_p.value <- cbind(Compound, p.value)
Compound_p.value <- Compound_p.value %>% as.data.frame()
colnames(Compound_p.value) <- c("Compound", "p.value")
```


```{r, echo=FALSE}
Compound_p.value %>% knitr::kable()
```

## Effect of gram on tolerance 
```{r,message=FALSE, echo=FALSE}
database <- read_excel("Input/__Database_MRB_isolates_sequences.xlsx")
```

```{r, include=FALSE}
#AUC_tolind_MBOA_AMPO_BOA_APO_DMG_cor <- read_rds("Output/AUC_tolind_MBOA_AMPO_BOA_APO_DMG_cor.rds")
```

```{r,message=FALSE, echo=FALSE}
AUC_tolind_MBOA_AMPO_BOA_APO_DMG_cor_gram <- 
  left_join(AUC_tolind_MBOA_AMPO_BOA_APO_DMG_cor, database %>% dplyr::select(Strain, gram), by = "Strain") 
```

```{r, include=FALSE}
AUC_tolind_MBOA_AMPO_BOA_APO_DMG_cor_gram$cols_Family <- as.character(AUC_tolind_MBOA_AMPO_BOA_APO_DMG_cor_gram$Family)

AUC_tolind_MBOA_AMPO_BOA_APO_DMG_cor_gram[ AUC_tolind_MBOA_AMPO_BOA_APO_DMG_cor_gram$Family=="Moraxellaceae", ]$cols_Family <- "#a73e62"
AUC_tolind_MBOA_AMPO_BOA_APO_DMG_cor_gram[ AUC_tolind_MBOA_AMPO_BOA_APO_DMG_cor_gram$Family=="Bacillaceae" , ]$cols_Family <- "#004586"
AUC_tolind_MBOA_AMPO_BOA_APO_DMG_cor_gram[ AUC_tolind_MBOA_AMPO_BOA_APO_DMG_cor_gram$Family=="Pseudomonadaceae" , ]$cols_Family <- "#4b1f6f"
AUC_tolind_MBOA_AMPO_BOA_APO_DMG_cor_gram[ AUC_tolind_MBOA_AMPO_BOA_APO_DMG_cor_gram$Family=="Paenibacillaceae" , ]$cols_Family <- "#8f9ec9"
AUC_tolind_MBOA_AMPO_BOA_APO_DMG_cor_gram[ AUC_tolind_MBOA_AMPO_BOA_APO_DMG_cor_gram$Family=="Rhizobiaceae", ]$cols_Family <- "#ff950e"
AUC_tolind_MBOA_AMPO_BOA_APO_DMG_cor_gram[ AUC_tolind_MBOA_AMPO_BOA_APO_DMG_cor_gram$Family=="Sphingomonadaceae", ]$cols_Family <- "#c5000b"
AUC_tolind_MBOA_AMPO_BOA_APO_DMG_cor_gram[ AUC_tolind_MBOA_AMPO_BOA_APO_DMG_cor_gram$Family=="Xanthomonadaceae", ]$cols_Family <- "#0084d1"
AUC_tolind_MBOA_AMPO_BOA_APO_DMG_cor_gram[ AUC_tolind_MBOA_AMPO_BOA_APO_DMG_cor_gram$Family=="Microbacteriaceae" , ]$cols_Family <- "#7e0021"
AUC_tolind_MBOA_AMPO_BOA_APO_DMG_cor_gram[ AUC_tolind_MBOA_AMPO_BOA_APO_DMG_cor_gram$Family=="Micrococcaceae" , ]$cols_Family <- "#83caff"
AUC_tolind_MBOA_AMPO_BOA_APO_DMG_cor_gram[ AUC_tolind_MBOA_AMPO_BOA_APO_DMG_cor_gram$Family=="Chitinophagaceae" , ]$cols_Family <- "#ff420e"
AUC_tolind_MBOA_AMPO_BOA_APO_DMG_cor_gram[ AUC_tolind_MBOA_AMPO_BOA_APO_DMG_cor_gram$Family=="Enterobacteriaceae" , ]$cols_Family <- "#ffd320"
AUC_tolind_MBOA_AMPO_BOA_APO_DMG_cor_gram[ AUC_tolind_MBOA_AMPO_BOA_APO_DMG_cor_gram$Family=="Erwiniaceae" , ]$cols_Family <- "#579d1c"
AUC_tolind_MBOA_AMPO_BOA_APO_DMG_cor_gram[ AUC_tolind_MBOA_AMPO_BOA_APO_DMG_cor_gram$Family=="Nocardioidaceae" , ]$cols_Family <- "#314004"
AUC_tolind_MBOA_AMPO_BOA_APO_DMG_cor_gram[ AUC_tolind_MBOA_AMPO_BOA_APO_DMG_cor_gram$Family=="Oxalobacteraceae" , ]$cols_Family <- "#aecf00"

## collapsed color vector for each level
temp <- data.frame(AUC_tolind_MBOA_AMPO_BOA_APO_DMG_cor_gram$Family, AUC_tolind_MBOA_AMPO_BOA_APO_DMG_cor_gram$cols_Family)
temp <- plyr::ddply(temp, .variables="AUC_tolind_MBOA_AMPO_BOA_APO_DMG_cor_gram.cols_Family", .fun=unique)   #library(plyr)
AUC_tolind_MBOA_AMPO_BOA_APO_DMG_cor_gram_level_cols_Family <- as.character(temp[,2])
names(AUC_tolind_MBOA_AMPO_BOA_APO_DMG_cor_gram_level_cols_Family) <- temp[,1]
```

```{r, message=FALSE, echo=FALSE, error=FALSE, warning = FALSE, fig.dim = c(5, 5)}
Fig.supp.MBOAgram <- AUC_tolind_MBOA_AMPO_BOA_APO_DMG_cor_gram %>% 
  filter(!gram %in% "NA") %>% 
  filter(TI_norm_MBOA < 1.2) %>% 
  ggplot(aes(x= gram, y = TI_norm_MBOA)) +
  geom_boxplot(aes(fill = gram), show.legend = FALSE) +
  geom_jitter(aes(colour = Family), alpha = 0.5, width = 0.2, show.legend = FALSE) +
  ylim(0, 1.4) +
  stat_compare_means(label = "p.signif", method = "t.test", ref.group = "negative", label.y = 1.3, cex = 15/.pt) +
  scale_fill_manual(values = c("grey50", "grey90"))+
  scale_color_manual(values = AUC_tolind_MBOA_AMPO_BOA_APO_DMG_cor_gram_level_cols_Family)+
  theme_bw() +
  theme(axis.text.x=element_text(size = 35/.pt),
        axis.text.y=element_text(size = 60/.pt),
        axis.title = element_text(size = 60/.pt)) +
  labs(x = " ", y = "TI", title = "MBOA")

Fig.supp.MBOAgram

# ggsave(plot = Fig.supp.MBOAgram,  filename = "Fig.supp.MBOAgram.pdf", width = 6.6, height = 12, dpi = 300, scale = 1, units = "cm")

# ggsave(plot = Fig.supp.MBOAgram,  filename = "Fig.supp.MBOAgram.svg", width = 6.6, height = 12, dpi = 300, scale = 1, units = "cm")
```

```{r, message=FALSE, echo=FALSE, error=FALSE, warning = FALSE, fig.dim = c(5, 5)}
Fig.supp.BOAgram <- AUC_tolind_MBOA_AMPO_BOA_APO_DMG_cor_gram %>% 
  filter(!gram %in% "NA") %>%
  filter(TI_norm_BOA < 1.2) %>% 
  ggplot(aes(x= gram, y = TI_norm_BOA)) +
  geom_boxplot(aes(fill = gram), show.legend = FALSE) +
  geom_jitter(aes(colour = Family), alpha = 0.5, width = 0.2, show.legend = FALSE) +
  ylim(0, 1.4) +
  stat_compare_means(label = "p.signif", method = "t.test", ref.group = "negative", label.y = 1.3, cex = 15/.pt) +
  scale_fill_manual(values = c("grey50", "grey90"))+
  scale_color_manual(values = AUC_tolind_MBOA_AMPO_BOA_APO_DMG_cor_gram_level_cols_Family)+
  theme_bw() +
  theme(axis.text.x=element_text(size = 35/.pt),
        axis.text.y=element_text(size = 60/.pt),
        axis.title = element_text(size = 60/.pt)) +
  labs(x = " ", y = "TI", title = "BOA")

Fig.supp.BOAgram

# ggsave(plot = Fig.supp.BOAgram,  filename = "Fig.supp.BOAgram.pdf", width = 6.6, height = 12, dpi = 300, scale = 1, units = "cm")

# ggsave(plot = Fig.supp.BOAgram,  filename = "Fig.supp.BOAgram.svg", width = 6.6, height = 12, dpi = 300, scale = 1, units = "cm")
```

```{r, message=FALSE, echo=FALSE, error=FALSE, warning = FALSE,  fig.dim = c(5, 5)}
Fig.supp.AMPOgram <- AUC_tolind_MBOA_AMPO_BOA_APO_DMG_cor_gram %>% 
  filter(!gram %in% "NA") %>%
  filter(TI_norm_AMPO < 1.2) %>% 
  ggplot(aes(x= gram, y = TI_norm_AMPO)) +
  geom_boxplot(aes(fill = gram), show.legend = FALSE) +
  geom_jitter(aes(colour = Family), alpha = 0.5, width = 0.2, show.legend = FALSE) +
  ylim(0, 1.4) +
  stat_compare_means(label = "p.signif", method = "t.test", ref.group = "negative", label.y = 1.3, cex = 15/.pt) +
  scale_fill_manual(values = c("grey50", "grey90"))+
  scale_color_manual(values = AUC_tolind_MBOA_AMPO_BOA_APO_DMG_cor_gram_level_cols_Family)+
  theme_bw() +
  theme(axis.text.x=element_text(size = 35/.pt),
        axis.text.y=element_text(size = 60/.pt),
        axis.title = element_text(size = 60/.pt)) +
  labs(x = " ", y = "TI", title = "AMPO")

Fig.supp.AMPOgram

# ggsave(plot = Fig.supp.AMPOgram,  filename = "Fig.supp.AMPOgram.pdf", width = 6.6, height = 12, dpi = 300, scale = 1, units = "cm")

# ggsave(plot = Fig.supp.AMPOgram,  filename = "Fig.supp.AMPOgram.svg", width = 6.6, height = 12, dpi = 300, scale = 1, units = "cm")
```

```{r, message=FALSE, echo=FALSE, error=FALSE, warning = FALSE, fig.dim = c(5, 5)}
Fig.supp.APOgram <- AUC_tolind_MBOA_AMPO_BOA_APO_DMG_cor_gram %>% 
  filter(!gram %in% "NA") %>%
  filter(TI_norm_APO < 1.2) %>% 
  ggplot(aes(x= gram, y = TI_norm_APO)) +
  geom_boxplot(aes(fill = gram), show.legend = FALSE) +
  geom_jitter(aes(colour = Family), alpha = 0.5, width = 0.2, show.legend = FALSE) +
  ylim(0, 1.4) +
  stat_compare_means(label = "p.signif", method = "t.test", ref.group = "negative", label.y = 1.3, cex = 15/.pt) +
  scale_fill_manual(values = c("grey50", "grey90"))+
  scale_color_manual(values = AUC_tolind_MBOA_AMPO_BOA_APO_DMG_cor_gram_level_cols_Family)+
  theme_bw() +
theme(axis.text.x=element_text(size = 35/.pt),
        axis.text.y=element_text(size = 60/.pt),
        axis.title = element_text(size = 60/.pt)) +
  labs(x = " ", y = "TI", title = "APO")

Fig.supp.APOgram

# ggsave(plot = Fig.supp.APOgram,  filename = "Fig.supp.APOgram.pdf",width = 6.6, height = 12, dpi = 300, scale = 1, units = "cm")

# ggsave(plot = Fig.supp.APOgram,  filename = "Fig.supp.APOgram.svg", width = 6.6, height = 12, dpi = 300, scale = 1, units = "cm")
```

```{r, message=FALSE, echo=FALSE, error=FALSE, warning = FALSE, fig.dim = c(5, 5)}
Fig.supp.DMGgram <- AUC_tolind_MBOA_AMPO_BOA_APO_DMG_cor_gram %>% 
  filter(!gram %in% "NA") %>%
  # filter(AUC_norm_DMG < 1.2) %>% 
  filter(TI_norm_DMG < 1.2) %>% 
  # ggplot(aes(x= gram, y = AUC_norm_DMG)) +
  ggplot(aes(x= gram, y = TI_norm_DMG)) +
  geom_boxplot(aes(fill = gram), show.legend = FALSE) +
  geom_jitter(aes(colour = Family), alpha = 0.5, width = 0.2, show.legend = FALSE) +
  ylim(0, 1.4) +
  stat_compare_means(label = "p.signif", method = "t.test", ref.group = "negative", label.y = 1.3, cex = 15/.pt) +
  scale_fill_manual(values = c("grey50", "grey90"))+
  scale_color_manual(values = AUC_tolind_MBOA_AMPO_BOA_APO_DMG_cor_gram_level_cols_Family)+
  theme_bw() +
  theme(axis.text.x=element_text(size = 35/.pt),
        axis.text.y=element_text(size = 60/.pt),
        axis.title = element_text(size = 60/.pt)) +
  labs(x = " ", y = expression("TI"), title = "DIMBOA-Glc")

Fig.supp.DMGgram

# ggsave(plot = Fig.supp.DMGgram,  filename = "Fig.supp.DMGgram.pdf", width = 6.6, height = 12, dpi = 300, scale = 1, units = "cm")
# 
# ggsave(plot = Fig.supp.DMGgram,  filename = "Fig.supp.DMGgram.svg", width = 6.6, height = 12, dpi = 300, scale = 1, units = "cm")
```

```{r, include=FALSE}
MBOA_gram_aov <- aov(TI_norm_MBOA ~ gram, data = AUC_tolind_MBOA_AMPO_BOA_APO_DMG_cor_gram %>% filter(TI_norm_MBOA < 1.2)) %>% summary()

aov(TI_norm_BOA ~ gram, data = AUC_tolind_MBOA_AMPO_BOA_APO_DMG_cor_gram %>% filter(TI_norm_BOA < 1.2)) %>% summary()

aov(TI_norm_AMPO ~ gram, data = AUC_tolind_MBOA_AMPO_BOA_APO_DMG_cor_gram %>% filter(TI_norm_AMPO < 1.2)) %>% summary()

aov(TI_norm_APO ~ gram, data = AUC_tolind_MBOA_AMPO_BOA_APO_DMG_cor_gram %>% filter(TI_norm_APO < 1.2)) %>% summary()

# aov(AUC_norm_DMG ~ gram, data = AUC_tolind_MBOA_AMPO_BOA_APO_DMG_cor_gram %>% filter(AUC_norm_DMG < 1.2)) %>% summary()

aov(TI_norm_DMG ~ gram, data = AUC_tolind_MBOA_AMPO_BOA_APO_DMG_cor_gram %>% filter(TI_norm_DMG < 1.2)) %>% summary()
```

# II) ATSPHERE BACTERIA

**Load metadata**: Metadata strains with plate layout (indicating which strain is in which well) and database with taxonomic information about the strains. 

```{r, include=FALSE}
Strain_Data_raw <- read_excel("Input/Metadata_60_strains_AtSphere_screen.xlsx", sheet = "metadata_Run.1")  %>% as.data.frame()
Strain_Data_raw_2 <- read_excel("Input/Metadata_60_strains_AtSphere_screen.xlsx", sheet = "metadata_Run2") %>% as.data.frame()

database <- read_excel("Input/220122_ExpDesign_MRB_At_BX_CM_tolerance_testing.xlsx", sheet = 10) %>% mutate(Strain = gsub(" ", "", Strain)) %>% as.data.frame()
Strain_Data <- left_join(Strain_Data_raw, database, by = "Strain") %>% 
  dplyr::select(Strain, Column, Row, Well, Replicate, Plate, Phylum, Class, Order, Family, Genus) %>% 
  mutate(Well_Plate = interaction(Well, Plate, sep = "_"))

Strain_Data_2 <- left_join(Strain_Data_raw_2, database, by = "Strain") %>% 
  dplyr::select(Strain, Column, Row, Well, Replicate, Plate, Phylum, Class, Order, Family, Genus) %>% 
  mutate(Well_Plate = interaction(Well, Plate, sep = "_"))
```

**Load growth data**: Growth data from the plates. 

```{r, warning=FALSE, include=FALSE}
results1 <- data_frame()

# Provide a lookup table for plate information, treatments are spread accross sheets in the table to be read.
# in the first column describe the plates and in the second column the experiment (rep(times = # plates - 2))
treatlist <- data_frame("Treat" =
              c("Dummy",
                "0_uM_DMSO_TSB_A",
                "250_uM_MBOA_TSB_A",
                "500_uM_MBOA_TSB_A",
                "625_uM_MBOA_TSB_A",
                "1250_uM_MBOA_TSB_A",
                "2500_uM_MBOA_TSB_A",
                "5000_uM_MBOA_TSB_A",
                "media",
                "0_uM_DMSO_TSB_B",
                "250_uM_MBOA_TSB_B",
                "500_uM_MBOA_TSB_B",
                "625_uM_MBOA_TSB_B",
                "1250_uM_MBOA_TSB_B",
                "2500_uM_MBOA_TSB_B",
                "5000_uM_MBOA_TSB_B",
                "Dummy"),
  "Run" = c("NULL", rep("R1", times = 15), "NULL"), # times = all plates but without the dummy plates
  "Plate" = c("NULL", rep("Plate_A", times = 7), "NULL", rep("Plate_B", times = 7), "NULL"), # times = all plates but without the dummy plates
  "Media" = c("NULL", rep ("TSB", times = 15), "NULL"),
  "Chem" = c("NULL", "DMSO", rep("MBOA", times = 6), "NULL", "DMSO", rep("MBOA", times = 6), "NULL"),
  "Conc" = c("NULL", c("0", "250", "500", "625", "1250", "2500", "5000"), "NULL", c("0", "250", "500", "625", "1250", "2500", "5000"), "NULL"))




#Initialize / clear df

#Loop through sheets.

for (i in 1:nrow(treatlist)) {
  tmp <- read_excel("Input/220613_MBOA_AtSphere_run1.xlsx", sheet = i, range = "B26:CU69") %>%   # change for duration runs
    dplyr::select(-`T° 600`) %>% # Drop Temperature
    #pivot_longer(names_to = "Well", values_to = "Density") %>%  #Make long
    gather(key = "Well", value = "Density", 2:ncol(.)) %>%
    mutate(Treat = as.character(treatlist[i,1]), #Add treatment
    Run = as.character(treatlist[i,2]),
    Plate = as.character(treatlist[i,3]),
    Media = as.character(treatlist[i,4]),
    Chem = as.character(treatlist[i,5]),
    Conc = as.character(treatlist[i,6]),
    Well_Plate = interaction(Well, Plate, sep = "_")) #Add Replicate
results1 <- bind_rows(results1, tmp)
}
```

```{r, include=FALSE}
results2 <- data_frame()

# Provide a lookup table for plate information, treatments are spread accross sheets in the table to be read.
# in the first column describe the plates and in the second column the experiment (rep(times = # plates - 2))
treatlist <- data_frame("Treat" =
              c("Dummy",
                "0_uM_DMSO_TSB_A",
                "250_uM_MBOA_TSB_A",
                "500_uM_MBOA_TSB_A",
                "625_uM_MBOA_TSB_A",
                "1250_uM_MBOA_TSB_A",
                "2500_uM_MBOA_TSB_A",
                "5000_uM_MBOA_TSB_A",
                "media",
                "0_uM_DMSO_TSB_B",
                "250_uM_MBOA_TSB_B",
                "500_uM_MBOA_TSB_B",
                "625_uM_MBOA_TSB_B",
                "1250_uM_MBOA_TSB_B",
                "2500_uM_MBOA_TSB_B",
                "5000_uM_MBOA_TSB_B",
                "Dummy"),
  "Run" = c("NULL", rep("R2", times = 15), "NULL"), # times = all plates but without the dummy plates
  "Plate" = c("NULL", rep("Plate_A", times = 7), "NULL", rep("Plate_B", times = 7), "NULL"), # times = all plates but without the dummy plates
  "Media" = c("NULL", rep ("TSB", times = 15), "NULL"),
  "Chem" = c("NULL", "DMSO", rep("MBOA", times = 6), "NULL", "DMSO", rep("MBOA", times = 6), "NULL"),
  "Conc" = c("NULL", c("0", "250", "500", "625", "1250", "2500", "5000"), "NULL", c("0", "250", "500", "625", "1250", "2500", "5000"), "NULL"))




#Initialize / clear df

#Loop through sheets.

for (i in 1:nrow(treatlist)) {
  tmp <- read_excel("Input/220620_MBOA_AtSpehre_Run2.xlsx", sheet = i, range = "B26:CU69") %>%   # change for duration runs
    dplyr::select(-`T° 600`) %>% # Drop Temperature
    #pivot_longer(names_to = "Well", values_to = "Density") %>%  #Make long
    gather(key = "Well", value = "Density", 2:ncol(.)) %>%
    mutate(Treat = as.character(treatlist[i,1]), #Add treatment
    Run = as.character(treatlist[i,2]),
    Plate = as.character(treatlist[i,3]),
    Media = as.character(treatlist[i,4]),
    Chem = as.character(treatlist[i,5]),
    Conc = as.character(treatlist[i,6]),
    Well_Plate = interaction(Well, Plate, sep = "_")) #Add Replicate
results2 <- bind_rows(results2, tmp)
}
```

```{r, include=FALSE, warning=FALSE}
###
###Get times based on difftime. 
results1 %<>%
  filter(!Treat %in% "Dummy") %>%
  filter(!Density %in% "NA") %>% 
  mutate(mins = difftime(Time, Time[1], units = "mins") %>% as.numeric(),
         hrs = difftime(Time, Time[1], units = "hours") %>% as.numeric()) %>% 
  left_join(., Strain_Data, by = "Well_Plate")
```

```{r, include=FALSE, warning=FALSE}
###
###Get times based on difftime. 
results2 %<>%
  filter(!Treat %in% "Dummy") %>%
  filter(!Density %in% "NA") %>% 
  mutate(mins = difftime(Time, Time[1], units = "mins") %>% as.numeric(),
         hrs = difftime(Time, Time[1], units = "hours") %>% as.numeric()) %>% 
  left_join(., Strain_Data_2, by = "Well_Plate") 
```

```{r, include=FALSE, warning=FALSE}
results <- rbind(results1 %>% dplyr::select(-Well.x, -Well.y, -Plate.x) %>% dplyr::rename(Plate = Plate.y), 
                 results2 %>% dplyr::select(-Well.x, -Well.y, -Plate.x) %>% dplyr::rename(Plate = Plate.y)) 
                 # results3 %>% dplyr::select(-Well.x, -Well.y, -Plate.x) %>% dplyr::rename(Plate = Plate.y))

results %<>% mutate(Conc = as.character(Conc)) %>% dplyr::select(-Time)
```

Calculate density increase: substract initial density from density of each time point 
```{r, include=FALSE, warning=FALSE}
results %<>%
  filter(!Strain %in% c("Ignore","None")) %>%
  group_by(Run, Well_Plate, Treat) %>%
  mutate(Density_increase = Density - Density[1]) %>%
  ungroup %>% as.data.frame()
```

```{r, include=FALSE, warning=FALSE}
results_raw <- results

#write_rds(results, paste0("Output/results_AtSphere_MBOA_screen.rds"))
```

## Growth curves
First the density increase in the medium control plate is plotted to check that there was no general contamination detected in the experiment.

```{r, warning=FALSE, message=FALSE, echo=FALSE, error=FALSE, fig.dim = c(6, 5)}
results %>%  filter(Treat %in% c("media")) %>%
  separate(Well_Plate, c("Well", "Plate")) %>%
  mutate(Row = Well %>% str_extract(.,"[A-H]"),
                    Column = Well %>% str_extract(.,"[0-9]+") %>% as.numeric()) %>% 
  ggplot(aes(x=hrs, y= Density_increase)) +
  geom_line(aes(colour = Run), linewidth = 1) +
  facet_grid(Row ~ Column, scales = "fixed") +
  theme_bw() +
  labs(x = "Time [hours]",
       y = "OD 600", 
       title = "growth curves in media control plate",
       color = "Run")
```

Next the growth of all strains in the control treatment and the no bacteria control is checked. 

```{r, echo=FALSE, error=FALSE, warning = FALSE, fig.dim = c(8, 8)}
results %>%  
  filter(Chem %in% "DMSO") %>% 
  unique() %>% 
  ggplot(aes(x=hrs, y= Density_increase, color = interaction(Chem, Conc))) +
  geom_line(aes(colour = Run, linetype = as.factor(Column))) +
  facet_wrap(~ Strain, scales = "fixed") +
  theme_bw() +
  labs(x = "Time [hours]",
       y = "OD 600", 
       title = "growth curves",
       color = "Run")
```

Growth of individual strains in different concentrations of MBOA. 

```{r, echo=FALSE, error=FALSE, warning = FALSE, fig.dim = c(8, 8)}
results %>%  
  filter(Conc %in% c("5000", "2500", "1250", "625", "500", "250", "0")) %>% 
  mutate(Conc = factor(Conc, levels = c("5000", "2500", "1250", "625", "500", "250", "0"))) %>% 
  filter(Chem %in% c("DMSO", "MBOA")) %>% 
  ggplot(aes(x=hrs, y= Density_increase, color = Conc)) +
  geom_line(aes(linetype = interaction(as.factor(Replicate), Run))) +
  facet_wrap(~interaction(Strain, Family), scales = "fixed") +
  theme_bw() +
  scale_color_manual(values = c("#9ecae1", "#6baed6", "#4292c6", "#2171b5", "#08519c", "#08306b", "#000740"))+
  labs(x = "Time [hours]",
       y = "OD 600", 
       title = "MBOA response",
       color = "Concentration",
       linetype = "Replicate")
```

Some strains did not grew and were therefore excluded from the analysis. In both runs: Root1294 Sphigomonas, Root420 Flavobacterium, Root482 Rhizobium, Root559 Lysobacter, Root630 Pseudoxanthomonas, in run 1: Root318D1 Variovorax, in run 2: Root166 Microbacterium.

```{r, include=FALSE}
results %<>% filter(!Strain %in% c("Root1294", "Root420", "Root482", "Root559", "Root630", "Root318D1", "Root166"))
```

## Calculate area under the curve
To quantify the total bacterial growth over time, we calculate the total area under the curve. This is done with the function "auc()" For comparison between treatments, the total AUC (AUC_raw) is normalized with the AUC of the strain grown in the control treatment (no chemicals added, just normal growth media with DMSO). The AUC_norm is used for all further analysis, plotting and calculations. 

```{r, warning=FALSE, include=FALSE}
results %<>% dplyr::select(-Well_Plate) %>% unique()

results_AUC_MBOA_AtSphere <- results %>% 
  filter(Media %in% "TSB") %>% 
  filter(Chem %in% c("DMSO", "MBOA")) %>% 
  mutate(Conc = as.numeric(Conc)) %>% 
  arrange((Conc))  %>%
  # mutate(Conc = factor(Conc, levels = c("0", "250", "500", "625", "1250", "2500", "5000"))) %>% 
  group_by(Strain, Replicate, Media, Treat, Conc, Plate, Run) %>%
  dplyr::summarize(AUC = auc(hrs, Density_increase), 
            AUC_raw = auc(hrs, Density)) %>%
  ungroup %>% 
  mutate(Treat = as.factor(Treat),
         Replicate = as.factor(Replicate)) %>% 
  mutate(AUC_pos = abs(AUC)) %>% 
  mutate(Strain_Replicate_Media_Run = paste(Strain, Replicate, Media, Run)) %>% 
  arrange(desc(Conc))  %>% # this a super important step to normalize with the growth in Conc = 0 (Conc has to be a numeric, see above)
  #Split by Rep
  base::split(., .$Strain_Replicate_Media_Run) %>% #Make each strain / rep combination its own data.frane
  map_dfr(~ 
  mutate(.x, AUC_norm = AUC_pos / AUC_pos[length(AUC_pos)])) %>% #in each df, divide the AUC by the last AUC value; since Control does not start with a number, it is the last treatment
  left_join(. , Strain_Data %>% dplyr::select(Strain, Phylum, Class, Order, Family, Genus), by = "Strain") %>% 
  # mutate(Well_Row = Well %>% str_extract(.,"[A-H]"),
  #                   Well_Col = Well %>% str_extract(.,"[0-9]+") %>% as.numeric()) %>% 
  as.data.frame() #transform into df
## Save table with AUC
#write_rds(results_AUC_MBOA_AtSphere,paste0("Output/results_AUC_MBOA_AtSphere.rds"))
```

```{r, include=FALSE}
results_AUC_MBOA_AtSphere$Chem <- "MBOA"

results_AUC_all_raw <- results_AUC_MBOA_AtSphere

results_AUC_all_raw %<>% mutate(Chem_Conc = interaction(Chem, Conc, sep = "_"))

results_AUC_all_AtSphere <- results_AUC_all_raw # %>% filter(!Strain %in% c("LMU1", "LMG1", "LMF1", "LMI14", "LST17", "LAR4")) 

#write_rds(results_AUC_all_AtSphere, "Output/results_AUC_all_AtSphere.rds")
```

```{r, include=FALSE}
# scales::show_col(calc_pal()(12))
results_AUC_all_AtSphere %<>% na.omit()

results_AUC_all_AtSphere$cols_Family <- as.character(results_AUC_all_AtSphere$Family)

results_AUC_all_AtSphere[ results_AUC_all_AtSphere$Family=="Moraxellaceae", ]$cols_Family <- "#a73e62"
results_AUC_all_AtSphere[ results_AUC_all_AtSphere$Family=="Bacillaceae" , ]$cols_Family <- "#004586"
results_AUC_all_AtSphere[ results_AUC_all_AtSphere$Family=="Pseudomonadaceae" , ]$cols_Family <- "#4b1f6f"
# results_AUC_all_AtSphere[ results_AUC_all_AtSphere$Family=="Paenibacillaceae" , ]$cols_Family <- "#8f9ec9"
results_AUC_all_AtSphere[ results_AUC_all_AtSphere$Family=="Rhizobiaceae", ]$cols_Family <- "#ff950e"
results_AUC_all_AtSphere[ results_AUC_all_AtSphere$Family=="Sphingomonadaceae", ]$cols_Family <- "#c5000b"
results_AUC_all_AtSphere[ results_AUC_all_AtSphere$Family=="Xanthomonadaceae", ]$cols_Family <- "#0084d1"
results_AUC_all_AtSphere[ results_AUC_all_AtSphere$Family=="Microbacteriaceae" , ]$cols_Family <- "#7e0021"
results_AUC_all_AtSphere[ results_AUC_all_AtSphere$Family=="Micrococcaceae" , ]$cols_Family <- "#83caff"
# results_AUC_all_AtSphere[ results_AUC_all_AtSphere$Family=="Chitinophagaceae" , ]$cols_Family <- "#ff420e"
results_AUC_all_AtSphere[ results_AUC_all_AtSphere$Family=="Enterobacteriaceae" , ]$cols_Family <- "#ffd320"
# results_AUC_all_AtSphere[ results_AUC_all_AtSphere$Family=="Erwiniaceae" , ]$cols_Family <- "#579d1c"
# results_AUC_all_AtSphere[ results_AUC_all_AtSphere$Family=="Nocardioidaceae" , ]$cols_Family <- "#314004"
results_AUC_all_AtSphere[ results_AUC_all_AtSphere$Family=="Oxalobacteraceae" , ]$cols_Family <- "#aecf00"
# results_AUC_all_AtSphere[ results_AUC_all_AtSphere$Family=="Streptomycetaceae" , ]$cols_Family <- "#8b995a"

results_AUC_all_AtSphere[ results_AUC_all_AtSphere$Family=="Flavobacteriaceae" , ]$cols_Family <- "#F0D5B4"
# results_AUC_all_AtSphere[ results_AUC_all_AtSphere$Family=="Planococcaceae", ]$cols_Family <- "#7F5757"
# results_AUC_all_AtSphere[ results_AUC_all_AtSphere$Family=="Chitinophagaceae", ]$cols_Family <- "#6D4600"
results_AUC_all_AtSphere[ results_AUC_all_AtSphere$Family=="Sphingobacteriaceae", ]$cols_Family <- "#c75536"
# results_AUC_all_AtSphere[ results_AUC_all_AtSphere$Family=="Deinococcaceae", ]$cols_Family <- "#031e33"

# families not represented in MRB - ev change these colours
results_AUC_all_AtSphere[ results_AUC_all_AtSphere$Family=="Phyllobacteriaceae" , ]$cols_Family <- "#F0D5B4"
results_AUC_all_AtSphere[ results_AUC_all_AtSphere$Family=="Comamonadaceae", ]$cols_Family <- "#031e33"
results_AUC_all_AtSphere[ results_AUC_all_AtSphere$Family=="Alcaligenaceae", ]$cols_Family <- "#7F5757"

# results_AUC_all_AtSphere[ results_AUC_all_AtSphere$Family==NA, ]$cols_Family <- "dimgrey"
# table(results_AUC_all_AtSphere$cols_Family)

## collapsed color vector for each level
temp <- data.frame(results_AUC_all_AtSphere$Family, results_AUC_all_AtSphere$cols_Family)
temp <- plyr::ddply(temp, .variables="results_AUC_all_AtSphere.cols_Family", .fun=unique)   #library(plyr)
results_AUC_all_AtSphere_level_cols_Family <- as.character(temp[,2])
names(results_AUC_all_AtSphere_level_cols_Family) <- temp[,1]
```

```{r, include=FALSE}
#write_rds(results_AUC_all_AtSphere, "Output/results_AUC_all_AtSphere.rds")
```

```{r, include=FALSE}
sorted <- results_AUC_all_AtSphere %>% dplyr::arrange(Family) %>% dplyr::select(Family) %>% unique()
sorted_Family <- sorted$Family
```

## Plotting AUC 
```{r, message=FALSE, echo=FALSE, error=FALSE, warning = FALSE, fig.dim = c(10, 10)}
AUC_all_MBOA_AtSphere <- results_AUC_all_AtSphere %>% 
  filter(Chem %in% "MBOA") %>% 
  ggplot(aes(x = as.factor(Conc), y = AUC_norm)) +
  stat_summary(fun.y = mean, fun.ymax = max, fun.ymin = min, alpha = 0.4) +
  geom_point(aes(colour = Family), alpha=0.4, show.legend = FALSE) + 
  stat_compare_means(label = "p.signif", method = "t.test", ref.group = "0", label.y = 1.4, symnum.args = symnum.args) +
  ylim(c(0, 1.8))+
  facet_wrap(~ reorder(Strain, desc(Family)), nrow = 5, scales = "fixed") +
  theme_bw() +
  theme(axis.text.x = element_text(angle = -90, size = 8, hjust = 0, vjust = 0.5)) +
  scale_color_manual(values = results_AUC_all_AtSphere_level_cols_Family) +
  labs(y = "AUC",
       x = "concentration [μM]", 
       title = "growth in MBOA (ARB)",
       color = "Family")

AUC_all_MBOA_AtSphere

# ggsave(plot = AUC_all_MBOA_AtSphere,  filename = "AUC_all_MBOA_AtSphere.pdf", width = 25, height = 18, dpi = 300, scale = 1, units = "cm")
# 
# ggsave(plot = AUC_all_MBOA_AtSphere,  filename = "AUC_all_MBOA_AtSphere.svg", width = 25, height = 18, dpi = 300, scale = 1, units = "cm")
```

## Tolerance index 

To compare the tolerance among different bacterial strains, we use the tolerance index. The tolerance index is calculated from the area under the curve of the AUC of each strain in the different concentrations of the compounds, as further normalization the tolerance index is calculated from AUC_norm instead of AUC. Accordingly, a strain with TI_norm = 1 is completely tolerant to the compound in each concentration (not inhibited by the compound). Tolerance index is calculated for each compound separately using results_AUC from the respective compound. 

```{r, warning=FALSE, include=FALSE, echo=F}
AUC_tolind_MBOA_AtSphere <- results_AUC_all_AtSphere %>% 
  filter(AUC_norm < 1.2) %>% 
  filter(Chem %in% "MBOA") %>% 
  filter(!Strain %in% c("no", NA)) %>% 
  group_by(Strain, Replicate, Run) %>% 
  mutate(AUC_max = paste("1")) %>%
  dplyr::summarize(TI = auc(Conc, AUC_norm), 
            TI_raw = auc(Conc, AUC_raw),
            TI_max = auc(Conc, AUC_max)) %>% 
  mutate(TI_norm = TI / TI_max)

AUC_tolind_MBOA_AtSphere <- left_join(AUC_tolind_MBOA_AtSphere, Strain_Data %>% dplyr::select(Strain, Phylum, Family, Genus), by = "Strain") %>% unique()

## Save table with AUC
#write_rds(AUC_tolind_MBOA_AtSphere, paste0("Output/AUC_tolind_MBOA_AtSphere.rds"))
```

```{r, message=FALSE, echo=FALSE, error=FALSE, warning = FALSE, fig.dim = c(6, 5)}
TI_MBOA_ALL_AtSphere <- AUC_tolind_MBOA_AtSphere %>% 
  na.omit() %>% 
  filter(!Strain %in% "NBC") %>% 
  filter(TI_norm < 1.3) %>% 
  ggplot(aes(x = reorder(Strain, desc(TI_norm)), y = TI_norm)) + # to sort the x levels using y value descending
  geom_bar(aes(fill = Family), alpha = 0.5, stat = "summary", position = "dodge", show.legend = FALSE) +
  ggbeeswarm::geom_quasirandom(aes(colour = Family), size = 0.01/.pt, alpha = 0.7,  width = 0.3, show.legend = FALSE)+
  geom_errorbar(stat = "summary", size = 0.5/.pt, width=0.2) +
  geom_hline(colour = "grey50", yintercept = 0.75, size = 0.5/.pt) +
  geom_hline(colour = "grey50", yintercept = 0.5, size = 0.5/.pt) +
  # ylim(0, 1.4)+
  #stat_compare_means(method = "anova", aes(x = reorder(Strain, desc(TI_norm)), y = TI_norm), cex = 10/.pt, label.y = 1.2, label.x.npc = 0.6) +
  #stat_compare_means(label = "p.signif", method = "t.test", ref.group = "Soil745", symnum.args = symnum.args, hide.ns = TRUE, cex = 4/.pt, label.y = 1.1) +
  scale_fill_manual(values = results_AUC_all_AtSphere_level_cols_Family) +
  scale_colour_manual(values = results_AUC_all_AtSphere_level_cols_Family) +
  theme_classic()+
  scale_y_continuous(limits = c(0, 1.5), expand = expansion(mult = c(0, 0)))+
  theme(axis.text.x=element_text(angle = -90, hjust = 0, vjust = 0.5, size = 15/.pt),
        axis.text.y=element_text(size = 25/.pt),
        axis.title = element_text(size = 25/.pt),
        plot.title = element_text(size = 25/.pt)) +
  labs(title = "MBOA (AtSphere)", 
       fill = "Family", 
       y = "TI", 
       x = "")

TI_MBOA_ALL_AtSphere

# ggsave(plot = TI_MBOA_ALL_AtSphere,  filename = "TI_MBOA_ALL_AtSphere.pdf", width = 8.8, height = 6, dpi = 300, scale = 1, units = "cm")
# 
# ggsave(plot = TI_MBOA_ALL_AtSphere,  filename = "TI_MBOA_ALL_AtSphere.svg", width = 8.8, height = 6, dpi = 300, scale = 1, units = "cm")
```

```{r, include=FALSE}
aov(TI_norm ~ Family, data = AUC_tolind_MBOA_AtSphere) %>% summary()
```

## p-value heatmaps

```{r, message=FALSE, echo=FALSE, error=FALSE, warning = FALSE, fig.dim = c(6, 5)}
AUC_tolind_MBOA_AtSphere_mean <- AUC_tolind_MBOA_AtSphere %>% filter(TI_norm < 1.3) %>%  group_by(Strain) %>% dplyr::summarise(TI_norm_mean = mean(TI_norm))
AUC_tolind_MBOA_AtSphere_sorted <- left_join(AUC_tolind_MBOA_AtSphere %>% filter(TI_norm < 1.3), AUC_tolind_MBOA_AtSphere_mean, by = "Strain") %>% as.data.frame() %>% dplyr::arrange(desc(TI_norm_mean))

AUC_tolind_MBOA_AtSphere_sorted_strains <- AUC_tolind_MBOA_AtSphere_sorted$Strain %>% unique()

MBOA_AtSphere.TI.aov <- AUC_tolind_MBOA_AtSphere_sorted %>% aov(TI_norm ~ Strain, data = .)

MBOA_AtSphere.TI.tukey <- AUC_tolind_MBOA_AtSphere_sorted %>% aov(TI_norm ~ Strain, data = .) %>% tukey_hsd(ordered = TRUE, conf.level = 0.95)

MBOA_AtSphere.TI.tukey.sorted <- left_join(MBOA_AtSphere.TI.tukey, AUC_tolind_MBOA_AtSphere_mean, join_by(group1 == Strain) )

MBOA_AtSphere.AUC.TI.all.tukey.sorted <- left_join(AUC_tolind_MBOA_AtSphere %>% filter(TI_norm < 1.3), MBOA_AtSphere.TI.tukey, join_by(Strain == group1))


MBOA_AtSphere_p_HM_bin <-
  MBOA_AtSphere.AUC.TI.all.tukey.sorted %>%
  mutate(binary = case_when(p.adj > 0.05 ~ "ns", p.adj < 0.05 ~ "sig")) %>% 
  mutate(group2 = factor(group2, level = rev(AUC_tolind_MBOA_AtSphere_sorted_strains)),
          group1 = factor(Strain, level = AUC_tolind_MBOA_AtSphere_sorted_strains)) %>%
  ggplot(aes(x=group1, y = group2))+
  geom_tile(aes(fill = binary), color = "white", show.legend = TRUE)+
  scale_fill_manual(values = c("darkblue", "darkred"), labels = c("ns", "sig", ""), na.value = "white") +
  theme_classic()+ 
  theme(axis.text.x = element_text(angle = -90, size = 8, hjust = 0, vjust = 0.5), axis.text.y = element_text(angle = 0, size = 8, hjust = 0, vjust = 0))+
  theme(legend.position = c(0.2, 0.2))+
  labs(x = "", y = "")

# MBOA_AtSphere_p_HM_bin

MBOA_AtSphere_p_HM_bar <- AUC_tolind_MBOA_AtSphere %>% 
  filter(!Strain %in% "NBC") %>% 
  filter(TI_norm < 1.3) %>% 
  ggplot(aes(x = reorder(Strain, desc(TI_norm)), y = TI_norm)) + # to sort the x levels using y value descending
  geom_bar(aes(fill = Family), stat = "summary", position = "dodge", show.legend = FALSE, alpha = 0.5) +
  ggbeeswarm::geom_quasirandom(aes(color = Family), size = 0.01/.pt, alpha = 0.7,  width = 0.3, show.legend = FALSE)+
  geom_errorbar(stat = "summary", size = 0.5/.pt, width=0.2) +
  scale_fill_manual(values = results_AUC_all_AtSphere_level_cols_Family) +
  scale_color_manual(values = results_AUC_all_AtSphere_level_cols_Family) +
  theme_classic()+
  scale_y_continuous(limits = c(0, 2), expand = expansion(mult = c(0, 0)))+
  theme(axis.text.x=element_blank())+
  labs(x = "", y = "TI")
  
#MBOA_AtSphere_p_HM_bar


g2 <- ggplotGrob(MBOA_AtSphere_p_HM_bar)
g3 <- ggplotGrob(MBOA_AtSphere_p_HM_bin)
MBOA_AtSphere_p_HM_bar_bin <- rbind(g2, g3)
MBOA_AtSphere_p_HM_bar_bin$widths <- unit.pmax(g2$widths, g3$widths)
grid.newpage()
grid.draw(MBOA_AtSphere_p_HM_bar_bin)
 
# svg("MBOA_AtSphere_p_HM_bar_bin.svg", height = 12, width = 7)
# grid.draw(MBOA_AtSphere_p_HM_bar_bin)
# dev.off()

```

## Effect of gram on tolerance 
```{r, include=FALSE}
AUC_tolind_MBOA_AtSphere_tax <- left_join(AUC_tolind_MBOA_AtSphere, database %>% dplyr::select(Strain, gram), by = "Strain") 
```

```{r, include=FALSE}
# scales::show_col(calc_pal()(12))
AUC_tolind_MBOA_AtSphere_tax %<>% na.omit()

AUC_tolind_MBOA_AtSphere_tax$cols_Family <- as.character(AUC_tolind_MBOA_AtSphere_tax$Family)

AUC_tolind_MBOA_AtSphere_tax[ AUC_tolind_MBOA_AtSphere_tax$Family=="Moraxellaceae", ]$cols_Family <- "#a73e62"
AUC_tolind_MBOA_AtSphere_tax[ AUC_tolind_MBOA_AtSphere_tax$Family=="Bacillaceae" , ]$cols_Family <- "#004586"
AUC_tolind_MBOA_AtSphere_tax[ AUC_tolind_MBOA_AtSphere_tax$Family=="Pseudomonadaceae" , ]$cols_Family <- "#4b1f6f"
# AUC_tolind_MBOA_AtSphere_tax[ AUC_tolind_MBOA_AtSphere_tax$Family=="Paenibacillaceae" , ]$cols_Family <- "#8f9ec9"
AUC_tolind_MBOA_AtSphere_tax[ AUC_tolind_MBOA_AtSphere_tax$Family=="Rhizobiaceae", ]$cols_Family <- "#ff950e"
AUC_tolind_MBOA_AtSphere_tax[ AUC_tolind_MBOA_AtSphere_tax$Family=="Sphingomonadaceae", ]$cols_Family <- "#c5000b"
AUC_tolind_MBOA_AtSphere_tax[ AUC_tolind_MBOA_AtSphere_tax$Family=="Xanthomonadaceae", ]$cols_Family <- "#0084d1"
AUC_tolind_MBOA_AtSphere_tax[ AUC_tolind_MBOA_AtSphere_tax$Family=="Microbacteriaceae" , ]$cols_Family <- "#7e0021"
AUC_tolind_MBOA_AtSphere_tax[ AUC_tolind_MBOA_AtSphere_tax$Family=="Micrococcaceae" , ]$cols_Family <- "#83caff"
# AUC_tolind_MBOA_AtSphere_tax[ AUC_tolind_MBOA_AtSphere_tax$Family=="Chitinophagaceae" , ]$cols_Family <- "#ff420e"
AUC_tolind_MBOA_AtSphere_tax[ AUC_tolind_MBOA_AtSphere_tax$Family=="Enterobacteriaceae" , ]$cols_Family <- "#ffd320"
# AUC_tolind_MBOA_AtSphere_tax[ AUC_tolind_MBOA_AtSphere_tax$Family=="Erwiniaceae" , ]$cols_Family <- "#579d1c"
# AUC_tolind_MBOA_AtSphere_tax[ AUC_tolind_MBOA_AtSphere_tax$Family=="Nocardioidaceae" , ]$cols_Family <- "#314004"
AUC_tolind_MBOA_AtSphere_tax[ AUC_tolind_MBOA_AtSphere_tax$Family=="Oxalobacteraceae" , ]$cols_Family <- "#aecf00"
# AUC_tolind_MBOA_AtSphere_tax[ AUC_tolind_MBOA_AtSphere_tax$Family=="Streptomycetaceae" , ]$cols_Family <- "#8b995a"

AUC_tolind_MBOA_AtSphere_tax[ AUC_tolind_MBOA_AtSphere_tax$Family=="Flavobacteriaceae" , ]$cols_Family <- "#F0D5B4"
# AUC_tolind_MBOA_AtSphere_tax[ AUC_tolind_MBOA_AtSphere_tax$Family=="Planococcaceae", ]$cols_Family <- "#7F5757"
# AUC_tolind_MBOA_AtSphere_tax[ AUC_tolind_MBOA_AtSphere_tax$Family=="Chitinophagaceae", ]$cols_Family <- "#6D4600"
AUC_tolind_MBOA_AtSphere_tax[ AUC_tolind_MBOA_AtSphere_tax$Family=="Sphingobacteriaceae", ]$cols_Family <- "#c75536"
# AUC_tolind_MBOA_AtSphere_tax[ AUC_tolind_MBOA_AtSphere_tax$Family=="Deinococcaceae", ]$cols_Family <- "#031e33"

# families not represented in MRB - ev change these colours
AUC_tolind_MBOA_AtSphere_tax[ AUC_tolind_MBOA_AtSphere_tax$Family=="Phyllobacteriaceae" , ]$cols_Family <- "#F0D5B4"
AUC_tolind_MBOA_AtSphere_tax[ AUC_tolind_MBOA_AtSphere_tax$Family=="Comamonadaceae", ]$cols_Family <- "#031e33"
AUC_tolind_MBOA_AtSphere_tax[ AUC_tolind_MBOA_AtSphere_tax$Family=="Alcaligenaceae", ]$cols_Family <- "#7F5757"

# AUC_tolind_MBOA_AtSphere_tax[ AUC_tolind_MBOA_AtSphere_tax$Family==NA, ]$cols_Family <- "dimgrey"
# table(AUC_tolind_MBOA_AtSphere_tax$cols_Family)

## collapsed color vector for each level
temp <- data.frame(AUC_tolind_MBOA_AtSphere_tax$Family, AUC_tolind_MBOA_AtSphere_tax$cols_Family)
temp <- plyr::ddply(temp, .variables="AUC_tolind_MBOA_AtSphere_tax.cols_Family", .fun=unique)   #library(plyr)
AUC_tolind_MBOA_AtSphere_tax_level_cols_Family <- as.character(temp[,2])
names(AUC_tolind_MBOA_AtSphere_tax_level_cols_Family) <- temp[,1]
```

```{r, message=FALSE, echo=FALSE, error=FALSE, warning = FALSE, fig.dim = c(6, 5)}
Fig.supp.MBOAgram_AtSphere <- AUC_tolind_MBOA_AtSphere_tax %>% 
  filter(!gram %in% "NA") %>% 
  filter(TI_norm < 1.2) %>% 
  ggplot(aes(x= gram, y = TI_norm)) +
  geom_boxplot(aes(fill = gram), show.legend = FALSE) +
  geom_jitter(aes(colour = Family), alpha = 0.5, width = 0.2, show.legend = FALSE) +
  ylim(0, 1.4) +
  stat_compare_means(label = "p.signif", method = "t.test", ref.group = "negative", label.y = 1.3, cex = 15/.pt) +
  scale_fill_manual(values = c("grey50", "grey90"))+
  scale_color_manual(values = results_AUC_all_AtSphere_level_cols_Family)+
  theme_bw() +
  theme(axis.text.x=element_text(size = 35/.pt),
        axis.text.y=element_text(size = 60/.pt),
        axis.title = element_text(size = 60/.pt)) +
  labs(x = " ", y = "TI", title = "MBOA (ARB)")

Fig.supp.MBOAgram_AtSphere

# ggsave(plot = Fig.supp.MBOAgram_AtSphere,  filename = "Fig.supp.MBOAgram_AtSphere.pdf", width = 6.6, height = 12, dpi = 300, scale = 1, units = "cm")
 
# ggsave(plot = Fig.supp.MBOAgram_AtSphere,  filename = "Fig.supp.MBOAgram_AtSphere.svg", width = 6.6, height = 12, dpi = 300, scale = 1, units = "cm")
```

## Tolerance groups
```{r, include=FALSE}
BX_tolerance <- AUC_tolind_MBOA_AtSphere_tax %>% as.data.frame() %>%
  filter(TI_norm < 1.2) %>%
  # filter(TI_norm_MBOA < 1.2 & TI_norm_AMPO < 1.2 & TI_norm_BOA < 1.2 & TI_norm_APO < 1.2) %>%
  dplyr::group_by(Strain, Family) %>%
  dplyr::summarize(TI_norm_MBOA_mean = mean(TI_norm)) %>%
  dplyr::select(Strain, Family, TI_norm_MBOA_mean)

BX_tolerance %<>%
  mutate(MBOA_type = case_when(TI_norm_MBOA_mean > 0.75 ~ "tolerant",
                              TI_norm_MBOA_mean < 0.5 ~ "susceptible")) %>%
  replace_na(list(MBOA_type = "intermediate"))
```

```{r, echo=FALSE}
BX_tolerance %<>% as.data.frame()

BX_tolerance_table <- BX_tolerance %>% dplyr::select(MBOA_type) %>% table() %>% as.data.frame() %>% dplyr::rename(MBOA = Freq)

write.csv(BX_tolerance_table, "Output/BX_tolerance_table.csv")
```

fisher exact
```{r, echo=FALSE}
MBOA_type_test <- fisher.test(BX_tolerance$Family, BX_tolerance$MBOA_type,
            alternative = "two.sided",
            conf.int = TRUE, conf.level = 0.95)

p.value <- rbind(MBOA_type_test$p.value)
Compound <- rbind("MBOA")
Compound_p.value <- cbind(Compound, p.value)
Compound_p.value <- Compound_p.value %>% as.data.frame()
colnames(Compound_p.value) <- c("Compound", "p.value")

Compound_p.value %>% knitr::kable()
```

## Correlation TI norm MBOA MRB ~ AtSphere families
```{r, include=FALSE}
#AUC_tolind_MBOA_cor <- read_rds("Output/AUC_tolind_MBOA_AMPO_BOA_APO_DMG_cor.rds") 

AUC_tolind_MBOA_cor <- AUC_tolind_MBOA_AMPO_BOA_APO_DMG_cor %>% dplyr::select(-TI_norm_AMPO, -TI_norm_APO, -TI_norm_BOA, -TI_norm_DMG)
```

```{r, include=FALSE}
AUC_tolind_MBOA_AtSphere_cor_fam <- AUC_tolind_MBOA_cor %>% as.data.frame() %>% group_by(Strain, Family) %>% 
  dplyr::summarize(TI_norm_MBOA_mean = mean(TI_norm_MBOA)) %>% 
  dplyr::select(Family, TI_norm_MBOA_mean) %>% as.data.frame() %>% dplyr::select(-Strain)  %>% unique()

AUC_tolind_MBOA_AtSphere_tax_fam <- AUC_tolind_MBOA_AtSphere_tax %>% as.data.frame() %>% group_by(Strain, Family) %>% 
  dplyr::summarize(TI_norm_mean = mean(TI_norm)) %>% 
  dplyr::select(Family, TI_norm_mean) %>% as.data.frame() %>% dplyr::select(-Strain) %>% unique()
```

```{r, include=FALSE}
AUC_tolind_MBOA_AtSphere_MRB_AtSphere <- left_join(AUC_tolind_MBOA_AtSphere_cor_fam %>% unique(), AUC_tolind_MBOA_AtSphere_tax_fam %>% unique(), by = "Family") %>% dplyr::rename(TI_norm_MBOA_MRB = TI_norm_MBOA_mean, TI_norm_AtSphere = TI_norm_mean)
```

```{r, message=FALSE, echo=FALSE, error=FALSE, warning = FALSE, fig.dim = c(5, 5)}
Fig.TIcor_MRB_AtSphere <- AUC_tolind_MBOA_AtSphere_MRB_AtSphere %>% unique() %>% 
  ggplot(aes(x = TI_norm_MBOA_MRB, y = TI_norm_AtSphere)) +
  geom_point(aes(colour = Family), show.legend = FALSE) +
  geom_smooth(method = "lm", color= "black", formula = y ~ x)  + 
  xlim(0, 1.25)+
  ylim(0, 1.25)+
  stat_cor(method = "pearson", label.x = 0.4, label.y = 1.1) +
  # stat_cor(aes(label =paste(..r.label.., cut(..p.., 
  #                                             breaks = c(-Inf, 0.0001, 0.001, 0.01, 0.05, Inf),
  #                                             labels = c("'****'", "'***'", "'**'", "'*'", "'ns'")), 
  #                           sep = "~")), method = "pearson", label.x = 0.2, label.y = 1.1, cex = 15/.pt) +
  scale_colour_manual(values = results_AUC_all_AtSphere_level_cols_Family) +
  theme_bw() +
  theme(axis.text.x=element_text(size = 35/.pt),
        axis.text.y=element_text(size = 35/.pt),
        axis.title = element_text(size = 35/.pt)) +
  labs(title = "MBOA", 
       x = "TI MRB",
       y = "TI ARB")

Fig.TIcor_MRB_AtSphere

# ggsave(plot = Fig.TIcor_MRB_AtSphere,  filename = "Fig.TIcor_MRB_AtSphere.pdf", width = 9, height = 9, dpi = 300, scale = 1, units = "cm")
 
# ggsave(plot = Fig.TIcor_MRB_AtSphere,  filename = "Fig.TIcor_MRB_AtSphere.svg", width = 9, height = 9, dpi = 300, scale = 1, units = "cm")


```

```{r, include=FALSE}
AUC_tolind_MBOA_AtSphere_MRB_AtSphere_mean <- AUC_tolind_MBOA_AtSphere_MRB_AtSphere %>% group_by(Family) %>% unique() %>% 
  dplyr::summarize(TI_norm_MBOA_MRB_mean = mean(TI_norm_MBOA_MRB), TI_norm_AtSphere_mean = mean(TI_norm_AtSphere)) 
```

```{r, message=FALSE, echo=FALSE, error=FALSE, warning = FALSE, fig.dim = c(5, 5)}
Fig.TIcor_MRB_AtSphere_mean <- AUC_tolind_MBOA_AtSphere_MRB_AtSphere_mean %>% na.omit() %>% unique() %>% 
  ggplot(aes(x = TI_norm_MBOA_MRB_mean, y = TI_norm_AtSphere_mean)) +
  geom_point(aes(colour = Family), show.legend = TRUE) +
  geom_smooth(method = "lm", color= "black", formula = y ~ x)  + 
  xlim(0, 1.25)+
  ylim(0, 1.25)+
  stat_cor(aes(label =paste(..rr.label.., cut(..p.., 
                                              breaks = c(-Inf, 0.0001, 0.001, 0.01, 0.05, Inf),
                                              labels = c("'****'", "'***'", "'**'", "'*'", "'ns'")), 
                            sep = "~")), method = "pearson", label.x = 0.2, label.y = 1.1, cex = 15/.pt) +
  scale_colour_manual(values = results_AUC_all_AtSphere_level_cols_Family) +
  theme_bw() +
  labs(title = "MBOA", 
       x = "TI MRB",
       y = "TI AtSphere")


Fig.TIcor_MRB_AtSphere_mean
```

## Compare families MRB ~ AtSphere

```{r, include=FALSE}
#AUC_tolind_MBOA_AtSphere_cor <- read_rds("Output/AUC_tolind_MBOA_AMPO_BOA_APO_DMG_cor.rds") 

AUC_tolind_MBOA_AtSphere_cor <-  AUC_tolind_MBOA_AMPO_BOA_APO_DMG_cor %>% 
  dplyr::select(-TI_norm_AMPO, -TI_norm_APO, -TI_norm_BOA, -TI_norm_DMG) %>% 
  dplyr::rename(TI_norm = TI_norm_MBOA)

```

```{r, include=FALSE}
TI_MBOA_AtSphere <- rbind(AUC_tolind_MBOA_AtSphere_tax %>% 
                            dplyr::select(Strain, Replicate, Phylum, Family, Genus, Run, TI_norm) %>% mutate(Plant = "arabidopsis"),
                            AUC_tolind_MBOA_AtSphere_cor %>% mutate(Plant = "maize"))
```

```{r,  message=FALSE, echo=FALSE, error=FALSE, warning = FALSE, fig.dim = c(6, 5)}
AtSphereFamilyplant <- TI_MBOA_AtSphere %>% 
  na.omit() %>% 
  filter(Family %in% c("Bacillaceae", "Enterobacteriaceae", "Microbacteriaceae", "Micrococcaceae", "Pseudomonadaceae", "Rhizobiaceae", "Sphingomonadaceae", "Xanthomonadaceae")) %>% 
  filter(TI_norm < 1.3) %>% 
  unique() %>% 
  mutate(Plant = factor(Plant, levels = c("maize", "arabidopsis"))) %>% 
  ggplot(aes(x = Plant, y = TI_norm)) +
  geom_bar(aes(fill = Family), alpha = 0.5, stat = "summary", position = "dodge", show.legend = FALSE) +
  ggbeeswarm::geom_quasirandom(aes(colour = Family), size = 0.1/.pt, alpha = 0.7,  width = 0.3, show.legend = FALSE)+
  geom_errorbar(stat = "summary", size = 0.5/.pt, width=0.2) +
  geom_hline(colour = "grey50", yintercept = 0.75, size = 0.5/.pt) +
  geom_hline(colour = "grey50", yintercept = 0.5, size = 0.5/.pt) +
  geom_errorbar(stat = "summary", width=0.2) +
  stat_compare_means(label = "p.signif", method = "t.test", ref.group = "maize", hide.ns = TRUE, cex = 4/.pt, label.y = 1.1, symnum.args = symnum.args) +
  scale_fill_manual(values = results_AUC_all_level_cols_Family) +
  scale_colour_manual(values = results_AUC_all_level_cols_Family) +
  theme_classic()+
  theme(strip.background = element_rect(color = NULL, fill= "white", linetype="blank"), strip.text.x = NULL)+
  theme(axis.text.x=element_text(angle = -90, hjust = 0, vjust = 0.5, size = 25/.pt),
        axis.text.y=element_text(size = 25/.pt),
        axis.title = element_text(size = 25/.pt),
        plot.title = element_text(size = 25/.pt)) +
  scale_y_continuous(expand = expansion(mult = c(0, 0.1)))+
  scale_x_discrete(labels= c("MRB","ARB"))+
  facet_wrap(~Family, nrow = 1) +
  labs(title = "MBOA (ARB)", 
       fill = "Family", 
       y = "TI", 
       x = "")


AtSphereFamilyplant


# ggsave(plot = AtSphereFamilyplant,  filename = "AtSphereFamilyplant.pdf", width = 4.4, height = 3, dpi = 300, scale = 3, units = "cm")
# 
# ggsave(plot = AtSphereFamilyplant,  filename = "AtSphereFamilyplant.svg", width = 4.4, height = 3, dpi = 300, scale = 3, units = "cm")
```

# III) MICROBIOME DATA
We mapped the 16s rRNA sequences to the microbiome dataset published by Hu et al. 2018. In this study, maize wild-type plants (producing BXs) were grown along with bx1 mutant plants in the field. Then the plants were harvested and the root, the rhizosphere and the soil compartment were sequenced. The bacterial community profiles of wild-type and bx1 mutant plants differed significantely. 
We found that most of the MRB isolates map to taxonomic units (OTUs) in the dataset and many of them map to abundant OTUs. 
Here we investigate the BX-dependent colonization of these OTUs in the microbiome dataset and correlate those with the tolerance index of the strains. 

```{r, include=FALSE}
BX_abundance_OTU_Ch_new <- read_rds("Input/iso_abundance_rank_selected_field.rds")
BX_abundance_OTU_Ch_new_single_data <- read_rds("Input/iso_abundance_single_data_selected_field.rds")
```

```{r, include=FALSE}
BX_abundance_OTU_Ch_new_single_data$cols_Family <- as.character(BX_abundance_OTU_Ch_new_single_data$family)

BX_abundance_OTU_Ch_new_single_data$cols_family <- "dimgrey"
BX_abundance_OTU_Ch_new_single_data[ BX_abundance_OTU_Ch_new_single_data$family=="Moraxellaceae", ]$cols_family <- "#a73e62"
BX_abundance_OTU_Ch_new_single_data[ BX_abundance_OTU_Ch_new_single_data$family=="Bacillaceae" , ]$cols_family <- "#004586"
BX_abundance_OTU_Ch_new_single_data[ BX_abundance_OTU_Ch_new_single_data$family=="Pseudomonadaceae" , ]$cols_family <- "#4b1f6f"
BX_abundance_OTU_Ch_new_single_data[ BX_abundance_OTU_Ch_new_single_data$family=="Paenibacillaceae" , ]$cols_family <- "#8f9ec9"
BX_abundance_OTU_Ch_new_single_data[ BX_abundance_OTU_Ch_new_single_data$family=="Rhizobiaceae", ]$cols_family <- "#ff950e"
BX_abundance_OTU_Ch_new_single_data[ BX_abundance_OTU_Ch_new_single_data$family=="Sphingomonadaceae", ]$cols_family <- "#c5000b"
BX_abundance_OTU_Ch_new_single_data[ BX_abundance_OTU_Ch_new_single_data$family=="Xanthomonadaceae", ]$cols_family <- "#0084d1"
BX_abundance_OTU_Ch_new_single_data[ BX_abundance_OTU_Ch_new_single_data$family=="Microbacteriaceae" , ]$cols_family <- "#7e0021"
BX_abundance_OTU_Ch_new_single_data[ BX_abundance_OTU_Ch_new_single_data$family=="Micrococcaceae" , ]$cols_family <- "#83caff"
BX_abundance_OTU_Ch_new_single_data[ BX_abundance_OTU_Ch_new_single_data$family=="Chitinophagaceae" , ]$cols_family <- "#ff420e"
BX_abundance_OTU_Ch_new_single_data[ BX_abundance_OTU_Ch_new_single_data$family=="Enterobacteriaceae" , ]$cols_family <- "#ffd320"
BX_abundance_OTU_Ch_new_single_data[ BX_abundance_OTU_Ch_new_single_data$family=="Erwiniaceae" , ]$cols_family <- "#579d1c"
BX_abundance_OTU_Ch_new_single_data[ BX_abundance_OTU_Ch_new_single_data$family=="Nocardioidaceae" , ]$cols_family <- "#314004"
BX_abundance_OTU_Ch_new_single_data[ BX_abundance_OTU_Ch_new_single_data$family=="Oxalobacteraceae" , ]$cols_family <- "#aecf00"
# BX_abundance_OTU_Ch_new_single_data[ BX_abundance_OTU_Ch_new_single_data$Family=="Streptomycetaceae" , ]$cols_Family <- "#8b995a"
# BX_abundance_OTU_Ch_new_single_data[ BX_abundance_OTU_Ch_new_single_data$Family=="Flavobacteriaceae" , ]$cols_Family <- "#F0D5B4"
#BX_abundance_OTU_Ch_new_single_data[ BX_abundance_OTU_Ch_new_single_data$Family=="Planococcaceae", ]$cols_Family <- "#7F5757"
# BX_abundance_OTU_Ch_new_single_data[ BX_abundance_OTU_Ch_new_single_data$Family=="Chitinophagales", ]$cols_Family <- "#6D4600"
#BX_abundance_OTU_Ch_new_single_data[ BX_abundance_OTU_Ch_new_single_data$Family=="Sphingobacteriaceae", ]$cols_Family <- "#c75536"
#BX_abundance_OTU_Ch_new_single_data[ BX_abundance_OTU_Ch_new_single_data$Family=="Deinococcaceae", ]$cols_Family <- "#031e33"

#BX_abundance_OTU_Ch_new_single_data[ BX_abundance_OTU_Ch_new_single_data$family==NA, ]$cols_family <- "dimgrey"
#table(BX_abundance_OTU_Ch_new_single_data$cols_family)

## collapsed color vector for each level
temp <- data.frame(BX_abundance_OTU_Ch_new_single_data$family, BX_abundance_OTU_Ch_new_single_data$cols_family)
temp <- plyr::ddply(temp, .variables="BX_abundance_OTU_Ch_new_single_data.cols_family", .fun=unique)   #library(plyr)
BX_abundance_OTU_Ch_new_single_data_level_cols_family <- as.character(temp[,2])
names(BX_abundance_OTU_Ch_new_single_data_level_cols_family) <- temp[,1]
```

In this graph the log2foldchange of OTUs on WT/bx1 roots in the field is visualized.

Stats was done like this: iso_abundance_single_data_long %>% group_by(OTU, compartment) %>%
  do(tidy(t.test(Abundance~genotype, data=.))) %>% 
  add_significance("p.value") %>% 
  dplyr::select(OTU, p.value, p.value.signif) 

```{r, message=FALSE, echo=FALSE, error=FALSE, warning = FALSE, fig.dim = c(6, 5)}
BX_abundace_L2FC <- BX_abundance_OTU_Ch_new_single_data %>% 
  filter(!family %in% c("Streptomycetaceae", "Sphingobacteriaceae")) %>% 
  mutate(p.value.signif = gsub("ns", "", p.value.signif)) %>% 
  mutate(p.value.signif = gsub("?", "", p.value.signif)) %>% 
   dplyr::select(-Strain) %>% 
  unique() %>% 
  mutate(compartment = gsub("rhizo", "rhizosphere", compartment)) %>% 
  mutate(compartment = factor(compartment, levels = c("root", "rhizosphere"))) %>% 
  filter(!log2FC %in% c(-Inf, Inf, NA, NaN)) %>%
  ggplot(aes(x = log2FC, y = reorder(OTU, Abundance_WT))) +
  geom_boxplot(aes(colour = family)) + 
  geom_point(aes(colour = family)) +
  # stat_compare_means(label = "p.signif", method = "t.test", ref.group = "LMI1x", symnum.args = symnum.args, hide.ns = TRUE, cex = 4/.pt, label.y = 1.1) +
  geom_text(aes(label=p.value.signif), colour = "black", x = -5, size = 5)+
  facet_wrap(~ compartment)+
  theme_bw()+
  scale_color_manual(values = BX_abundance_OTU_Ch_new_single_data_level_cols_family) +
  labs(y = "OTU")

BX_abundace_L2FC

# ggsave(plot = BX_abundace_L2FC,  filename = "BX_abundace_L2FC.pdf", width = 15, height = 12, dpi = 300, scale = 1, units = "cm")
# 
# ggsave(plot = BX_abundace_L2FC,  filename = "BX_abundace_L2FC.svg", width = 15, height = 12, dpi = 300, scale = 1, units = "cm")

```

## Correlation with tolerance data

```{r, include=FALSE}
#AUC_tolind_MBOA_AMPO_BOA_APO_DMG_cor <- read_rds("Output/AUC_tolind_MBOA_AMPO_BOA_APO_DMG_cor.rds")
```

```{r, include=FALSE}
isomapTolOTU_new <- left_join(AUC_tolind_MBOA_AMPO_BOA_APO_DMG_cor, BX_abundance_OTU_Ch_new, by = "Strain")
```

```{r, include=FALSE}
isomapTolOTU_new$cols_Family <- as.character(isomapTolOTU_new$Family)

isomapTolOTU_new[ isomapTolOTU_new$Family=="Moraxellaceae", ]$cols_Family <- "#a73e62"
isomapTolOTU_new[ isomapTolOTU_new$Family=="Bacillaceae" , ]$cols_Family <- "#004586"
isomapTolOTU_new[ isomapTolOTU_new$Family=="Pseudomonadaceae" , ]$cols_Family <- "#4b1f6f"
isomapTolOTU_new[ isomapTolOTU_new$Family=="Paenibacillaceae" , ]$cols_Family <- "#8f9ec9"
isomapTolOTU_new[ isomapTolOTU_new$Family=="Rhizobiaceae", ]$cols_Family <- "#ff950e"
isomapTolOTU_new[ isomapTolOTU_new$Family=="Sphingomonadaceae", ]$cols_Family <- "#c5000b"
isomapTolOTU_new[ isomapTolOTU_new$Family=="Xanthomonadaceae", ]$cols_Family <- "#0084d1"
isomapTolOTU_new[ isomapTolOTU_new$Family=="Microbacteriaceae" , ]$cols_Family <- "#7e0021"
isomapTolOTU_new[ isomapTolOTU_new$Family=="Micrococcaceae" , ]$cols_Family <- "#83caff"
isomapTolOTU_new[ isomapTolOTU_new$Family=="Chitinophagaceae" , ]$cols_Family <- "#ff420e"
isomapTolOTU_new[ isomapTolOTU_new$Family=="Enterobacteriaceae" , ]$cols_Family <- "#ffd320"
isomapTolOTU_new[ isomapTolOTU_new$Family=="Erwiniaceae" , ]$cols_Family <- "#579d1c"
isomapTolOTU_new[ isomapTolOTU_new$Family=="Nocardioidaceae" , ]$cols_Family <- "#314004"
isomapTolOTU_new[ isomapTolOTU_new$Family=="Oxalobacteraceae" , ]$cols_Family <- "#aecf00"

## collapsed color vector for each level
temp <- data.frame(isomapTolOTU_new$Family, isomapTolOTU_new$cols_Family)
temp <- plyr::ddply(temp, .variables="isomapTolOTU_new.cols_Family", .fun=unique)   #library(plyr)
isomapTolOTU_new_level_cols_Family <- as.character(temp[,2])
names(isomapTolOTU_new_level_cols_Family) <- temp[,1]
```

```{r, include=FALSE}
isomapTolOTU_new_single <- left_join(AUC_tolind_MBOA_AMPO_BOA_APO_DMG_cor, BX_abundance_OTU_Ch_new_single_data, by = "Strain")
```

```{r, include=FALSE}

isomapTolOTU_new_single$cols_Family <- as.character(isomapTolOTU_new_single$Family)

isomapTolOTU_new_single[ isomapTolOTU_new_single$Family=="Moraxellaceae", ]$cols_Family <- "#a73e62"
isomapTolOTU_new_single[ isomapTolOTU_new_single$Family=="Bacillaceae" , ]$cols_Family <- "#004586"
isomapTolOTU_new_single[ isomapTolOTU_new_single$Family=="Pseudomonadaceae" , ]$cols_Family <- "#4b1f6f"
isomapTolOTU_new_single[ isomapTolOTU_new_single$Family=="Paenibacillaceae" , ]$cols_Family <- "#8f9ec9"
isomapTolOTU_new_single[ isomapTolOTU_new_single$Family=="Rhizobiaceae", ]$cols_Family <- "#ff950e"
isomapTolOTU_new_single[ isomapTolOTU_new_single$Family=="Sphingomonadaceae", ]$cols_Family <- "#c5000b"
isomapTolOTU_new_single[ isomapTolOTU_new_single$Family=="Xanthomonadaceae", ]$cols_Family <- "#0084d1"
isomapTolOTU_new_single[ isomapTolOTU_new_single$Family=="Microbacteriaceae" , ]$cols_Family <- "#7e0021"
isomapTolOTU_new_single[ isomapTolOTU_new_single$Family=="Micrococcaceae" , ]$cols_Family <- "#83caff"
isomapTolOTU_new_single[ isomapTolOTU_new_single$Family=="Chitinophagaceae" , ]$cols_Family <- "#ff420e"
isomapTolOTU_new_single[ isomapTolOTU_new_single$Family=="Enterobacteriaceae" , ]$cols_Family <- "#ffd320"
isomapTolOTU_new_single[ isomapTolOTU_new_single$Family=="Erwiniaceae" , ]$cols_Family <- "#579d1c"
isomapTolOTU_new_single[ isomapTolOTU_new_single$Family=="Nocardioidaceae" , ]$cols_Family <- "#314004"
isomapTolOTU_new_single[ isomapTolOTU_new_single$Family=="Oxalobacteraceae" , ]$cols_Family <- "#aecf00"

## collapsed color vector for each level
temp <- data.frame(isomapTolOTU_new_single$Family, isomapTolOTU_new_single$cols_Family)
temp <- plyr::ddply(temp, .variables="isomapTolOTU_new_single.cols_Family", .fun=unique)   #library(plyr)
isomapTolOTU_new_single_level_cols_Family <- as.character(temp[,2])
names(isomapTolOTU_new_single_level_cols_Family) <- temp[,1]
```

```{r, include=FALSE}
isomapTolOTU_new_mean_LFC <- isomapTolOTU_new_single %>% 
  filter(!log2FC %in% c(-Inf, Inf, NA, NaN)) %>%
  mutate(compartment = factor(compartment, levels = c("root", "rhizo"))) %>%
  # mutate(soil = factor(soil, levels = c("BXp", "BXm"))) %>% 
  group_by(Strain, Family, compartment) %>% 
  dplyr::summarize(TI_norm_MBOA_mean = mean(TI_norm_MBOA), 
                   TI_norm_BOA_mean = mean(TI_norm_BOA), 
                   TI_norm_AMPO_mean = mean(TI_norm_AMPO), 
                   TI_norm_APO_mean = mean(TI_norm_APO),
                   #AUC_norm_DMG_mean = mean(AUC_norm_DMG %>% na.omit()),
                   TI_norm_DMG_mean = mean(TI_norm_DMG %>% na.omit()),
                   WT_bx_dif_mean = mean(WT_bx_dif), 
                   log2FC_mean = mean(log2FC), 
                   Family = Family)
```

*MBOA root*

```{r, message=FALSE, echo=FALSE, error=FALSE, warning = FALSE, fig.dim = c(6, 5)}
fIG4A.TIMBOA_cor_logFC_field_root <- isomapTolOTU_new_mean_LFC %>% 
  filter(compartment %in% "root") %>% 
  unique() %>% 
  ggplot(aes(y = TI_norm_MBOA_mean, x = log2FC_mean)) + 
  geom_point(aes(color = Family), size = 2, show.legend = FALSE) +
  geom_smooth(method = "lm", color="black", formula = y ~ x)  + 
  # stat_cor(aes(label =paste(..r.label.., cut(..p.., 
  #                                              breaks = c(-Inf, 0.0001, 0.001, 0.01, 0.05, Inf),
  #                                              labels = c("'****'", "'***'", "'**'", "'*'", "'ns'")), 
  #                            sep = "~")), method = "pearson", label.x = -1.5, label.y = 1.1, cex = 15/.pt) +
  stat_cor(method = "pearson", label.x = -1.5, label.y = 1.1) +
  scale_colour_manual(values = isomapTolOTU_new_single_level_cols_Family) +
  theme_bw() +
  # facet_grid(~compartment, scales = "free") +
  labs(y = "TI",
       x = "log2FC",
       color = "Family",
       title = expression("TI"["MBOA"]* "~ root microbiome"))

fIG4A.TIMBOA_cor_logFC_field_root

# ggsave(plot = fIG4A.TIMBOA_cor_logFC_field_root,  filename = "fIG4A.TIMBOA_cor_logFC_field_root.pdf", width = 9, height = 9, dpi = 300, scale = 1, units = "cm")
# #
# ggsave(plot = fIG4A.TIMBOA_cor_logFC_field_root,  filename = "fIG4A.TIMBOA_cor_logFC_field_root.svg", width = 9, height = 9, dpi = 300, scale = 1, units = "cm")
```

*MBOA rhizosphere*

```{r, message=FALSE, echo=FALSE, error=FALSE, warning = FALSE, fig.dim = c(6, 5)}
fIG4A.TIMBOA_cor_logFC_field_rhizo <- isomapTolOTU_new_mean_LFC %>% 
  filter(compartment %in% "rhizo") %>% 
  unique() %>% 
  ggplot(aes(y = TI_norm_MBOA_mean, x = log2FC_mean)) + 
  geom_point(aes(color = Family), size = 2, show.legend = FALSE) +
  geom_smooth(method = "lm", color="black", formula = y ~ x)  + 
  # stat_cor(aes(label =paste(..r.label.., cut(..p.., 
  #                                             breaks = c(-Inf, 0.0001, 0.001, 0.01, 0.05, Inf),
  #                                             labels = c("'****'", "'***'", "'**'", "'*'", "'ns'")), 
  #                           sep = "~")), method = "pearson", label.x = -1.5, label.y = 1.1, cex = 15/.pt)+
  stat_cor(method = "pearson", label.x = -1.5, label.y = 1.1) +
  scale_colour_manual(values = isomapTolOTU_new_single_level_cols_Family) +
  theme_bw() +
  labs(y = "TI",
       x = "log2FC",
       color = "Family",
       title = expression("TI"["MBOA"]* "~ rhizosphere microbiome"))

fIG4A.TIMBOA_cor_logFC_field_rhizo

# ggsave(plot = fIG4A.TIMBOA_cor_logFC_field_rhizo,  filename = "fIG4A.TIMBOA_cor_logFC_field_rhizo.pdf", width = 9, height = 9, dpi = 300, scale = 1, units = "cm")
# 
# ggsave(plot = fIG4A.TIMBOA_cor_logFC_field_rhizo,  filename = "fIG4A.TIMBOA_cor_logFC_field_rhizo.svg", width = 9, height = 9, dpi = 300, scale = 1, units = "cm")
```

*BOA*

```{r, message=FALSE, echo=FALSE, error=FALSE, warning = FALSE, fig.dim = c(6, 5)}
fIGsupp.TIBOA_cor_logFC_field_root_rhizo <- isomapTolOTU_new_mean_LFC %>% 
  # filter(OTU %in% selected_OTU) %>% 
  unique() %>% 
  mutate(compartment = gsub("rhizo", "rhizosphere", compartment)) %>% 
  mutate(compartment = factor(compartment, levels = c("root", "rhizosphere"))) %>% 
  ggplot(aes(y = TI_norm_BOA_mean, x = log2FC_mean)) + 
  geom_point(aes(color = Family), size = 2, show.legend = FALSE) +
  geom_smooth(method = "lm", color="black", formula = y ~ x)  + 
  # geom_text(aes(label = OTU))+
   # stat_cor(aes(label =paste(..r.label.., cut(..p.., 
   #                                            breaks = c(-Inf, 0.0001, 0.001, 0.01, 0.05, Inf),
   #                                            labels = c("'****'", "'***'", "'**'", "'*'", "'ns'")), 
   #                          sep = "~")), method = "pearson", label.x = -1.5, label.y = 1.4, cex = 15/.pt)+
  stat_cor(method = "pearson", label.x = -1.5, label.y = 1.5) +
  ylim(0, 1.5)+
  # xlim(-3, 2)+
  # geom_text(aes(label  = Strain)) +
  scale_colour_manual(values = isomapTolOTU_new_single_level_cols_Family) +
  theme_bw() +
  facet_grid(~compartment, scales = "free") +
  labs(y = "TI",
       x = "log2FC",
       color = "Family",
       title = expression("TI"["BOA"]* "~ root & rhizosphere microbiome"))

fIGsupp.TIBOA_cor_logFC_field_root_rhizo

# ggsave(plot = fIGsupp.TIBOA_cor_logFC_field_root_rhizo,  filename = "fIGsupp.TIBOA_cor_logFC_field_root_rhizo.pdf", width = 6, height = 3, dpi = 300, scale = 3, units = "cm")
# 
# ggsave(plot = fIGsupp.TIBOA_cor_logFC_field_root_rhizo,  filename = "fIGsupp.TIBOA_cor_logFC_field_root_rhizo.svg", width = 6, height = 3, dpi = 300, scale = 3, units = "cm")
```

*AMPO*

```{r, message=FALSE, echo=FALSE, error=FALSE, warning = FALSE, fig.dim = c(6, 5)}
fIGsupp.TIAMPO_cor_logFC_field_root_rhizo <- isomapTolOTU_new_mean_LFC %>% 
  # filter(OTU %in% selected_OTU) %>% 
  unique() %>% 
  mutate(compartment = gsub("rhizo", "rhizosphere", compartment)) %>% 
  mutate(compartment = factor(compartment, levels = c("root", "rhizosphere"))) %>% 
  ggplot(aes(y = TI_norm_AMPO_mean, x = log2FC_mean)) + 
  geom_point(aes(color = Family), size = 2, show.legend = FALSE) +
  geom_smooth(method = "lm", color="black", formula = y ~ x)  + 
  # geom_text(aes(label = OTU))+
  # stat_cor(aes(label =paste(..r.label.., cut(..p.., 
  #                                             breaks = c(-Inf, 0.0001, 0.001, 0.01, 0.05, Inf),
  #                                             labels = c("'****'", "'***'", "'**'", "'*'", "'ns'")), 
  #                           sep = "~")), method = "pearson", label.x = -2, label.y = 1.4) +
  stat_cor(method = "pearson", label.x = -1.5, label.y = 1.5) +
  ylim(0, 1.5)+
  # xlim(-3, 2)+
  # geom_text(aes(label  = Strain)) +
  scale_colour_manual(values = isomapTolOTU_new_single_level_cols_Family) +
  theme_bw() +
  facet_grid(~compartment, scales = "free") +
  labs(y = "TI",
       x = "log2FC",
       color = "Family",
       title = expression("TI"["AMPO"]* "~ root & rhizosphere microbiome"))

fIGsupp.TIAMPO_cor_logFC_field_root_rhizo

# ggsave(plot = fIGsupp.TIAMPO_cor_logFC_field_root_rhizo,  filename = "fIGsupp.TIAMPO_cor_logFC_field_root_rhizo.pdf", width = 6, height = 3, dpi = 300, scale = 3, units = "cm")
# 
# ggsave(plot = fIGsupp.TIAMPO_cor_logFC_field_root_rhizo,  filename = "fIGsupp.TIAMPO_cor_logFC_field_root_rhizo.svg", width = 6, height = 3, dpi = 300, scale = 3, units = "cm")
```

*APO*

```{r, message=FALSE, echo=FALSE, error=FALSE, warning = FALSE,  fig.dim = c(6, 5)}
fIGsupp.TIAPO_cor_logFC_field_root_rhizo <- isomapTolOTU_new_mean_LFC %>% 
  # filter(OTU %in% selected_OTU) %>% 
  mutate(compartment = gsub("rhizo", "rhizosphere", compartment)) %>% 
  mutate(compartment = factor(compartment, levels = c("root", "rhizosphere"))) %>% 
  unique() %>% 
  ggplot(aes(y = TI_norm_APO_mean, x = log2FC_mean)) + 
  geom_point(aes(color = Family), size = 2, show.legend = FALSE) +
  geom_smooth(method = "lm", color="black", formula = y ~ x)  + 
  # geom_text(aes(label = OTU))+
  # stat_cor(aes(label =paste(..r.label.., cut(..p.., 
  #                                             breaks = c(-Inf, 0.0001, 0.001, 0.01, 0.05, Inf),
  #                                             labels = c("'****'", "'***'", "'**'", "'*'", "'ns'")), 
  #                           sep = "~")), method = "pearson", label.x = -2, label.y = 1.4) +
  stat_cor(method = "pearson", label.x = -1.5, label.y = 1.5) +
  ylim(0, 1.5)+
  # xlim(-3, 2)+
  # geom_text(aes(label  = Strain)) +
  scale_colour_manual(values = isomapTolOTU_new_single_level_cols_Family) +
  theme_bw() +
  facet_grid(~compartment, scales = "free") +
  labs(y = "TI",
       x = "log2FC",
       color = "Family",
       title = expression("TI"["APO"]* "~ root & rhizosphere microbiome"))

fIGsupp.TIAPO_cor_logFC_field_root_rhizo

# ggsave(plot = fIGsupp.TIAPO_cor_logFC_field_root_rhizo,  filename = "fIGsupp.TIAPO_cor_logFC_field_root_rhizo.pdf", width = 6, height = 3, dpi = 300, scale = 3, units = "cm")
# 
# ggsave(plot = fIGsupp.TIAPO_cor_logFC_field_root_rhizo,  filename = "fIGsupp.TIAPO_cor_logFC_field_root_rhizo.svg", width = 6, height = 3, dpi = 300, scale = 3, units = "cm")
```

*DIMBOA-Glc*

```{r, message=FALSE, echo=FALSE, error=FALSE, warning = FALSE, fig.dim = c(6, 5)}
fIGsupp.TIDMG_cor_logFC_field_root_rhizo <- isomapTolOTU_new_mean_LFC %>% 
  # filter(OTU %in% selected_OTU) %>% 
  mutate(compartment = gsub("rhizo", "rhizosphere", compartment)) %>% 
  mutate(compartment = factor(compartment, levels = c("root", "rhizosphere"))) %>% 
  unique() %>% 
  ggplot(aes(y = TI_norm_DMG_mean, x = log2FC_mean)) + 
  geom_point(aes(color = Family), size = 2, show.legend = FALSE) +
  geom_smooth(method = "lm", color="black", formula = y ~ x)  + 
  # geom_text(aes(label = OTU))+
  # stat_cor(aes(label =paste(..r.label.., cut(..p.., 
  #                                             breaks = c(-Inf, 0.0001, 0.001, 0.01, 0.05, Inf),
  #                                             labels = c("'****'", "'***'", "'**'", "'*'", "'ns'")), 
  #                           sep = "~")), method = "pearson", label.x = -2, label.y = 1.4) +
  stat_cor(method = "pearson", label.x = -1.5, label.y = 1.5) +
  ylim(0, 1.5)+
  # xlim(-3, 2)+
  # geom_text(aes(label  = Strain)) +
  scale_colour_manual(values = isomapTolOTU_new_single_level_cols_Family) +
  theme_bw() +
  facet_grid(~compartment, scales = "free") +
  labs(y = "TI",
       x = "log2FC",
       color = "Family",
       title = expression("TI"["DIMBOA-Glc"]* "~ root & rhizosphere microbiome"))


fIGsupp.TIDMG_cor_logFC_field_root_rhizo

# ggsave(plot = fIGsupp.TIDMG_cor_logFC_field_root_rhizo,  filename = "fIGsupp.TIDMG_cor_logFC_field_root_rhizo.pdf", width = 6, height = 3, dpi = 300, scale = 3, units = "cm")
# # 
# ggsave(plot = fIGsupp.TIDMG_cor_logFC_field_root_rhizo,  filename = "fIGsupp.TIDMG_cor_logFC_field_root_rhizo.svg", width = 6, height = 3, dpi = 300, scale = 3, units = "cm")
```


DIMBOA-Glc

```{r, message=FALSE, echo=FALSE, error=FALSE, warning = FALSE, fig.dim = c(6, 5)}
# fIGsupp.TIDMG_cor_logFC_field_root_rhizo <- isomapTolOTU_new_mean_LFC %>% 
#   # filter(OTU %in% selected_OTU) %>% 
#   mutate(compartment = gsub("rhizo", "rhizosphere", compartment)) %>% 
#   mutate(compartment = factor(compartment, levels = c("root", "rhizosphere"))) %>% 
#   unique() %>% 
#   ggplot(aes(y = AUC_norm_DMG_mean, x = log2FC_mean)) + 
#   geom_point(aes(color = Family), size = 2, show.legend = FALSE) +
#   geom_smooth(method = "lm", color="black", formula = y ~ x)  + 
#   # geom_text(aes(label = OTU))+
#   # stat_cor(aes(label =paste(..r.label.., cut(..p.., 
#   #                                             breaks = c(-Inf, 0.0001, 0.001, 0.01, 0.05, Inf),
#   #                                             labels = c("'****'", "'***'", "'**'", "'*'", "'ns'")), 
#   #                           sep = "~")), method = "pearson", label.x = -2, label.y = 1.4) +
#   stat_cor(method = "pearson", label.x = -1.5, label.y = 1.5) +
#   ylim(0, 1.5)+
#   # xlim(-3, 2)+
#   # geom_text(aes(label  = Strain)) +
#   scale_colour_manual(values = isomapTolOTU_new_single_level_cols_Family) +
#   theme_bw() +
#   facet_grid(~compartment, scales = "free") +
#   labs(y = "AUC",
#        x = "log2FC",
#        color = "Family",
#        title = expression("TI"["DIMBOA-Glc"]* "~ root & rhizosphere microbiome"))
# 
# 
# fIGsupp.TIDMG_cor_logFC_field_root_rhizo

# ggsave(plot = fIGsupp.TIDMG_cor_logFC_field_root_rhizo,  filename = "fIGsupp.TIDMG_cor_logFC_field_root_rhizo.pdf", width = 6, height = 3, dpi = 300, scale = 3, units = "cm")
# 
# ggsave(plot = fIGsupp.TIDMG_cor_logFC_field_root_rhizo,  filename = "fIGsupp.TIDMG_cor_logFC_field_root_rhizo.svg", width = 6, height = 3, dpi = 300, scale = 3, units = "cm")
```


## Corplot log2FC field

```{r, message=FALSE, echo=FALSE, error=FALSE, warning = FALSE, include = FALSE}
# DIMBOA-Glc
cor.DMG <- isomapTolOTU_new_mean_LFC  %>% 
  dplyr::select(TI_norm_DMG_mean, log2FC_mean, compartment) %>% 
  # filter(TI_norm_MBOA_mean != "NA") %>%
  # filter(TI_norm_MBOA_mean < 1.2) %>% 
  # filter(!log2FC_mean %in% c(-Inf, Inf, NaN, NA)) %>% 
  mutate(compartment_soil = interaction(compartment)) %>% 
  unique() %>% 
  nest(data = -compartment) %>% 
  # filter(!location_background_compartment %in% NA) %>% 
  mutate(
    test = map(data, ~ cor.test(.x$TI_norm_DMG_mean, .x$log2FC_mean, method = "pearson")), # S3 list-col
    tidied = map(test, tidy)
  ) %>% 
  unnest(tidied)  %>% 
  mutate(Compound = "DIMBOA-Glc")

# MBOA
cor.MBOA <- isomapTolOTU_new_mean_LFC  %>% 
  dplyr::select(TI_norm_MBOA_mean, log2FC_mean, compartment) %>% 
  # filter(TI_norm_MBOA_mean != "NA") %>%
  # filter(TI_norm_MBOA_mean < 1.2) %>% 
  # filter(!log2FC_mean %in% c(-Inf, Inf, NaN, NA)) %>% 
  mutate(compartment_soil = interaction(compartment)) %>% 
  unique() %>% 
  nest(data = -compartment) %>% 
  # filter(!location_background_compartment %in% NA) %>% 
  mutate(
    test = map(data, ~ cor.test(.x$TI_norm_MBOA_mean, .x$log2FC_mean, method = "pearson")), # S3 list-col
    tidied = map(test, tidy)
  ) %>% 
  unnest(tidied)  %>% 
  mutate(Compound = "MBOA")

# BOA
cor.BOA <- isomapTolOTU_new_mean_LFC  %>% 
  dplyr::select(Strain, TI_norm_BOA_mean, log2FC_mean, compartment) %>% 
  filter(TI_norm_BOA_mean != "NA") %>%
  filter(TI_norm_BOA_mean < 1.5) %>% 
  # filter(!log2FC_mean %in% c(-Inf, Inf, NaN, NA)) %>% 
  mutate(compartment_soil = interaction(compartment)) %>% 
  unique() %>% 
  nest(data = -compartment) %>% 
  # filter(!location_background_compartment %in% NA) %>% 
  mutate(
    test = map(data, ~ cor.test(.x$TI_norm_BOA_mean, .x$log2FC_mean)), # S3 list-col
    tidied = map(test, tidy)
  ) %>% 
  unnest(tidied)  %>% 
  mutate(Compound = "BOA")


# AMPO
cor.AMPO <- isomapTolOTU_new_mean_LFC  %>% 
  dplyr::select(Strain, TI_norm_AMPO_mean, log2FC_mean, compartment) %>% 
  filter(TI_norm_AMPO_mean != "NA") %>%
  filter(TI_norm_AMPO_mean < 1.5) %>% 
  filter(!log2FC_mean %in% c(-Inf, Inf, NaN, NA)) %>% 
  mutate(compartment_soil = interaction(compartment)) %>% 
  unique() %>% 
  nest(data = -compartment) %>% 
  # filter(!location_background_compartment %in% NA) %>% 
  mutate(
    test = map(data, ~ cor.test(.x$TI_norm_AMPO_mean, .x$log2FC_mean)), # S3 list-col
    tidied = map(test, tidy)
  ) %>% 
  unnest(tidied)  %>% 
  mutate(Compound = "AMPO")

# APO
cor.APO <- isomapTolOTU_new_mean_LFC  %>% 
  dplyr::select(Strain, TI_norm_APO_mean, log2FC_mean, compartment) %>% 
  filter(TI_norm_APO_mean != "NA") %>%
  filter(TI_norm_APO_mean < 1.5) %>% 
  filter(!log2FC_mean %in% c(-Inf, Inf, NaN, NA)) %>% 
  mutate(compartment_soil = interaction(compartment)) %>% 
  unique() %>% 
  nest(data = -compartment) %>% 
  # filter(!location_background_compartment %in% NA) %>% 
  mutate(
    test = map(data, ~ cor.test(.x$TI_norm_APO_mean, .x$log2FC_mean)), # S3 list-col
    tidied = map(test, tidy)
  ) %>% 
  unnest(tidied)  %>% 
  mutate(Compound = "APO")

cor.all.compounds_new_field <- rbind(cor.DMG, cor.MBOA, cor.BOA, cor.AMPO, cor.APO) %>% 
  add_significance("p.value")
```

```{r, message=FALSE, echo=FALSE, error=FALSE, warning = FALSE, fig.dim = c(3,3)}
cor.all.compounds_new_field_Fig4B <- cor.all.compounds_new_field %>% 
  mutate(Compound = factor(Compound, levels = c("APO", "BOA", "AMPO", "MBOA", "DIMBOA-Glc"))) %>% 
  ggplot(aes(x = compartment, y = Compound))+
  geom_tile(aes(fill = estimate))+
  geom_text(aes(label = p.value.signif))+
  scale_fill_gradient2(low = "darkblue", mid = "white",   high = "darkred", na.value = "white", midpoint = 0) +
  theme_bw()+
  scale_x_discrete(labels= c("root", "rhizosphere"))+
  labs(y = " ", 
       x = " ",
       fill = "correlation")


cor.all.compounds_new_field_Fig4B

# ggsave(plot = cor.all.compounds_new_field_Fig4B,  filename = "cor.all.compounds_new_field_Fig4B.pdf", width = 7.7, height = 7, dpi = 300, scale = 1, units = "cm")
#
# ggsave(plot = cor.all.compounds_new_field_Fig4B,  filename = "cor.all.compounds_new_field_Fig4B.svg", width = 7.7, height = 7, dpi = 300, scale = 1, units = "cm")

```

```{r, message=FALSE, echo=FALSE, error=FALSE, warning = FALSE, fig.dim = c(3, 2)}
cor.all.compounds_new_field %>% 
  filter(Compound %in% c("MBOA", "AMPO")) %>% 
  # mutate(location_background_compartment = factor(location_background_compartment, levels = c("Changins.B73.root", "Changins.B73.rhizo",  "Zurich.B73.root", "Zurich.B73.rhizo", "Zurich.W22.root", "Zurich.W22.rhizo", "Ithaca.W22.root", "Ithaca.W22.rhizo", "Sheffield.W22.root"))) %>% 
  mutate(Compound = factor(Compound, levels = c("APO", "BOA", "AMPO", "MBOA"))) %>% 
  # filter(!location_background_compartment %in% NA) %>% 
  # mutate(Compound = gsub("DMG","DIMBOA-Glc", Compound)) %>% 
  ggplot(aes(x = compartment, y = Compound))+
  geom_tile(aes(fill = estimate))+
  geom_text(aes(label = p.value.signif))+
  #geom_text(aes(label = COR))+
  scale_fill_gradient2(low = "darkblue", mid = "white",   high = "darkred", na.value = "white", midpoint = 0) +
  theme_bw()+
  scale_x_discrete(labels= c("root", "rhizosphere"))+
  # theme(axis.text.x = element_text(angle = 90, hjust = 0)) + 
  labs(y = " ", 
       x = " ",
       fill = "correlation")
```

```{r}
sessionInfo()
```


